// @wix/advanced-seo-utils/dist/src/private/item-types/bookings-form/adapter-bookings-form.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.IDs = void 0;
var utils_1 = require("../../adapters/utils");
var ContextProps_1 = require("../../types/ContextProps");
var get_1 = require("../../utils/get");
var sanitize_url_1 = require("../../utils/sanitize-url");
exports.IDs = {
    FORM_NAME: 'bookings-form.form-name',
    FORM_DESCRIPTION: 'bookings-form.form-description',
    PAGE_URL: 'bookings-form.page-url',
};
var SchemaTypes = {};
var userVisible = [];
exports.default = {
    IDs: exports.IDs,
    SchemaTypes: SchemaTypes,
    getData: function(item) {
        var _a;
        return _a = {},
            _a[exports.IDs.FORM_NAME] = get_1.get(item, 'form.name'),
            _a[exports.IDs.FORM_DESCRIPTION] = get_1.get(item, 'form.description'),
            _a[exports.IDs.PAGE_URL] = sanitize_url_1.sanitizeUrl(get_1.get(item, "context." + ContextProps_1.CONTEXT_PROPS.DEFAULT_URL)),
            _a;
    },
    getSdStatus: function() {},
    getLegacySeoBlob: function() {},
    getSlug: function() {
        return '';
    },
    updateItemDataWithSlug: function(item) {
        return item;
    },
    getBiData: function() {},
    getKeys: function() {
        return utils_1.getKeys(exports.IDs, userVisible);
    },
    getSdKeys: function() {
        return [];
    },
};
