// @wix/advanced-seo-utils/dist/src/private/item-types/events-page/schema-presets/events-page.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
var adapter_events_page_1 = __importDefault(require("../adapter-events-page"));
var IDs = adapter_events_page_1.default.IDs,
    SchemaTypes = adapter_events_page_1.default.SchemaTypes;
exports.default = {
    type: SchemaTypes.EVENTS_PAGE,
    schema: {
        '@context': 'https://schema.org',
        '@type': 'Event',
        name: "{{" + IDs.NAME + "}}",
        description: "{{" + IDs.DESCRIPTION + "}}",
        startDate: "{{" + IDs.START_DATE + "}}",
        endDate: "{{" + IDs.END_DATE + "}}",
        eventStatus: "{{" + IDs.SD_STATUS + "}}",
        eventAttendanceMode: "{{" + IDs.SD_ATTANDANCE_MODE + "}}",
        location: "{{" + IDs.SD_JSON_LOCATION + "}}",
        offers: "{{" + IDs.SD_JSON_OFFERS + "}}",
        image: {
            '@type': 'ImageObject',
            url: "{{" + IDs.CALCULATED_MAIN_IMAGE_URL + "}}",
            width: "{{" + IDs.IMAGE_WIDTH + "}}",
            height: "{{" + IDs.IMAGE_HEIGHT + "}}",
        },
    },
};
