// @wix/bookings-adapter-ooi-wix-sdk/dist/src/AppSettingsClientAdapter.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.AppSettingsClientAdapter = void 0;
var tslib_1 = require("tslib");
var AppSettingsClientAdapter = /** @class */ (function() {
    function AppSettingsClientAdapter(_a) {
        var appDefId = _a.appDefId,
            instanceId = _a.instanceId,
            externalId = _a.externalId;
        this.appDefId = appDefId;
        this.instanceId = instanceId;
        this.externalId = externalId;
    }
    AppSettingsClientAdapter.prototype.getAppDefId = function() {
        return this.appDefId;
    };
    AppSettingsClientAdapter.prototype.getInstanceId = function() {
        return this.instanceId;
    };
    AppSettingsClientAdapter.prototype.getExternalId = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                return [2 /*return*/ , this.externalId];
            });
        });
    };
    AppSettingsClientAdapter.prototype.setExternalId = function(id) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                return [2 /*return*/ ];
            });
        });
    };
    AppSettingsClientAdapter.prototype.triggerSettingsUpdated = function(data, scope) {};
    AppSettingsClientAdapter.prototype.onSettingsUpdated = function(scope, cb) {};
    return AppSettingsClientAdapter;
}());
exports.AppSettingsClientAdapter = AppSettingsClientAdapter;
