// @wix/bookings-adapter-ooi-wix-sdk/dist/src/WixOOISDKAdapter.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.WixOOISDKAdapter = exports.PREVIEW_MODAL_DIMENSIONS = exports.DEFAULT_MODAL_DIMENSIONS = exports.BookingsQueryParams = void 0;
var tslib_1 = require("tslib");
var bookings_const_1 = require("./bookings.const");
var bookings_uou_types_1 = require("@wix/bookings-uou-types");
var BookingsQueryParams;
(function(BookingsQueryParams) {
    BookingsQueryParams["REFERRAL"] = "referral";
    BookingsQueryParams["STAFF"] = "staff";
    BookingsQueryParams["LOCATION"] = "location";
    BookingsQueryParams["BOOKING_ID"] = "bookingId";
    BookingsQueryParams["SERVICE_ID"] = "service";
    BookingsQueryParams["AVAILABILITY_SLOT"] = "slot";
    BookingsQueryParams["TIMEZONE"] = "timezone";
    BookingsQueryParams["FILLED_FIELDS"] = "filledFields";
    BookingsQueryParams["REFERRER"] = "referrer";
})(BookingsQueryParams = exports.BookingsQueryParams || (exports.BookingsQueryParams = {}));
exports.DEFAULT_MODAL_DIMENSIONS = {
    height: 380,
    width: 580
};
exports.PREVIEW_MODAL_DIMENSIONS = {
    height: 600,
    width: 600
};
var WixOOISDKAdapter = /** @class */ (function() {
    function WixOOISDKAdapter(wixCodeApi, platformAPIs, appParams, compId) {
        var _this = this;
        this.wixCodeApi = wixCodeApi;
        this.platformAPIs = platformAPIs;
        this.appParams = appParams;
        this.compId = compId;
        this.navigateToMembersArea = function() {
            return (0, tslib_1.__awaiter)(_this, void 0, void 0, function() {
                var navigateToSection;
                return (0, tslib_1.__generator)(this, function(_a) {
                    switch (_a.label) {
                        case 0:
                            return [4 /*yield*/ , this.getMembersAreaAPI()];
                        case 1:
                            navigateToSection = (_a.sent()).navigateToSection;
                            return [4 /*yield*/ , navigateToSection({
                                appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                                sectionId: bookings_const_1.MEMBERS_AREA_SECTION_ID,
                            })];
                        case 2:
                            _a.sent();
                            return [2 /*return*/ ];
                    }
                });
            });
        };
        this.sessionInstanceKey = compId + "_instance";
        this.currentLanguage = this.getCurrentLanguageFromPlatform();
    }
    WixOOISDKAdapter.prototype.openWaitingListModal = function(serviceType) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var modalType;
            return (0, tslib_1.__generator)(this, function(_a) {
                modalType = 'waitlist';
                return [2 /*return*/ , this.openModal(modalType, serviceType, exports.DEFAULT_MODAL_DIMENSIONS)];
            });
        });
    };
    WixOOISDKAdapter.prototype.openUoUPremiumModal = function(serviceType, referralInfo) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var modalType;
            return (0, tslib_1.__generator)(this, function(_a) {
                modalType = 'uou_not_bookable_modal';
                return [2 /*return*/ , this.openModal(modalType, serviceType, exports.DEFAULT_MODAL_DIMENSIONS, referralInfo)];
            });
        });
    };
    WixOOISDKAdapter.prototype.openPreviewPremiumModal = function(serviceType, referralInfo) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var modalType;
            return (0, tslib_1.__generator)(this, function(_a) {
                modalType = 'preview_premium';
                return [2 /*return*/ , this.openModal(modalType, serviceType, exports.PREVIEW_MODAL_DIMENSIONS, referralInfo)];
            });
        });
    };
    WixOOISDKAdapter.prototype.openModal = function(modalType, serviceType, dimensions, referralInfo) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var baseUrl, modalUrl, referralInfoParam;
            return (0, tslib_1.__generator)(this, function(_a) {
                baseUrl = 'https://bookings.wixapps.net/_api/bookings-viewer';
                modalUrl = baseUrl + "/viewer-modals";
                referralInfoParam = referralInfo ?
                    "&referralInfo=" + referralInfo :
                    "";
                return [2 /*return*/ , this.wixCodeApi.window.openModal(modalUrl + "?deviceType=" + this.getDeviceType() + "&instance=" + this.getInstance() + "&locale=" + this.wixCodeApi.site.language + "&modal=" + modalType + "&serviceType=" + serviceType + referralInfoParam, (0, tslib_1.__assign)((0, tslib_1.__assign)({}, dimensions), {
                    theme: 'BARE'
                }))];
            });
        });
    };
    WixOOISDKAdapter.prototype.getCurrentUser = function() {
        return this.wixCodeApi.user.currentUser;
    };
    WixOOISDKAdapter.prototype.promptUserLogin = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                return [2 /*return*/ , this.wixCodeApi.user.promptLogin({
                    mode: 'login',
                    lang: this.currentLanguage,
                })];
            });
        });
    };
    WixOOISDKAdapter.prototype.onUserLogin = function(callback) {
        this.wixCodeApi.user.onLogin(callback);
    };
    WixOOISDKAdapter.prototype.isPricingPlanInstalled = function() {
        if (this.wixCodeApi.site) {
            if (this.wixCodeApi.site.isAppSectionInstalled) {
                return Promise.resolve(this.wixCodeApi.site.isAppSectionInstalled({
                    appDefinitionId: bookings_const_1.PRICING_PLAN_APP_DEF_ID,
                    sectionId: bookings_const_1.PRICING_PLAN_SECTION_ID,
                }));
            }
            // TODO: remove when wixCodeApi.site.getAppToken is added to viewerWrapper
            if (this.wixCodeApi.site.getAppToken) {
                return Promise.resolve(!!this.wixCodeApi.site.getAppToken(bookings_const_1.PRICING_PLAN_APP_DEF_ID));
            }
        }
        return this.isPricingPlanInstalledLegacyMode();
    };
    WixOOISDKAdapter.prototype.isPricingPlanInstalledLegacyMode = function() {
        return new Promise(function(resolve) {
            if (window.Wix.Utils.getViewMode() !== 'standalone') {
                window.Wix.isAppSectionInstalled(bookings_const_1.PRICING_PLAN_SECTION_ID, {
                    appDefinitionId: bookings_const_1.PRICING_PLAN_APP_DEF_ID
                }, function(isInstalled) {
                    resolve(isInstalled);
                });
            } else {
                resolve(false);
            }
        });
    };
    WixOOISDKAdapter.prototype.getPricingPlanUrl = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var fullPricingPlanUrl, baseUrl, urlParts;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.wixCodeApi.site.getSectionUrl({
                            sectionId: bookings_const_1.PRICING_PLAN_SECTION_ID,
                            appDefinitionId: bookings_const_1.PRICING_PLAN_APP_DEF_ID,
                        })];
                    case 1:
                        fullPricingPlanUrl = (_a.sent()).url;
                        baseUrl = this.wixCodeApi.location.baseUrl;
                        if (baseUrl) {
                            urlParts = fullPricingPlanUrl.split(baseUrl);
                            return [2 /*return*/ , urlParts.length > 1 ? urlParts[1] : fullPricingPlanUrl];
                        }
                        return [2 /*return*/ , fullPricingPlanUrl];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToPricingPlanPreview = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var pricingPlanUrl;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.getPricingPlanUrl()];
                    case 1:
                        pricingPlanUrl = _a.sent();
                        this.wixCodeApi.location.to(pricingPlanUrl);
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToPricingPlan = function(_a) {
        var redirectTo = _a.redirectTo,
            planIds = _a.planIds,
            maxStartDate = _a.maxStartDate,
            queryParams = _a.queryParams;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var pricingPlanUrl, callbackData, pricingPlanSectionParams;
            return (0, tslib_1.__generator)(this, function(_b) {
                switch (_b.label) {
                    case 0:
                        return [4 /*yield*/ , this.getPricingPlanUrl()];
                    case 1:
                        pricingPlanUrl = _b.sent();
                        callbackData = btoa(JSON.stringify({
                            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                            sectionId: redirectTo.sectionId,
                            state: redirectTo.relativePath,
                            queryParams: queryParams,
                        }));
                        pricingPlanSectionParams = encodeURI(JSON.stringify({
                            navigateToSectionProps: callbackData,
                            planIds: planIds.join(','),
                            maxStartDate: maxStartDate,
                        }));
                        this.wixCodeApi.location.to(pricingPlanUrl + "?appSectionParams=" + pricingPlanSectionParams);
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.isPreviewMode = function() {
        return this.getViewMode() === 'preview';
    };
    WixOOISDKAdapter.prototype.isEditorMode = function() {
        return this.getViewMode() === 'editor';
    };
    WixOOISDKAdapter.prototype.isSiteMode = function() {
        return this.getViewMode() === 'site';
    };
    WixOOISDKAdapter.prototype.isDemoMode = function() {
        return this.getDecodedInstance().demoMode;
    };
    WixOOISDKAdapter.prototype.isTemplateMode = function() {
        return this.getDecodedInstance().siteIsTemplate;
    };
    WixOOISDKAdapter.prototype.setContainerDimensions = function(newWidth, newHeight) {
        if (this.isRunningInIframe() && this.isEditorMode()) {
            window.Wix.resizeComponent({
                width: newWidth,
                height: newHeight,
            });
        }
    };
    WixOOISDKAdapter.prototype.setContainerHeight = function(newHeight) {
        if (this.isRunningInIframe()) {
            window.Wix.setHeight(newHeight);
        }
    };
    WixOOISDKAdapter.prototype.getExternalId = function() {
        var _this = this;
        return new Promise(function(resolve, reject) {
            if (_this.isRunningInIframe()) {
                window.Wix.getExternalId(function(externalId) {
                    return resolve(externalId);
                });
            } else {
                reject('should not be called in OOI mode');
            }
        });
    };
    WixOOISDKAdapter.prototype.getViewMode = function() {
        return this.wixCodeApi.window.viewMode.toLowerCase();
    };
    WixOOISDKAdapter.prototype.getCurrentLanguage = function() {
        return this.currentLanguage;
    };
    WixOOISDKAdapter.prototype.getRegionalSettings = function() {
        return this.wixCodeApi.site.regionalSettings;
    };
    WixOOISDKAdapter.prototype.setCurrentLanguage = function(locale) {
        this.currentLanguage = this.getCurrentLanguageFromLocale(locale);
    };
    WixOOISDKAdapter.prototype.getCurrentLanguageFromPlatform = function() {
        return this.getCurrentLanguageFromLocale(this.wixCodeApi.site.language);
    };
    WixOOISDKAdapter.prototype.getCurrentLanguageFromLocale = function(locale) {
        var localeToSubstring = locale ? locale.substring(0, 2) : undefined;
        return localeToSubstring || 'en';
    };
    WixOOISDKAdapter.prototype.isRunningInIframe = function() {
        return !(typeof WorkerGlobalScope !== 'undefined' &&
            self instanceof WorkerGlobalScope);
    };
    WixOOISDKAdapter.prototype.getVisitorId = function() {
        return this.platformAPIs.bi.visitorId;
    };
    WixOOISDKAdapter.prototype.getMetaSiteId = function() {
        return this.platformAPIs.bi.metaSiteId;
    };
    WixOOISDKAdapter.prototype.getOwnerId = function() {
        return this.platformAPIs.bi.ownerId;
    };
    WixOOISDKAdapter.prototype.getMemberId = function() {
        return this.wixCodeApi.user.currentUser.id;
    };
    WixOOISDKAdapter.prototype.getBiToken = function() {
        return this.platformAPIs.bi.biToken;
    };
    WixOOISDKAdapter.prototype.getInstanceId = function() {
        return this.appParams.instanceId;
    };
    WixOOISDKAdapter.prototype.getInstance = function() {
        return (this.wixCodeApi.site.getAppToken(bookings_const_1.BOOKINGS_APP_DEF_ID) ||
            this.appParams.instance);
    };
    WixOOISDKAdapter.prototype.getAppDefId = function() {
        return this.appParams.appDefinitionId;
    };
    WixOOISDKAdapter.prototype.getStaticsBaseUrl = function() {
        return this.appParams.baseUrls && this.appParams.baseUrls.staticsBaseUrl;
    };
    WixOOISDKAdapter.prototype.getPlatformAPIs = function() {
        return this.platformAPIs;
    };
    WixOOISDKAdapter.prototype.getMembersAreaAPI = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var api;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        if (!!this.membersAreaApi) return [3 /*break*/ , 2];
                        return [4 /*yield*/ , this.wixCodeApi.site.getPublicAPI(bookings_const_1.SANTA_MEMBERS_APP_ID)];
                    case 1:
                        api = _a.sent();
                        this.membersAreaApi = api;
                        _a.label = 2;
                    case 2:
                        return [2 /*return*/ , this.membersAreaApi];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getServiceListStaticsBaseUrl = function() {
        return (this.appParams.baseUrls &&
            this.appParams.baseUrls.serviceListStaticsBaseUrl);
    };
    WixOOISDKAdapter.prototype.getPlatformAppStaticsBaseUrl = function() {
        return (this.appParams.baseUrls && this.appParams.baseUrls.platformAppsBaseUrl);
    };
    WixOOISDKAdapter.prototype.getServerBaseUrl = function() {
        return this.isSSR() && this.appParams.baseUrls ?
            this.appParams.baseUrls.serverBaseUrl :
            '/';
    };
    WixOOISDKAdapter.prototype.isOwner = function() {
        return (this.getOwnerId() === this.getVisitorId() ||
            this.getOwnerId() === this.getMemberId());
    };
    WixOOISDKAdapter.prototype.trackEvent = function(evid, payload) {
        this.wixCodeApi.window.trackEvent(evid, payload);
    };
    WixOOISDKAdapter.prototype.trackAnalytics = function(_a) {
        var eventId = _a.eventId,
            payload = _a.payload;
        try {
            if (!this.isEditorMode()) {
                this.trackEvent(eventId, payload);
            }
        } catch (e) {
            console.log(e);
        }
    };
    WixOOISDKAdapter.prototype.getSiteRevision = function() {
        return this.wixCodeApi.site.revision;
    };
    WixOOISDKAdapter.prototype.getSiteTimezone = function() {
        return this.wixCodeApi.site.timezone;
    };
    WixOOISDKAdapter.prototype.getInstanceValue = function(key) {
        var decodedInstance = this.getDecodedInstance();
        if (decodedInstance) {
            return decodedInstance[key] || null;
        }
        return null;
    };
    WixOOISDKAdapter.prototype.getDeviceType = function() {
        return this.wixCodeApi.window.formFactor;
    };
    WixOOISDKAdapter.prototype.getBookingsRelativeUrl = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var isBookingCheckoutInstalled, fullBookingsUrl, baseUrl, urlParts;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.isBookCheckoutInstalled()];
                    case 1:
                        isBookingCheckoutInstalled = _a.sent();
                        return [4 /*yield*/ , this.wixCodeApi.site.getSectionUrl({
                            sectionId: isBookingCheckoutInstalled ?
                                bookings_const_1.BOOKINGS_CHECKOUT_SECTION_ID :
                                bookings_const_1.BOOKINGS_SCHEDULER_SECTION_ID,
                            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                        })];
                    case 2:
                        fullBookingsUrl = (_a.sent()).url;
                        baseUrl = this.wixCodeApi.location.baseUrl;
                        if (baseUrl) {
                            urlParts = fullBookingsUrl.split(baseUrl);
                            return [2 /*return*/ , urlParts.length > 1 ? urlParts[1] : fullBookingsUrl];
                        }
                        return [2 /*return*/ , fullBookingsUrl];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getServicePageRelativeUrl = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var isServicePageInstalled;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.isServicePageInstalled()];
                    case 1:
                        isServicePageInstalled = _a.sent();
                        if (isServicePageInstalled) {
                            return [2 /*return*/ , this.getSectionRelativeUrl(bookings_const_1.BOOKINGS_SERVICE_PAGE_SECTION_ID)];
                        }
                        return [2 /*return*/ , this.getBookingsRelativeUrl()];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getBookingCalendarRelativeUrl = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var isBookingCalendarInstalled;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.isBookingCalendarInstalled()];
                    case 1:
                        isBookingCalendarInstalled = _a.sent();
                        if (isBookingCalendarInstalled) {
                            return [2 /*return*/ , this.getSectionRelativeUrl(bookings_const_1.BOOKINGS_CALENDAR_SECTION_ID)];
                        }
                        return [2 /*return*/ , this.getBookingsRelativeUrl()];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getSectionAbsoluteUrl = function(sectionId) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var baseUrl, _a;
            return (0, tslib_1.__generator)(this, function(_b) {
                switch (_b.label) {
                    case 0:
                        baseUrl = this.wixCodeApi.location.baseUrl;
                        _a = baseUrl;
                        return [4 /*yield*/ , this.getSectionRelativeUrl(bookings_const_1.BOOKINGS_FORM_SECTION_ID)];
                    case 1:
                        return [2 /*return*/ , (_a + (_b.sent()))];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getSectionRelativeUrl = function(sectionId) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var fullBookingsUrl, baseUrl, urlParts;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.wixCodeApi.site.getSectionUrl({
                            sectionId: sectionId,
                            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                        })];
                    case 1:
                        fullBookingsUrl = (_a.sent()).url;
                        baseUrl = this.wixCodeApi.location.baseUrl;
                        if (baseUrl) {
                            urlParts = fullBookingsUrl.split(baseUrl);
                            return [2 /*return*/ , urlParts.length > 1 ? urlParts[1] : fullBookingsUrl];
                        }
                        return [2 /*return*/ , fullBookingsUrl];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToServiceList = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var bookOnlineRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.getBookOnlineRelativeUrl()];
                    case 1:
                        bookOnlineRelativeUrl = _a.sent();
                        this.wixCodeApi.location.to(bookOnlineRelativeUrl);
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsWithSuffix = function(suffix) {
        if (suffix === void 0) {
            suffix = '';
        }
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var bookingsRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.getBookingsRelativeUrl()];
                    case 1:
                        bookingsRelativeUrl = _a.sent();
                        this.wixCodeApi.location.to(bookingsRelativeUrl + suffix);
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsServicePage = function(serviceSlug, _a) {
        var _b = _a === void 0 ? {} : _a,
            referral = _b.referral,
            location = _b.location;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var suffix, query, servicePageRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_c) {
                switch (_c.label) {
                    case 0:
                        suffix = "" + serviceSlug;
                        query = this.queryBuilder({
                            referral: referral,
                            location: location
                        });
                        return [4 /*yield*/ , this.getServicePageRelativeUrl()];
                    case 1:
                        servicePageRelativeUrl = _c.sent();
                        this.wixCodeApi.location.to(servicePageRelativeUrl + "/" + (suffix + query));
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsCalendarPage = function(serviceSlug, navigationContext) {
        if (navigationContext === void 0) {
            navigationContext = {};
        }
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                return [2 /*return*/ , this.navigateToBookingsBookAction(true, serviceSlug, navigationContext)];
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsBookAction = function(isSlotSelectionNeeded, serviceSlug, _a) {
        var _b = _a === void 0 ? {} : _a,
            referral = _b.referral,
            staff = _b.staff,
            location = _b.location,
            timezone = _b.timezone;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var suffix, _c, query, bookingCalendarRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_d) {
                switch (_d.label) {
                    case 0:
                        _c = isSlotSelectionNeeded;
                        if (!_c) return [3 /*break*/ , 2];
                        return [4 /*yield*/ , this.isBookingCalendarInstalled()];
                    case 1:
                        _c = (_d.sent());
                        _d.label = 2;
                    case 2:
                        suffix = _c ? "/" + serviceSlug :
                            "/" + serviceSlug + "/book";
                        query = this.queryBuilder({
                            referral: referral,
                            staff: staff,
                            location: location,
                            timezone: timezone
                        });
                        return [4 /*yield*/ , (isSlotSelectionNeeded ?
                            this.getBookingCalendarRelativeUrl() :
                            this.getBookingsRelativeUrl())];
                    case 3:
                        bookingCalendarRelativeUrl = _d.sent();
                        this.wixCodeApi.location.to("" + bookingCalendarRelativeUrl + (suffix + query));
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.queryBuilder = function(queryParams) {
        var currentParams = this.getUrlQueryParams();
        var params = [];
        for (var _i = 0, _a = Object.entries((0, tslib_1.__assign)((0, tslib_1.__assign)({}, currentParams), queryParams)); _i < _a.length; _i++) {
            var _b = _a[_i],
                key = _b[0],
                value = _b[1];
            if (value) {
                params.push([key + "=" + value]);
            }
        }
        return params.length ? "?" + params.join('&') : '';
    };
    WixOOISDKAdapter.prototype.navigateToBookingsFormPage = function(_a) {
        var _b, _c;
        var slotAvailability = _a.slotAvailability,
            serviceId = _a.serviceId,
            serviceSlug = _a.serviceSlug,
            referral = _a.referral,
            timezone = _a.timezone;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var formUrl, sessionDetails;
            return (0, tslib_1.__generator)(this, function(_d) {
                switch (_d.label) {
                    case 0:
                        return [4 /*yield*/ , this.isBookingFormInstalled()];
                    case 1:
                        if (!_d.sent()) return [3 /*break*/ , 3];
                        return [4 /*yield*/ , this.getSectionRelativeUrl(bookings_const_1.BOOKINGS_FORM_SECTION_ID)];
                    case 2:
                        formUrl = _d.sent();
                        this.removeFromSessionStorage(BookingsQueryParams.AVAILABILITY_SLOT);
                        this.removeFromSessionStorage(BookingsQueryParams.SERVICE_ID);
                        this.removeFromSessionStorage(BookingsQueryParams.TIMEZONE);
                        slotAvailability &&
                            this.setToSessionStorage(BookingsQueryParams.AVAILABILITY_SLOT, JSON.stringify(slotAvailability));
                        serviceId &&
                            this.setToSessionStorage(BookingsQueryParams.SERVICE_ID, JSON.stringify(serviceId));
                        timezone &&
                            this.setToSessionStorage(BookingsQueryParams.TIMEZONE, JSON.stringify(timezone));
                        return [2 /*return*/ , this.wixCodeApi.location.to("" + formUrl + this.queryBuilder({
                            referral: referral
                        }))];
                    case 3:
                        if (slotAvailability) {
                            sessionDetails = {
                                id: slotAvailability.slot.sessionId,
                                slug: serviceSlug,
                                scheduleId: slotAvailability.slot.scheduleId,
                                start: new Date(slotAvailability.slot.startDate).valueOf(),
                                end: new Date(slotAvailability.slot.endDate).valueOf(),
                                staffMemberId: (_b = slotAvailability.slot.resource) === null || _b === void 0 ? void 0 : _b.id,
                                locationId: (_c = slotAvailability.slot.location) === null || _c === void 0 ? void 0 : _c.id,
                            };
                            return [2 /*return*/ , this.navigateToBookingsContactInfoPage(sessionDetails, referral, timezone)];
                        } else {
                            return [2 /*return*/ , this.navigateToCourseContactInfoPage(serviceSlug, referral, timezone)];
                        }
                        _d.label = 4;
                    case 4:
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToCourseContactInfoPage = function(serviceSlug, referral, timezone) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var suffix, queryParams;
            return (0, tslib_1.__generator)(this, function(_a) {
                suffix = "/" + serviceSlug + "/book";
                queryParams = {
                    referral: referral,
                    timezone: timezone,
                };
                return [2 /*return*/ , this.navigateToBookingsWithSuffix("" + suffix + this.queryBuilder(queryParams))];
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsContactInfoPage = function(sessionDetails, referral, timezone) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var suffix, queryParams, locationSuffix, sessionIdSuffix;
            return (0, tslib_1.__generator)(this, function(_a) {
                suffix = "/" + sessionDetails.slug + "/book/" + sessionDetails.scheduleId + "/" + sessionDetails.start + "/" + sessionDetails.end + "/" + sessionDetails.staffMemberId;
                queryParams = (0, tslib_1.__assign)((0, tslib_1.__assign)({}, (sessionDetails ? {
                    referral: referral
                } : {})), {
                    timezone: timezone
                });
                locationSuffix = sessionDetails.locationId ?
                    "/" + sessionDetails.locationId :
                    "/" + bookings_uou_types_1.ReservedLocationIds.UNSPECIFIED_LOCATION;
                sessionIdSuffix = sessionDetails.id ? "/" + sessionDetails.id : '';
                return [2 /*return*/ , this.navigateToBookingsWithSuffix("" + suffix + locationSuffix + sessionIdSuffix + this.queryBuilder(queryParams))];
            });
        });
    };
    WixOOISDKAdapter.prototype.encodeUTF8ToUrlSafeBase64 = function(str) {
        return btoa(unescape(encodeURIComponent(str))).replace('/', '-');
    };
    WixOOISDKAdapter.prototype.navigateToBookingsCheckout = function(bookedEntity, referral, timezone) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var encodedBooking, suffix, queryParams;
            return (0, tslib_1.__generator)(this, function(_a) {
                encodedBooking = this.encodeUTF8ToUrlSafeBase64(JSON.stringify(bookedEntity));
                suffix = "/checkout/" + encodedBooking;
                queryParams = {
                    referral: referral,
                    timezone: timezone,
                };
                return [2 /*return*/ , this.navigateToBookingsWithSuffix("" + suffix + this.queryBuilder(queryParams))];
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToBookingsSessionPage = function(session, referral) {
        var _a;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var locationId, slotDetails;
            return (0, tslib_1.__generator)(this, function(_b) {
                switch (_b.label) {
                    case 0:
                        locationId = ((_a = session.location) === null || _a === void 0 ? void 0 : _a.businessLocation) ?
                            "" + session.location.businessLocation.id :
                            '';
                        slotDetails = {
                            id: session.id,
                            slug: session.service.slugs[0],
                            scheduleId: session.bookingInfo.scheduleId,
                            start: session.bookingInfo.start,
                            end: session.bookingInfo.end,
                            staffMemberId: session.staffMember.id,
                            locationId: locationId,
                        };
                        return [4 /*yield*/ , this.navigateToBookingsContactInfoPage(slotDetails, referral)];
                    case 1:
                        _b.sent();
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.isSSR = function() {
        return this.wixCodeApi.window.rendering.env === 'backend';
    };
    WixOOISDKAdapter.prototype.isSEO = function() {
        return this.wixCodeApi.seo.isInSEO();
    };
    WixOOISDKAdapter.prototype.getCompId = function() {
        return this.compId;
    };
    WixOOISDKAdapter.prototype.isSessionInstanceSet = function() {
        return (this.platformAPIs.storage.session.getItem(this.sessionInstanceKey) ===
            this.getInstance());
    };
    WixOOISDKAdapter.prototype.setSessionInstance = function() {
        this.platformAPIs.storage.session.setItem(this.sessionInstanceKey, this.getInstance());
    };
    WixOOISDKAdapter.prototype.isViewDirectionRtl = function() {
        return bookings_const_1.RTL_LANGUAGES.indexOf(this.getCurrentLanguage()) > -1;
    };
    WixOOISDKAdapter.prototype.getScale = function() {
        var _this = this;
        return new Promise(function(resolve) {
            if (_this.isRunningInIframe()) {
                window.Wix.getBoundingRectAndOffsets(function(_a) {
                    var scale = _a.scale;
                    return resolve(scale);
                });
            } else {
                resolve(1);
            }
        });
    };
    WixOOISDKAdapter.prototype.getDecodedInstance = function() {
        var instance = this.getInstance();
        if (!instance) {
            return null;
        }
        var encodedInstance = instance.substring(instance.indexOf('.') + 1);
        return JSON.parse(this.decodeBase64(encodedInstance));
    };
    WixOOISDKAdapter.prototype.decodeBase64 = function(str) {
        if (typeof atob !== 'undefined') {
            return atob(str);
        }
        return '{}';
    };
    WixOOISDKAdapter.prototype.isBookingsSectionInstalled = function(sectionId) {
        if (this.isRunningInIframe()) {
            return new Promise(function(resolve) {
                window.Wix.isAppSectionInstalled(sectionId, {
                    appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID
                }, function(isInstalled) {
                    resolve(isInstalled);
                });
            });
        }
        return Promise.resolve(this.wixCodeApi.site.isAppSectionInstalled({
            sectionId: sectionId,
            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
        }));
    };
    WixOOISDKAdapter.prototype.isMemberAreaInstalled = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                return [2 /*return*/ , Promise.resolve(this.wixCodeApi.site.isAppSectionInstalled({
                    sectionId: bookings_const_1.MEMBERS_AREA_SECTION_ID,
                    appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                }))];
            });
        });
    };
    WixOOISDKAdapter.prototype.isBookCheckoutInstalled = function() {
        return this.isBookingsSectionInstalled(bookings_const_1.BOOKINGS_CHECKOUT_SECTION_ID);
    };
    WixOOISDKAdapter.prototype.isServicePageInstalled = function() {
        return this.isBookingsSectionInstalled(bookings_const_1.BOOKINGS_SERVICE_PAGE_SECTION_ID);
    };
    WixOOISDKAdapter.prototype.isBookingCalendarInstalled = function() {
        return this.isBookingsSectionInstalled(bookings_const_1.BOOKINGS_CALENDAR_SECTION_ID);
    };
    WixOOISDKAdapter.prototype.isBookingFormInstalled = function() {
        return this.isBookingsSectionInstalled(bookings_const_1.BOOKINGS_FORM_SECTION_ID);
    };
    WixOOISDKAdapter.prototype.getCsrfToken = function() {
        return this.platformAPIs.getCsrfToken();
    };
    WixOOISDKAdapter.prototype.getBookingsUrlSuffix = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var fullBookingsUrl, fullUrl, urlParts, bookingsSuffix;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.wixCodeApi.site.getSectionUrl({
                            sectionId: bookings_const_1.BOOKINGS_LIST_SECTION_ID,
                            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                        })];
                    case 1:
                        fullBookingsUrl = (_a.sent()).url;
                        fullUrl = this.wixCodeApi.location.url;
                        urlParts = fullUrl.split(fullBookingsUrl);
                        bookingsSuffix = urlParts.length > 1 && urlParts[1] ? urlParts[1] : '';
                        return [2 /*return*/ , this.isSuffixContainOnlyQueryParam(bookingsSuffix) ?
                            '' :
                            bookingsSuffix
                        ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getUrlQueryParams = function() {
        return this.wixCodeApi.location.query;
    };
    WixOOISDKAdapter.prototype.getSectionUrlParams = function() {
        try {
            var params = (0, tslib_1.__assign)({}, (this.getUrlQueryParams().appSectionParams ?
                JSON.parse(decodeURI(this.getUrlQueryParams().appSectionParams)) :
                {}));
            return params;
        } catch (_a) {
            return {};
        }
    };
    WixOOISDKAdapter.prototype.getUrlQueryParamValue = function(key) {
        var params = (0, tslib_1.__assign)((0, tslib_1.__assign)({}, this.getSectionUrlParams()), this.getUrlQueryParams());
        return params[key];
    };
    WixOOISDKAdapter.prototype.setUrlQueryParam = function(key, value) {
        var _a;
        this.wixCodeApi.location.queryParams.remove([key]);
        if (typeof value !== 'undefined') {
            this.wixCodeApi.location.queryParams.add((_a = {}, _a[key] = value, _a));
        }
    };
    WixOOISDKAdapter.prototype.isSuffixContainOnlyQueryParam = function(urlSuffix) {
        var splittedByQueryParam = urlSuffix.split('?');
        return (splittedByQueryParam.length > 1 &&
            !splittedByQueryParam[0] &&
            splittedByQueryParam[1]);
    };
    WixOOISDKAdapter.prototype.navigateToServiceListWithQuery = function(queryName, queryValue) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var bookOnlineRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.getBookOnlineRelativeUrl()];
                    case 1:
                        bookOnlineRelativeUrl = _a.sent();
                        this.wixCodeApi.location.to(bookOnlineRelativeUrl + "?" + queryName + "=" + queryValue);
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.getBookOnlineRelativeUrl = function() {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var isBookingCheckoutInstalled, bookOnlineFullUrl, baseUrl, urlParts, bookOnlineRelativeUrl;
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.isBookCheckoutInstalled()];
                    case 1:
                        isBookingCheckoutInstalled = _a.sent();
                        return [4 /*yield*/ , this.wixCodeApi.site.getSectionUrl({
                            sectionId: isBookingCheckoutInstalled ?
                                bookings_const_1.BOOKINGS_LIST_SECTION_ID :
                                bookings_const_1.BOOKINGS_SCHEDULER_SECTION_ID,
                            appDefinitionId: bookings_const_1.BOOKINGS_APP_DEF_ID,
                        })];
                    case 2:
                        bookOnlineFullUrl = (_a.sent()).url;
                        baseUrl = this.wixCodeApi.location.baseUrl;
                        urlParts = bookOnlineFullUrl.split(baseUrl);
                        bookOnlineRelativeUrl = urlParts.length > 1 ? urlParts[1] : bookOnlineFullUrl;
                        return [2 /*return*/ , bookOnlineRelativeUrl];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.navigateToSpecificSitePage = function(pageId) {
        this.wixCodeApi.location.navigateTo({
            type: 'PageLink',
            pageId: pageId
        });
    };
    WixOOISDKAdapter.prototype.getServiceSlug = function(pageName) {
        var _a;
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            var sitePageName, locationPaths, sitePageNameIndex, isSitePageNameExistInPath, isServiceSlugExistsInPath, serviceSlug, shouldFixProblematicSiteStructureInPreview;
            return (0, tslib_1.__generator)(this, function(_b) {
                sitePageName = (_a = this.wixCodeApi.site.currentPage.url) === null || _a === void 0 ? void 0 : _a.substring(1).toLowerCase();
                locationPaths = this.wixCodeApi.location.path.map(function(subPath) {
                    return subPath.toLowerCase();
                });
                sitePageNameIndex = locationPaths.indexOf(sitePageName);
                isSitePageNameExistInPath = sitePageNameIndex !== -1;
                if (!isSitePageNameExistInPath && this.isPreviewMode()) {
                    sitePageNameIndex = 1;
                }
                isServiceSlugExistsInPath = isSitePageNameExistInPath &&
                    sitePageNameIndex !== this.wixCodeApi.location.path.length - 1;
                serviceSlug = isServiceSlugExistsInPath ?
                    this.wixCodeApi.location.path[sitePageNameIndex + 1] :
                    '';
                shouldFixProblematicSiteStructureInPreview = this.isPreviewMode() &&
                    !isSitePageNameExistInPath &&
                    this.wixCodeApi.location.path.length === 1 &&
                    locationPaths[0] !== pageName;
                if (shouldFixProblematicSiteStructureInPreview) {
                    serviceSlug = this.wixCodeApi.location.path[0];
                }
                if (serviceSlug) {
                    serviceSlug = decodeURI(serviceSlug);
                }
                return [2 /*return*/ , serviceSlug];
            });
        });
    };
    WixOOISDKAdapter.prototype.renderSeoTags = function(itemType, itemData) {
        return (0, tslib_1.__awaiter)(this, void 0, void 0, function() {
            return (0, tslib_1.__generator)(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        return [4 /*yield*/ , this.wixCodeApi.seo.renderSEOTags({
                            itemType: itemType,
                            itemData: itemData,
                        })];
                    case 1:
                        _a.sent();
                        return [2 /*return*/ ];
                }
            });
        });
    };
    WixOOISDKAdapter.prototype.setToSessionStorage = function(key, value) {
        this.platformAPIs.storage.session.setItem(key, value);
    };
    WixOOISDKAdapter.prototype.getFromSessionStorage = function(key) {
        return this.platformAPIs.storage.session.getItem(key);
    };
    WixOOISDKAdapter.prototype.removeFromSessionStorage = function(key) {
        return this.platformAPIs.storage.session.removeItem(key);
    };
    WixOOISDKAdapter.prototype.getUserRoles = function() {
        return this.wixCodeApi.user.currentUser.getRoles();
    };
    return WixOOISDKAdapter;
}());
exports.WixOOISDKAdapter = WixOOISDKAdapter;
