// @wix/advanced-seo-utils/dist/src/private/renderer/get-tags-async/get-tags.js
"use strict";
var __assign = (this && this.__assign) || function() {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s)
                if (Object.prototype.hasOwnProperty.call(s, p))
                    t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function(thisArg, _arguments, P, generator) {
    function adopt(value) {
        return value instanceof P ? value : new P(function(resolve) {
            resolve(value);
        });
    }
    return new(P || (P = Promise))(function(resolve, reject) {
        function fulfilled(value) {
            try {
                step(generator.next(value));
            } catch (e) {
                reject(e);
            }
        }

        function rejected(value) {
            try {
                step(generator["throw"](value));
            } catch (e) {
                reject(e);
            }
        }

        function step(result) {
            result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected);
        }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function(thisArg, body) {
    var _ = {
            label: 0,
            sent: function() {
                if (t[0] & 1) throw t[1];
                return t[1];
            },
            trys: [],
            ops: []
        },
        f, y, t, g;
    return g = {
        next: verb(0),
        "throw": verb(1),
        "return": verb(2)
    }, typeof Symbol === "function" && (g[Symbol.iterator] = function() {
        return this;
    }), g;

    function verb(n) {
        return function(v) {
            return step([n, v]);
        };
    }

    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0:
                case 1:
                    t = op;
                    break;
                case 4:
                    _.label++;
                    return {
                        value: op[1],
                        done: false
                    };
                case 5:
                    _.label++;
                    y = op[1];
                    op = [0];
                    continue;
                case 7:
                    op = _.ops.pop();
                    _.trys.pop();
                    continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                        _ = 0;
                        continue;
                    }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                        _.label = op[1];
                        break;
                    }
                    if (op[0] === 6 && _.label < t[1]) {
                        _.label = t[1];
                        t = op;
                        break;
                    }
                    if (t && _.label < t[2]) {
                        _.label = t[2];
                        _.ops.push(op);
                        break;
                    }
                    if (t[2]) _.ops.pop();
                    _.trys.pop();
                    continue;
            }
            op = body.call(thisArg, _);
        } catch (e) {
            op = [6, e];
            y = 0;
        } finally {
            f = t = 0;
        }
        if (op[0] & 5) throw op[1];
        return {
            value: op[0] ? op[1] : void 0,
            done: true
        };
    }
};
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getTags = void 0;
var convert_site_level_meta_tags_1 = require("../../converters/convert-site-level-meta-tags");
var get_user_pattern_by_item_type_1 = require("../../patterns/get-user-pattern-by-item-type");
var get_context_1 = require("../../tags/context/get-context");
var ItemTypes_1 = require("../../types/ItemTypes");
var filter_page_level_tags_1 = require("../../tags/filters/filter-page-level-tags");
var adapters_async_1 = require("../../adapters-async");
var convert_seo_model_1 = require("../../converters/convert-seo-model");
var resolve_with_patterns_1 = require("../resolve-with-patterns");
var get_default_item_1 = require("../../patterns/get-default-item");
var handle_old_blog_post_1 = require("../../item-types/blog-post/handleOldBlogPost/handle-old-blog-post");
var ContextProps_1 = require("../../types/ContextProps");
var filter_payload_without_velo_1 = require("../../tags/filters/filter-payload-without-velo");
var convert_components_item_payload_1 = require("../convert-components-item-payload");
var get_default_patterns_schemas_async_1 = require("../get-default-patterns-schemas-async");
var get_dynamic_patterns_schema_1 = require("../get-dynamic-patterns-schema");
var get_multilingual_tags_1 = require("../get-multilingual-tags");
var has_adapter_multilingual_data_1 = require("../has-adapter-multilingual-data");

function _getEditorPayload(context, pageLevelSeoData) {
    return __awaiter(this, void 0, void 0, function() {
        var editorPanelLegacyData, editorPanelPageData;
        return __generator(this, function(_a) {
            switch (_a.label) {
                case 0:
                    return [4 /*yield*/ , adapters_async_1.getAdapter(ItemTypes_1.STATIC_PAGE_V2)];
                case 1:
                    editorPanelLegacyData = (_a.sent()).getLegacySeoBlob({
                        context: context,
                    });
                    editorPanelPageData = convert_seo_model_1.convertSeoModel(pageLevelSeoData.advancedSeoData);
                    return [2 /*return*/ , [editorPanelLegacyData, editorPanelPageData]];
            }
        });
    });
}

function _getUserPatternPayload(siteLevelSeoData, itemType, options) {
    if (options === void 0) {
        options = {
            logError: function() {}
        };
    }
    var userPatternSchema = get_user_pattern_by_item_type_1.getUserPatternByItemType(siteLevelSeoData.userPatterns, itemType, options);
    var _a = itemType === ItemTypes_1.STATIC_PAGE_V2 ?
        [userPatternSchema, {}] :
        [{}, userPatternSchema],
        userStaticPatternSchema = _a[0],
        userDynamicPatternSchema = _a[1];
    return {
        userPatternSchema: userPatternSchema,
        userStaticPatternSchema: userStaticPatternSchema,
        userDynamicPatternSchema: userDynamicPatternSchema,
    };
}

function _getPlatformPayload(adapter, itemData, seoData, isOriginalLang) {
    if (isOriginalLang === void 0) {
        isOriginalLang = true;
    }
    var platformLegacySeoOverrides = adapter.getLegacySeoBlob(itemData);
    var platformSchemaOverrides = seoData;
    return isOriginalLang ?
        [platformLegacySeoOverrides, platformSchemaOverrides] :
        [platformSchemaOverrides, platformLegacySeoOverrides];
}

function getTags(_a) {
    var _b = _a === void 0 ? {} : _a,
        _c = _b.siteLevelSeoData,
        siteLevelSeoData = _c === void 0 ? {} : _c,
        _d = _b.pageLevelSeoData,
        pageLevelSeoData = _d === void 0 ? {} : _d,
        _e = _b.veloOverrides,
        veloOverrides = _e === void 0 ? [] : _e,
        _f = _b.veloItemPayload,
        veloItemPayload = _f === void 0 ? get_default_item_1.getDefaultItemPayload() : _f,
        _g = _b.dynamicPageData,
        dynamicPageData = _g === void 0 ? [] : _g,
        _h = _b.tpaSeoEndpointData,
        tpaSeoEndpointData = _h === void 0 ? [] : _h,
        _j = _b.tpaOverrides,
        tpaOverrides = _j === void 0 ? [] : _j,
        _k = _b.componentsItemPayload,
        componentsItemPayload = _k === void 0 ? [] : _k,
        _l = _b.options,
        options = _l === void 0 ? {
            logError: function() {}
        } : _l;
    return __awaiter(this, void 0, void 0, function() {
        var siteLevelMetaTags, itemType, itemData, seoData, asNewPage, _m, defaultStaticPatternSchema, defaultDynamicPatternSchema, _o, userPatternSchema, userStaticPatternSchema, userDynamicPatternSchema, context, itemDataWithContext, adapter, platformPayload, staticPatterns, dynamicSchemaFromAdapter, hasMultilingualData, dynamicPatterns, payload, adaptedContext, multilingualTags, _p, componentsTags, componentContext, editorPanelPayload, resolvedTags;
        return __generator(this, function(_q) {
            switch (_q.label) {
                case 0:
                    siteLevelMetaTags = filter_page_level_tags_1.filterPageLevelTags(convert_site_level_meta_tags_1.convertSiteLevelMetaTags(siteLevelSeoData.metaTags));
                    itemType = veloItemPayload.itemType, itemData = veloItemPayload.itemData, seoData = veloItemPayload.seoData, asNewPage = veloItemPayload.asNewPage;
                    return [4 /*yield*/ , get_default_patterns_schemas_async_1.getDefaultPatternsSchemas(itemType)];
                case 1:
                    _m = _q.sent(), defaultStaticPatternSchema = _m[0], defaultDynamicPatternSchema = _m[1];
                    _o = _getUserPatternPayload(siteLevelSeoData, itemType, options), userPatternSchema = _o.userPatternSchema, userStaticPatternSchema = _o.userStaticPatternSchema, userDynamicPatternSchema = _o.userDynamicPatternSchema;
                    context = get_context_1.getContext(siteLevelSeoData, pageLevelSeoData, userPatternSchema);
                    itemDataWithContext = __assign(__assign({}, itemData), {
                        context: context
                    });
                    return [4 /*yield*/ , adapters_async_1.getAdapter(itemType)];
                case 2:
                    adapter = _q.sent();
                    platformPayload = _getPlatformPayload(adapter, itemData, seoData, context[ContextProps_1.CONTEXT_PROPS.CURR_LANG_IS_ORIGINAL]);
                    staticPatterns = [defaultStaticPatternSchema, userStaticPatternSchema];
                    dynamicSchemaFromAdapter = get_dynamic_patterns_schema_1.getDynamicPatternsSchema(adapter, itemDataWithContext);
                    hasMultilingualData = has_adapter_multilingual_data_1.hasAdapterMultilingualData(adapter, itemDataWithContext);
                    dynamicPatterns = [
                        defaultDynamicPatternSchema,
                        dynamicSchemaFromAdapter,
                        userDynamicPatternSchema,
                    ];
                    payload = [];
                    adaptedContext = __assign({}, adapter.getData(itemDataWithContext));
                    multilingualTags = !hasMultilingualData &&
                        get_multilingual_tags_1.getMultilingualSchemaFromContext(itemDataWithContext, {
                            translationsData: pageLevelSeoData === null || pageLevelSeoData === void 0 ? void 0 : pageLevelSeoData.translationsData,
                        });
                    if (!asNewPage) return [3 /*break*/ , 3];
                    payload = __spreadArrays([
                        siteLevelMetaTags,
                        multilingualTags
                    ], staticPatterns, dynamicPatterns, platformPayload);
                    return [3 /*break*/ , 6];
                case 3:
                    return [4 /*yield*/ , convert_components_item_payload_1.convertComponentsItemPayload({
                        componentsItemPayload: componentsItemPayload,
                        context: context,
                    })];
                case 4:
                    _p = _q.sent(), componentsTags = _p.componentsTags, componentContext = _p.componentContext;
                    return [4 /*yield*/ , _getEditorPayload(context, pageLevelSeoData)];
                case 5:
                    editorPanelPayload = _q.sent();
                    adaptedContext = __assign(__assign({}, adaptedContext), componentContext);
                    payload = __spreadArrays([
                        siteLevelMetaTags,
                        multilingualTags
                    ], staticPatterns, editorPanelPayload, dynamicPatterns, componentsTags, [
                        dynamicPageData,
                        tpaSeoEndpointData,
                        tpaOverrides
                    ], platformPayload);
                    payload = [filter_payload_without_velo_1.filterPayloadWithoutVelo(payload, veloOverrides), veloOverrides];
                    _q.label = 6;
                case 6:
                    resolvedTags = resolve_with_patterns_1.resolveWithPatterns(payload, adaptedContext, options);
                    return [2 /*return*/ , handle_old_blog_post_1.handleOldBlog(resolvedTags, siteLevelSeoData)];
            }
        });
    });
}
exports.getTags = getTags;
