// @wix/advanced-seo-utils/dist/src/private/item-types/blog-post/schema-presets/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.buildBlogPostSchema = void 0;
var adapter_blog_post_1 = __importDefault(require("../adapter-blog-post"));
var utils_1 = require("../../../adapters/utils");
var IDs = adapter_blog_post_1.default.IDs;
exports.buildBlogPostSchema = function(_a) {
    var schemaType = _a.schemaType,
        type = _a.type;
    return ({
        type: schemaType,
        schema: {
            '@context': 'https://schema.org',
            '@type': type,
            author: {
                '@type': 'Person',
                name: "{{" + IDs.AUTHOR + "}}",
                image: {
                    '@type': 'ImageObject',
                    url: "{{" + IDs.AUTHOR_IMAGE_URL + "}}",
                    width: "{{" + IDs.AUTHOR_IMAGE_WIDTH + "}}",
                    height: "{{" + IDs.AUTHOR_IMAGE_HEIGHT + "}}",
                },
            },
            articleSection: "{{" + IDs.CATEGORIES + "}}",
            dateModified: "{{" + IDs.LAST_MODIFIED_DATE + "}}",
            datePublished: "{{" + IDs.PUBLISH_DATE + "}}",
            headline: "{{" + IDs.HEADLINE + "}}",
            mainEntityOfPage: {
                '@id': "{{" + IDs.POST_PAGE_URL + "}}",
                '@type': 'itemPage',
                url: "{{" + IDs.POST_PAGE_URL + "}}",
            },
            description: "{{" + IDs.DESCRIPTION + "}}",
            url: "{{" + IDs.POST_PAGE_URL + "}}",
            image: {
                '@type': 'ImageObject',
                url: "{{" + IDs.COVER_IMAGE + "}}",
                width: "{{" + IDs.COVER_IMAGE_WIDTH + "}}",
                height: "{{" + IDs.COVER_IMAGE_HEIGHT + "}}",
            },
            publisher: {
                '@type': 'Organization',
                name: "{{" + utils_1.siteIDs.SITE_NAME + "}}",
                logo: {
                    '@type': 'ImageObject',
                    url: "{{" + utils_1.siteIDs.BUSINESS_LOGO + "}}",
                },
            },
        },
        disabled: "{{" + IDs.SD_DISABLE_SCHEMA + "}",
    });
};
