// @wix/advanced-seo-utils/dist/src/private/item-types/forum-post/pattern-forum-post.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.FORUM_POST = void 0;
var builders_1 = require("../../types/builders");
var adapter_forum_post_1 = __importDefault(require("./adapter-forum-post"));
var utils_1 = require("../../adapters/utils");
var forum_post_questions_and_answers_1 = __importDefault(require("./schema-presets/forum-post-questions-and-answers"));
var IDs = adapter_forum_post_1.default.IDs;
exports.FORUM_POST = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.TITLE + "}} | {{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildStandardMeta('description', "{{" + IDs.CONTENT + "}}")
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:url', "{{" + IDs.POST_PAGE_URL + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'article'),
        builders_1.buildOgMeta('og:image', "{{" + IDs.COVER_IMAGE + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.COVER_IMAGE_HEIGHT + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.COVER_IMAGE_WIDTH + "}}"),
        builders_1.buildOgMeta('article:author', "{{" + IDs.AUTHOR + "}}"),
        builders_1.buildOgMeta('article:published_time', "{{" + IDs.PUBLISH_DATE + "}}"),
        builders_1.buildOgMeta('article:modified_time', "{{" + IDs.LAST_MODIFIED_DATE + "}}"),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.POST_PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'prev',
            href: "{{" + IDs.PREV_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'next',
            href: "{{" + IDs.NEXT_URL + "}}"
        }),
        builders_1.buildStandardMeta('robots', 'index')
    ], builders_1.buildTwitterMetaRefs(), [
        builders_1.buildStructuredData({
            schema: forum_post_questions_and_answers_1.default.schema,
            schemaType: forum_post_questions_and_answers_1.default.type,
            disabled: [
                '{{options.disable-pattern-schema}}',
                "{{" + IDs.QUESTIONS_AND_ANSWERS_SD_DISABLE_SCHEMA + "}}",
            ],
        }),
    ]),
};
