// @wix/bookings-uou-mappers/dist/esm/policy/policy.formatter.js
var NUMBER_OF_MINUTES_IN_AN_HOUR = 60;
var NUMBER_OF_HOURS_IN_A_DAY = 24;
var NUMBER_OF_MINUTES_IN_A_DAY = 1440;
var PolicyFormatter = /** @class */ (function() {
    function PolicyFormatter(translations) {
        var _this = this;
        this.translations = translations;
        this.getPolicyLimitByUnits = function(limitInMinutes) {
            var days = limitInMinutes / NUMBER_OF_MINUTES_IN_A_DAY;
            var hours = (days - Math.floor(days)) * NUMBER_OF_HOURS_IN_A_DAY;
            var minutes = limitInMinutes % NUMBER_OF_MINUTES_IN_AN_HOUR;
            return {
                days: Math.floor(days),
                hours: Math.floor(hours),
                minutes: minutes,
            };
        };
        this.getPolicyLimit = function(limitInMinutes) {
            var _a = _this.getPolicyLimitByUnits(limitInMinutes),
                days = _a.days,
                hours = _a.hours,
                minutes = _a.minutes;
            if (days) {
                return days + " " + (days === 1 ? _this.translations.day : _this.translations.days);
            }
            var formattedTime = [];
            if (hours) {
                formattedTime.push(hours + " " + (hours === 1 ? _this.translations.hour : _this.translations.hours));
            }
            if (minutes) {
                formattedTime.push(minutes + " " + (minutes === 1 ? _this.translations.minute : _this.translations.minutes));
            }
            if (formattedTime.length > 1) {
                return formattedTime[0] + " " + _this.translations.and + " " + formattedTime[1];
            } else {
                return formattedTime[0];
            }
        };
    }
    return PolicyFormatter;
}());
export {
    PolicyFormatter
};
