// @wix/wixstores-client-storefront-sdk/dist/es/src/services/ProductLocation/LocationService.js
import {
    __awaiter,
    __generator
} from "tslib";
import {
    queryToString
} from '../utils';
import {
    CustomUrlApi
} from '../../utils/CustomUrl/CustomUrlApi';
import {
    PageMap
} from '@wix/wixstores-client-core/dist/es/src/constants';
import {
    parseUrl
} from '@wix/native-components-infra/dist/src/urlUtils';
var LocationService = /** @class */ (function() {
    function LocationService(_a) {
        var siteStore = _a.siteStore,
            _b = _a.sectionPage,
            sectionPage = _b === void 0 ? PageMap.PRODUCT : _b;
        this.beenInit = false;
        this.siteStore = siteStore;
        this.sectionPage = sectionPage;
        this.customUrlApi = new CustomUrlApi(this.siteStore.location.buildCustomizedUrl);
    }
    LocationService.prototype.init = function() {
        return __awaiter(this, void 0, void 0, function() {
            var _a, _b, _c;
            return __generator(this, function(_d) {
                switch (_d.label) {
                    case 0:
                        _a = this;
                        return [4 /*yield*/ , this.siteStore.getSectionUrl(this.sectionPage)];
                    case 1:
                        _a._sectionUrl = (_d.sent()).url;
                        _b = this;
                        return [4 /*yield*/ , this.customUrlApi.init()];
                    case 2:
                        _b._isUrlWithOverrides = _d.sent();
                        if (!this._isUrlWithOverrides) return [3 /*break*/ , 4];
                        _c = this;
                        return [4 /*yield*/ , this.getCustomizedUrlSegments()];
                    case 3:
                        _c._urlOverrideSegments = _d.sent();
                        _d.label = 4;
                    case 4:
                        this.beenInit = true;
                        return [2 /*return*/ , this];
                }
            });
        });
    };
    Object.defineProperty(LocationService.prototype, "sectionUrl", {
        get: function() {
            return this.validateInit() && this._sectionUrl;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(LocationService.prototype, "urlOverrideSegments", {
        get: function() {
            return this.validateInit() && this._urlOverrideSegments;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(LocationService.prototype, "isUrlWithOverrides", {
        get: function() {
            return this.validateInit() && this._isUrlWithOverrides;
        },
        enumerable: false,
        configurable: true
    });
    LocationService.prototype.validateInit = function() {
        if (!this.beenInit) {
            throw new Error('init LocationService wasn`t called');
        }
        return true;
    };
    LocationService.prototype.getProductUrl = function(productUrlPart, includeQueryParams) {
        if (includeQueryParams === void 0) {
            includeQueryParams = false;
        }
        if (this.isUrlWithOverrides && this.customUrlApi) {
            return this.customUrlApi.buildProductPageUrl({
                slug: productUrlPart
            });
        } else {
            var prefix = this.sectionUrl + "/" + productUrlPart;
            return !includeQueryParams || Object.keys(this.siteStore.location.query).length === 0 ?
                prefix :
                prefix + "?" + queryToString(this.siteStore.location.query);
        }
    };
    LocationService.prototype.getCurrentUrl = function() {
        return this.getUrlWithoutParams(this.siteStore.location.url);
    };
    LocationService.prototype.getCustomizedUrlSegments = function() {
        return __awaiter(this, void 0, void 0, function() {
            var currentUrl;
            return __generator(this, function(_a) {
                currentUrl = this.getCurrentUrl();
                return [2 /*return*/ , this.siteStore.siteApis.getCustomizedUrlSegments(currentUrl)];
            });
        });
    };
    LocationService.prototype.getUrlWithoutParams = function(url) {
        var parsedUrl = parseUrl(url);
        return parsedUrl.protocol + "://" + parsedUrl.host + parsedUrl.path;
    };
    return LocationService;
}());
export {
    LocationService
};
