// @wix/ambassador-wix-form-builder-web/types.js
/* eslint-disable */
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ViolationType = exports.ViewMode = exports.Version = exports.V2FormStatus = exports.V1FormStatus = exports.Type = exports.TransactionStatus = exports.TimeFormat = exports.SubscriptionSubscriptionMode = exports.SubscriptionSettingsSubscriptionMode = exports.SubscribeMode = exports.SubmissionsLimitRuleRequestTimeFrame = exports.Status = exports.SourceType = exports.SortOrder = exports.PublishedStatus = exports.Plugin = exports.PhoneInfoTag = exports.Operation = exports.MediaType = exports.LimitRulesTimeFrame = exports.LimitRulesDateOperator = exports.FormType = exports.FormFieldViewInfoFieldType = exports.FormFieldDefinitionFieldType = exports.FormFieldContactInfoContactField = exports.FormExposure = exports.FormDirection = exports.FormAttributesLimitSettingsSubmissionLimitTimeFrame = exports.FormAttributesLimitSettingsDateLimitDateOperator = exports.FormAttributeLimitSettingsSubmissionLimitTimeFrame = exports.FormAttributeLimitSettingsDateLimitDateOperator = exports.FirstDayOfWeek = exports.Feature = exports.EmptyState = exports.EmailInfoTag = exports.EmailGroup = exports.Currency = exports.ContactFormType = exports.AddressInfoTag = void 0;
var AddressInfoTag;
(function(AddressInfoTag) {
    AddressInfoTag["UNTAGGED"] = "UNTAGGED";
    AddressInfoTag["HOME"] = "HOME";
    AddressInfoTag["WORK"] = "WORK";
    AddressInfoTag["BILLING"] = "BILLING";
    AddressInfoTag["SHIPPING"] = "SHIPPING";
})(AddressInfoTag = exports.AddressInfoTag || (exports.AddressInfoTag = {}));
var ContactFormType;
(function(ContactFormType) {
    ContactFormType["SUBSCRIBE"] = "SUBSCRIBE";
})(ContactFormType = exports.ContactFormType || (exports.ContactFormType = {}));
var Currency;
(function(Currency) {
    Currency["UNDEFINED"] = "UNDEFINED";
    Currency["AFN"] = "AFN";
    Currency["USD"] = "USD";
    Currency["EUR"] = "EUR";
    Currency["BMD"] = "BMD";
    Currency["BGN"] = "BGN";
    Currency["BIF"] = "BIF";
    Currency["CVE"] = "CVE";
    Currency["KMF"] = "KMF";
    Currency["CDF"] = "CDF";
    Currency["DJF"] = "DJF";
    Currency["ERN"] = "ERN";
    Currency["FKP"] = "FKP";
    Currency["GHS"] = "GHS";
    Currency["GNF"] = "GNF";
    Currency["LAK"] = "LAK";
    Currency["LSL"] = "LSL";
    Currency["LYD"] = "LYD";
    Currency["MWK"] = "MWK";
    Currency["MRO"] = "MRO";
    Currency["MZN"] = "MZN";
    Currency["PAB"] = "PAB";
    Currency["RON"] = "RON";
    Currency["RUB"] = "RUB";
    Currency["SHP"] = "SHP";
    Currency["RSD"] = "RSD";
    Currency["SLL"] = "SLL";
    Currency["SBD"] = "SBD";
    Currency["SOS"] = "SOS";
    Currency["SZL"] = "SZL";
    Currency["TJS"] = "TJS";
    Currency["TOP"] = "TOP";
    Currency["TMT"] = "TMT";
    Currency["YER"] = "YER";
    Currency["ZMW"] = "ZMW";
    Currency["GIP"] = "GIP";
    Currency["ZAR"] = "ZAR";
    Currency["XPF"] = "XPF";
    Currency["XOF"] = "XOF";
    Currency["XCD"] = "XCD";
    Currency["XAF"] = "XAF";
    Currency["WST"] = "WST";
    Currency["VUV"] = "VUV";
    Currency["VND"] = "VND";
    Currency["VEF"] = "VEF";
    Currency["UZS"] = "UZS";
    Currency["UYU"] = "UYU";
    Currency["UGX"] = "UGX";
    Currency["UAH"] = "UAH";
    Currency["TZS"] = "TZS";
    Currency["TWD"] = "TWD";
    Currency["TTD"] = "TTD";
    Currency["TRY"] = "TRY";
    Currency["TND"] = "TND";
    Currency["THB"] = "THB";
    Currency["STD"] = "STD";
    Currency["SGD"] = "SGD";
    Currency["SEK"] = "SEK";
    Currency["SCR"] = "SCR";
    Currency["SAR"] = "SAR";
    Currency["RWF"] = "RWF";
    Currency["QAR"] = "QAR";
    Currency["PYG"] = "PYG";
    Currency["PLN"] = "PLN";
    Currency["PKR"] = "PKR";
    Currency["PHP"] = "PHP";
    Currency["PGK"] = "PGK";
    Currency["PEN"] = "PEN";
    Currency["OMR"] = "OMR";
    Currency["NZD"] = "NZD";
    Currency["NPR"] = "NPR";
    Currency["NOK"] = "NOK";
    Currency["NIO"] = "NIO";
    Currency["NGN"] = "NGN";
    Currency["MYR"] = "MYR";
    Currency["MXN"] = "MXN";
    Currency["MVR"] = "MVR";
    Currency["MUR"] = "MUR";
    Currency["MOP"] = "MOP";
    Currency["MNT"] = "MNT";
    Currency["MMK"] = "MMK";
    Currency["MKD"] = "MKD";
    Currency["MDL"] = "MDL";
    Currency["MAD"] = "MAD";
    Currency["LRD"] = "LRD";
    Currency["LKR"] = "LKR";
    Currency["LBP"] = "LBP";
    Currency["KZT"] = "KZT";
    Currency["KYD"] = "KYD";
    Currency["KWD"] = "KWD";
    Currency["KRW"] = "KRW";
    Currency["KHR"] = "KHR";
    Currency["KGS"] = "KGS";
    Currency["KES"] = "KES";
    Currency["JPY"] = "JPY";
    Currency["JOD"] = "JOD";
    Currency["JMD"] = "JMD";
    Currency["ISK"] = "ISK";
    Currency["INR"] = "INR";
    Currency["ILS"] = "ILS";
    Currency["IDR"] = "IDR";
    Currency["HUF"] = "HUF";
    Currency["HTG"] = "HTG";
    Currency["HRK"] = "HRK";
    Currency["HNL"] = "HNL";
    Currency["HKD"] = "HKD";
    Currency["GYD"] = "GYD";
    Currency["GTQ"] = "GTQ";
    Currency["GMD"] = "GMD";
    Currency["GEL"] = "GEL";
    Currency["GBP"] = "GBP";
    Currency["FJD"] = "FJD";
    Currency["ETB"] = "ETB";
    Currency["EGP"] = "EGP";
    Currency["DZD"] = "DZD";
    Currency["DOP"] = "DOP";
    Currency["DKK"] = "DKK";
    Currency["CZK"] = "CZK";
    Currency["CRC"] = "CRC";
    Currency["COP"] = "COP";
    Currency["CNY"] = "CNY";
    Currency["CLP"] = "CLP";
    Currency["CHF"] = "CHF";
    Currency["CAD"] = "CAD";
    Currency["BZD"] = "BZD";
    Currency["BYN"] = "BYN";
    Currency["BWP"] = "BWP";
    Currency["BTN"] = "BTN";
    Currency["BSD"] = "BSD";
    Currency["BRL"] = "BRL";
    Currency["BOB"] = "BOB";
    Currency["BND"] = "BND";
    Currency["BHD"] = "BHD";
    Currency["BDT"] = "BDT";
    Currency["BBD"] = "BBD";
    Currency["BAM"] = "BAM";
    Currency["AZN"] = "AZN";
    Currency["AWG"] = "AWG";
    Currency["AUD"] = "AUD";
    Currency["ARS"] = "ARS";
    Currency["AOA"] = "AOA";
    Currency["ANG"] = "ANG";
    Currency["AMD"] = "AMD";
    Currency["ALL"] = "ALL";
    Currency["AED"] = "AED";
    Currency["MGA"] = "MGA";
    Currency["NAD"] = "NAD";
    Currency["SRD"] = "SRD";
    Currency["IQD"] = "IQD";
})(Currency = exports.Currency || (exports.Currency = {}));
var EmailGroup;
(function(EmailGroup) {
    EmailGroup["SITE_USERS"] = "SITE_USERS";
    EmailGroup["NON_SITE_USERS"] = "NON_SITE_USERS";
    EmailGroup["CONFIRM_SUBSCRIPTION"] = "CONFIRM_SUBSCRIPTION";
})(EmailGroup = exports.EmailGroup || (exports.EmailGroup = {}));
var EmailInfoTag;
(function(EmailInfoTag) {
    EmailInfoTag["UNTAGGED"] = "UNTAGGED";
    EmailInfoTag["MAIN"] = "MAIN";
    EmailInfoTag["HOME"] = "HOME";
    EmailInfoTag["WORK"] = "WORK";
})(EmailInfoTag = exports.EmailInfoTag || (exports.EmailInfoTag = {}));
var EmptyState;
(function(EmptyState) {
    EmptyState["UNDEFINED"] = "UNDEFINED";
    EmptyState["EMAILS"] = "EMAILS";
    EmptyState["USERS"] = "USERS";
})(EmptyState = exports.EmptyState || (exports.EmptyState = {}));
var Feature;
(function(Feature) {
    Feature["UNDEFINED_ALLOWED_FIELD"] = "UNDEFINED_ALLOWED_FIELD";
    Feature["GENERAL_UPLOAD_BUTTON"] = "GENERAL_UPLOAD_BUTTON";
    Feature["DOWNLOAD_FILE"] = "DOWNLOAD_FILE";
    Feature["PAYMENT_FORM"] = "PAYMENT_FORM";
    Feature["SIGNATURE_FIELD"] = "SIGNATURE_FIELD";
    Feature["REMOVE_ASCEND_BRANDING"] = "REMOVE_ASCEND_BRANDING";
    Feature["MAX_EMAILS_PER_FORM"] = "MAX_EMAILS_PER_FORM";
})(Feature = exports.Feature || (exports.Feature = {}));
var FirstDayOfWeek;
(function(FirstDayOfWeek) {
    FirstDayOfWeek["UNDEFINED"] = "UNDEFINED";
    FirstDayOfWeek["SUNDAY"] = "SUNDAY";
    FirstDayOfWeek["MONDAY"] = "MONDAY";
})(FirstDayOfWeek = exports.FirstDayOfWeek || (exports.FirstDayOfWeek = {}));
var FormAttributeLimitSettingsDateLimitDateOperator;
(function(FormAttributeLimitSettingsDateLimitDateOperator) {
    FormAttributeLimitSettingsDateLimitDateOperator["UNDEFINED"] = "UNDEFINED";
    FormAttributeLimitSettingsDateLimitDateOperator["BEFORE"] = "BEFORE";
})(FormAttributeLimitSettingsDateLimitDateOperator = exports.FormAttributeLimitSettingsDateLimitDateOperator || (exports.FormAttributeLimitSettingsDateLimitDateOperator = {}));
var FormAttributeLimitSettingsSubmissionLimitTimeFrame;
(function(FormAttributeLimitSettingsSubmissionLimitTimeFrame) {
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["UNDEFINED"] = "UNDEFINED";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["HOURLY"] = "HOURLY";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["DAILY"] = "DAILY";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["WEEKLY"] = "WEEKLY";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["MONTHLY"] = "MONTHLY";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["YEARLY"] = "YEARLY";
    FormAttributeLimitSettingsSubmissionLimitTimeFrame["TOTAL"] = "TOTAL";
})(FormAttributeLimitSettingsSubmissionLimitTimeFrame = exports.FormAttributeLimitSettingsSubmissionLimitTimeFrame || (exports.FormAttributeLimitSettingsSubmissionLimitTimeFrame = {}));
var FormAttributesLimitSettingsDateLimitDateOperator;
(function(FormAttributesLimitSettingsDateLimitDateOperator) {
    FormAttributesLimitSettingsDateLimitDateOperator["UNDEFINED"] = "UNDEFINED";
    FormAttributesLimitSettingsDateLimitDateOperator["BEFORE"] = "BEFORE";
})(FormAttributesLimitSettingsDateLimitDateOperator = exports.FormAttributesLimitSettingsDateLimitDateOperator || (exports.FormAttributesLimitSettingsDateLimitDateOperator = {}));
var FormAttributesLimitSettingsSubmissionLimitTimeFrame;
(function(FormAttributesLimitSettingsSubmissionLimitTimeFrame) {
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["UNDEFINED"] = "UNDEFINED";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["HOURLY"] = "HOURLY";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["DAILY"] = "DAILY";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["WEEKLY"] = "WEEKLY";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["MONTHLY"] = "MONTHLY";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["YEARLY"] = "YEARLY";
    FormAttributesLimitSettingsSubmissionLimitTimeFrame["TOTAL"] = "TOTAL";
})(FormAttributesLimitSettingsSubmissionLimitTimeFrame = exports.FormAttributesLimitSettingsSubmissionLimitTimeFrame || (exports.FormAttributesLimitSettingsSubmissionLimitTimeFrame = {}));
var FormDirection;
(function(FormDirection) {
    FormDirection["UNDEFINED"] = "UNDEFINED";
    FormDirection["LTR"] = "LTR";
    FormDirection["RTL"] = "RTL";
})(FormDirection = exports.FormDirection || (exports.FormDirection = {}));
var FormExposure;
(function(FormExposure) {
    FormExposure["UNDEFINED_EXPOSURE"] = "UNDEFINED_EXPOSURE";
    FormExposure["SITE"] = "SITE";
    FormExposure["STANDALONE"] = "STANDALONE";
})(FormExposure = exports.FormExposure || (exports.FormExposure = {}));
var FormFieldContactInfoContactField;
(function(FormFieldContactInfoContactField) {
    FormFieldContactInfoContactField["UNDEFINED"] = "UNDEFINED";
    FormFieldContactInfoContactField["FIRST_NAME"] = "FIRST_NAME";
    FormFieldContactInfoContactField["LAST_NAME"] = "LAST_NAME";
    FormFieldContactInfoContactField["COMPANY"] = "COMPANY";
    FormFieldContactInfoContactField["POSITION"] = "POSITION";
    FormFieldContactInfoContactField["EMAIL"] = "EMAIL";
    FormFieldContactInfoContactField["PHONE"] = "PHONE";
    FormFieldContactInfoContactField["ADDRESS"] = "ADDRESS";
    FormFieldContactInfoContactField["BIRTHDATE"] = "BIRTHDATE";
    FormFieldContactInfoContactField["EXTENDED_FIELD"] = "EXTENDED_FIELD";
})(FormFieldContactInfoContactField = exports.FormFieldContactInfoContactField || (exports.FormFieldContactInfoContactField = {}));
var FormFieldDefinitionFieldType;
(function(FormFieldDefinitionFieldType) {
    FormFieldDefinitionFieldType["UNDEFINED"] = "UNDEFINED";
    FormFieldDefinitionFieldType["TEXT"] = "TEXT";
    FormFieldDefinitionFieldType["PARAGRAPH"] = "PARAGRAPH";
    FormFieldDefinitionFieldType["CHECKBOX"] = "CHECKBOX";
    FormFieldDefinitionFieldType["DATE_PICKER"] = "DATE_PICKER";
    FormFieldDefinitionFieldType["DROP_DOWN"] = "DROP_DOWN";
    FormFieldDefinitionFieldType["RADIO_GROUP"] = "RADIO_GROUP";
    FormFieldDefinitionFieldType["CHECKBOX_GROUP"] = "CHECKBOX_GROUP";
    FormFieldDefinitionFieldType["NUMBER"] = "NUMBER";
    FormFieldDefinitionFieldType["EMAIL"] = "EMAIL";
    FormFieldDefinitionFieldType["PHONE"] = "PHONE";
    FormFieldDefinitionFieldType["URL"] = "URL";
    FormFieldDefinitionFieldType["SIGNATURE"] = "SIGNATURE";
    FormFieldDefinitionFieldType["RATING"] = "RATING";
    FormFieldDefinitionFieldType["MULTI_LINE_ADDRESS"] = "MULTI_LINE_ADDRESS";
    FormFieldDefinitionFieldType["PHONE_COUNTRY_CODE"] = "PHONE_COUNTRY_CODE";
    FormFieldDefinitionFieldType["AUTO_COMPLETE_ADDRESS"] = "AUTO_COMPLETE_ADDRESS";
    FormFieldDefinitionFieldType["ATTACHMENTS"] = "ATTACHMENTS";
})(FormFieldDefinitionFieldType = exports.FormFieldDefinitionFieldType || (exports.FormFieldDefinitionFieldType = {}));
var FormFieldViewInfoFieldType;
(function(FormFieldViewInfoFieldType) {
    FormFieldViewInfoFieldType["UNDEFINED"] = "UNDEFINED";
    FormFieldViewInfoFieldType["TEXT"] = "TEXT";
    FormFieldViewInfoFieldType["PARAGRAPH"] = "PARAGRAPH";
    FormFieldViewInfoFieldType["CHECKBOX"] = "CHECKBOX";
    FormFieldViewInfoFieldType["DATE_PICKER"] = "DATE_PICKER";
    FormFieldViewInfoFieldType["DROP_DOWN"] = "DROP_DOWN";
    FormFieldViewInfoFieldType["RADIO_GROUP"] = "RADIO_GROUP";
    FormFieldViewInfoFieldType["CHECKBOX_GROUP"] = "CHECKBOX_GROUP";
    FormFieldViewInfoFieldType["NUMBER"] = "NUMBER";
    FormFieldViewInfoFieldType["EMAIL"] = "EMAIL";
    FormFieldViewInfoFieldType["PHONE"] = "PHONE";
    FormFieldViewInfoFieldType["URL"] = "URL";
    FormFieldViewInfoFieldType["SIGNATURE"] = "SIGNATURE";
    FormFieldViewInfoFieldType["RATING"] = "RATING";
    FormFieldViewInfoFieldType["MULTI_LINE_ADDRESS"] = "MULTI_LINE_ADDRESS";
    FormFieldViewInfoFieldType["PHONE_COUNTRY_CODE"] = "PHONE_COUNTRY_CODE";
    FormFieldViewInfoFieldType["AUTO_COMPLETE_ADDRESS"] = "AUTO_COMPLETE_ADDRESS";
    FormFieldViewInfoFieldType["ATTACHMENTS"] = "ATTACHMENTS";
})(FormFieldViewInfoFieldType = exports.FormFieldViewInfoFieldType || (exports.FormFieldViewInfoFieldType = {}));
var FormType;
(function(FormType) {
    FormType["UNDEFINED"] = "UNDEFINED";
    FormType["SUBSCRIPTION"] = "SUBSCRIPTION";
    FormType["REGISTRATION"] = "REGISTRATION";
    FormType["PAYMENT"] = "PAYMENT";
    FormType["CAPTCHA"] = "CAPTCHA";
    FormType["MULTI_STEP"] = "MULTI_STEP";
    FormType["NO_INBOX_NOTIFICATION"] = "NO_INBOX_NOTIFICATION";
    FormType["LIMITS"] = "LIMITS";
    FormType["APP"] = "APP";
})(FormType = exports.FormType || (exports.FormType = {}));
var LimitRulesDateOperator;
(function(LimitRulesDateOperator) {
    LimitRulesDateOperator["BEFORE"] = "BEFORE";
    LimitRulesDateOperator["AFTER"] = "AFTER";
})(LimitRulesDateOperator = exports.LimitRulesDateOperator || (exports.LimitRulesDateOperator = {}));
var LimitRulesTimeFrame;
(function(LimitRulesTimeFrame) {
    LimitRulesTimeFrame["TOTAL"] = "TOTAL";
    LimitRulesTimeFrame["MINUTE"] = "MINUTE";
    LimitRulesTimeFrame["HOUR"] = "HOUR";
    LimitRulesTimeFrame["DAY"] = "DAY";
    LimitRulesTimeFrame["WEEK"] = "WEEK";
    LimitRulesTimeFrame["MONTH"] = "MONTH";
    LimitRulesTimeFrame["YEAR"] = "YEAR";
})(LimitRulesTimeFrame = exports.LimitRulesTimeFrame || (exports.LimitRulesTimeFrame = {}));
var MediaType;
(function(MediaType) {
    MediaType["UNDEFINED"] = "UNDEFINED";
    MediaType["IMAGE_AND_VIDEO"] = "IMAGE_AND_VIDEO";
    MediaType["IMAGE"] = "IMAGE";
    MediaType["VIDEO"] = "VIDEO";
    MediaType["DOCUMENT"] = "DOCUMENT";
    MediaType["AUDIO"] = "AUDIO";
})(MediaType = exports.MediaType || (exports.MediaType = {}));
var Operation;
(function(Operation) {
    Operation["UNDEFINED"] = "UNDEFINED";
    Operation["SHOW"] = "SHOW";
    Operation["HIDE"] = "HIDE";
    Operation["REQUIRED"] = "REQUIRED";
    Operation["OPTIONAL"] = "OPTIONAL";
})(Operation = exports.Operation || (exports.Operation = {}));
var PhoneInfoTag;
(function(PhoneInfoTag) {
    PhoneInfoTag["UNTAGGED"] = "UNTAGGED";
    PhoneInfoTag["MAIN"] = "MAIN";
    PhoneInfoTag["HOME"] = "HOME";
    PhoneInfoTag["MOBILE"] = "MOBILE";
    PhoneInfoTag["WORK"] = "WORK";
    PhoneInfoTag["FAX"] = "FAX";
})(PhoneInfoTag = exports.PhoneInfoTag || (exports.PhoneInfoTag = {}));
var Plugin;
(function(Plugin) {
    Plugin["FORM_BUILDER"] = "FORM_BUILDER";
    Plugin["GET_SUBSCRIBERS"] = "GET_SUBSCRIBERS";
    Plugin["REGISTRATION_FORM"] = "REGISTRATION_FORM";
    Plugin["PAYMENT_FORM"] = "PAYMENT_FORM";
    Plugin["CAPTCHA"] = "CAPTCHA";
    Plugin["MULTI_STEP"] = "MULTI_STEP";
    Plugin["INBOX_OPT_OUT"] = "INBOX_OPT_OUT";
    Plugin["LIMIT_RULES"] = "LIMIT_RULES";
    Plugin["APP"] = "APP";
})(Plugin = exports.Plugin || (exports.Plugin = {}));
var PublishedStatus;
(function(PublishedStatus) {
    PublishedStatus["UNDEFINED"] = "UNDEFINED";
    PublishedStatus["PUBLISHED"] = "PUBLISHED";
    PublishedStatus["DELETED"] = "DELETED";
    PublishedStatus["DRAFT"] = "DRAFT";
    PublishedStatus["DISCARDED"] = "DISCARDED";
    PublishedStatus["ARCHIVED"] = "ARCHIVED";
})(PublishedStatus = exports.PublishedStatus || (exports.PublishedStatus = {}));
var SortOrder;
(function(SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder = exports.SortOrder || (exports.SortOrder = {}));
var SourceType;
(function(SourceType) {
    SourceType["UNDEFINED"] = "UNDEFINED";
    SourceType["APP"] = "APP";
})(SourceType = exports.SourceType || (exports.SourceType = {}));
var Status;
(function(Status) {
    Status["OK"] = "OK";
    Status["LIMIT_REACHED"] = "LIMIT_REACHED";
})(Status = exports.Status || (exports.Status = {}));
var SubmissionsLimitRuleRequestTimeFrame;
(function(SubmissionsLimitRuleRequestTimeFrame) {
    SubmissionsLimitRuleRequestTimeFrame["HOUR"] = "HOUR";
    SubmissionsLimitRuleRequestTimeFrame["MINUTE"] = "MINUTE";
    SubmissionsLimitRuleRequestTimeFrame["DAY"] = "DAY";
    SubmissionsLimitRuleRequestTimeFrame["WEEK"] = "WEEK";
    SubmissionsLimitRuleRequestTimeFrame["MONTH"] = "MONTH";
    SubmissionsLimitRuleRequestTimeFrame["YEAR"] = "YEAR";
})(SubmissionsLimitRuleRequestTimeFrame = exports.SubmissionsLimitRuleRequestTimeFrame || (exports.SubmissionsLimitRuleRequestTimeFrame = {}));
var SubscribeMode;
(function(SubscribeMode) {
    SubscribeMode["DIRECT"] = "DIRECT";
    SubscribeMode["DOUBLE_OPT"] = "DOUBLE_OPT";
})(SubscribeMode = exports.SubscribeMode || (exports.SubscribeMode = {}));
var SubscriptionSettingsSubscriptionMode;
(function(SubscriptionSettingsSubscriptionMode) {
    SubscriptionSettingsSubscriptionMode["UNDEFINED"] = "UNDEFINED";
    SubscriptionSettingsSubscriptionMode["ON_SUBMIT"] = "ON_SUBMIT";
    SubscriptionSettingsSubscriptionMode["AFTER_CONFIRMATION"] = "AFTER_CONFIRMATION";
})(SubscriptionSettingsSubscriptionMode = exports.SubscriptionSettingsSubscriptionMode || (exports.SubscriptionSettingsSubscriptionMode = {}));
var SubscriptionSubscriptionMode;
(function(SubscriptionSubscriptionMode) {
    SubscriptionSubscriptionMode["UNDEFINED"] = "UNDEFINED";
    SubscriptionSubscriptionMode["ON_SUBMIT"] = "ON_SUBMIT";
    SubscriptionSubscriptionMode["AFTER_CONFIRMATION"] = "AFTER_CONFIRMATION";
})(SubscriptionSubscriptionMode = exports.SubscriptionSubscriptionMode || (exports.SubscriptionSubscriptionMode = {}));
var TimeFormat;
(function(TimeFormat) {
    TimeFormat["AMPM"] = "AMPM";
    TimeFormat["FULL"] = "FULL";
})(TimeFormat = exports.TimeFormat || (exports.TimeFormat = {}));
var TransactionStatus;
(function(TransactionStatus) {
    TransactionStatus["APPROVED"] = "APPROVED";
    TransactionStatus["PENDING"] = "PENDING";
    TransactionStatus["FAILED"] = "FAILED";
    TransactionStatus["OTHER"] = "OTHER";
    TransactionStatus["MANUAL"] = "MANUAL";
})(TransactionStatus = exports.TransactionStatus || (exports.TransactionStatus = {}));
var Type;
(function(Type) {
    Type["UNDEFINED"] = "UNDEFINED";
    Type["FIRST_NAME"] = "FIRST_NAME";
    Type["LAST_NAME"] = "LAST_NAME";
    Type["COMPANY"] = "COMPANY";
    Type["POSITION"] = "POSITION";
    Type["UNTAGGED_EMAIL"] = "UNTAGGED_EMAIL";
    Type["MAIN_EMAIL"] = "MAIN_EMAIL";
    Type["HOME_EMAIL"] = "HOME_EMAIL";
    Type["WORK_EMAIL"] = "WORK_EMAIL";
    Type["UNTAGGED_PHONE"] = "UNTAGGED_PHONE";
    Type["MAIN_PHONE"] = "MAIN_PHONE";
    Type["HOME_PHONE"] = "HOME_PHONE";
    Type["WORK_PHONE"] = "WORK_PHONE";
    Type["MOBILE_PHONE"] = "MOBILE_PHONE";
    Type["FAX"] = "FAX";
    Type["UNTAGGED_ADDRESS"] = "UNTAGGED_ADDRESS";
    Type["HOME_ADDRESS"] = "HOME_ADDRESS";
    Type["WORK_ADDRESS"] = "WORK_ADDRESS";
    Type["BILLING_ADDRESS"] = "BILLING_ADDRESS";
    Type["SHIPPING_ADDRESS"] = "SHIPPING_ADDRESS";
    Type["BIRTHDATE"] = "BIRTHDATE";
    Type["EXTENDED_FIELD"] = "EXTENDED_FIELD";
})(Type = exports.Type || (exports.Type = {}));
var V1FormStatus;
(function(V1FormStatus) {
    V1FormStatus["UNDEFINED"] = "UNDEFINED";
    V1FormStatus["PUBLISHED"] = "PUBLISHED";
    V1FormStatus["DELETED"] = "DELETED";
    V1FormStatus["DRAFT"] = "DRAFT";
    V1FormStatus["DISCARDED"] = "DISCARDED";
    V1FormStatus["ARCHIVED"] = "ARCHIVED";
})(V1FormStatus = exports.V1FormStatus || (exports.V1FormStatus = {}));
var V2FormStatus;
(function(V2FormStatus) {
    V2FormStatus["UNDEFINED_FORM_STATUS"] = "UNDEFINED_FORM_STATUS";
    V2FormStatus["PUBLISHED"] = "PUBLISHED";
    V2FormStatus["DELETED"] = "DELETED";
    V2FormStatus["DRAFT"] = "DRAFT";
    V2FormStatus["ARCHIVED"] = "ARCHIVED";
})(V2FormStatus = exports.V2FormStatus || (exports.V2FormStatus = {}));
var Version;
(function(Version) {
    Version["UNDEFINED"] = "UNDEFINED";
    Version["V1"] = "V1";
    Version["V2"] = "V2";
})(Version = exports.Version || (exports.Version = {}));
var ViewMode;
(function(ViewMode) {
    ViewMode["SITE"] = "SITE";
    ViewMode["PREVIEW"] = "PREVIEW";
})(ViewMode = exports.ViewMode || (exports.ViewMode = {}));
var ViolationType;
(function(ViolationType) {
    ViolationType["QUOTA_REACHED"] = "QUOTA_REACHED";
    ViolationType["MISSING_IP_ADDRESS"] = "MISSING_IP_ADDRESS";
    ViolationType["SPAM_CONTENT"] = "SPAM_CONTENT";
})(ViolationType = exports.ViolationType || (exports.ViolationType = {}));
