// @wix/advanced-seo-utils/dist/src/private/item-types/blog-category/pattern-blog-category.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.BLOG_CATEGORY = void 0;
var builders_1 = require("../../types/builders");
var adapter_blog_category_1 = __importDefault(require("./adapter-blog-category"));
var utils_1 = require("../../adapters/utils");
var LinkTagTypes_1 = require("../../types/LinkTagTypes");
var utils_2 = require("../utils");
var IDs = adapter_blog_category_1.default.IDs;
exports.BLOG_CATEGORY = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.NAME + "}}")
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:url', "{{" + IDs.PAGE_URL + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'website'),
        builders_1.buildOgMeta('og:image', "{{" + IDs.IMAGE + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.IMAGE_HEIGHT + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.IMAGE_WIDTH + "}}"),
        builders_1.buildStandardMeta('robots', 'index'),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'prevUrl',
            href: "{{" + IDs.PREV_PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'nextUrl',
            href: "{{" + IDs.NEXT_PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'alternate',
            type: LinkTagTypes_1.LINK_TAG_TYPES.RSS,
            href: "{{" + IDs.BLOG_FEED_ROUTE + "}}",
            title: utils_2.getRssTitle("{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        })
    ], builders_1.buildTwitterMetaRefs()),
};
