// @wix/advanced-seo-utils/dist/src/private/item-types/forum-post/schema-presets/forum-post-questions-and-answers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
var adapter_forum_post_1 = __importDefault(require("../adapter-forum-post"));
var IDs = adapter_forum_post_1.default.IDs,
    SchemaTypes = adapter_forum_post_1.default.SchemaTypes;
exports.default = {
    type: SchemaTypes.FORUM_POST_QUESTIONS_AND_ANSWERS,
    schema: {
        '@context': 'https://schema.org',
        '@type': 'QAPage',
        mainEntity: {
            '@type': 'Question',
            name: "{{" + IDs.TITLE + "}}",
            text: "{{" + IDs.CONTENT_FULL + "}}",
            answerCount: "{{" + IDs.COMMENT_AMOUNT + "}}",
            upvoteCount: "{{" + IDs.LIKE_AMOUNT + "}}",
            dateCreated: "{{" + IDs.PUBLISH_DATE + "}}",
            author: {
                '@type': 'Person',
                name: "{{" + IDs.AUTHOR + "}}",
            },
            acceptedAnswer: "{{" + IDs.QUESTIONS_AND_ANSWERS_SD_ACCEPTED_ANSWER + "}}",
            suggestedAnswer: "{{" + IDs.QUESTIONS_AND_ANSWERS_SD_SUGGESTED_ANSWER + "}}",
        },
    },
};
