// @wix/wixstores-client-storefront-sdk/dist/es/src/apis/BIService.js
import {
    __assign
} from "tslib";
var BIService = /** @class */ (function() {
    function BIService(_a) {
        var _this = this;
        var siteStore = _a.siteStore,
            origin = _a.origin;
        this.updateBuyerNote = function(cart, hasNote) {
            var biParams = __assign({
                storeId: cart.storeId,
                cartId: cart.cartId
            }, _this.baseParams);
            hasNote
                ?
                _this.siteStore.biLogger.saveNoteToSellerSf(__assign(__assign({}, biParams), {
                    is_empty: false
                })) :
                _this.siteStore.biLogger.deleteNoteToSellerSf(biParams);
        };
        this.siteStore = siteStore;
        this.origin = origin;
    }
    Object.defineProperty(BIService.prototype, "baseParams", {
        get: function() {
            return {
                origin: this.origin,
            };
        },
        enumerable: false,
        configurable: true
    });
    return BIService;
}());
export {
    BIService
};
