// @wix/advanced-seo-utils/dist/src/renderer.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getMultilingualTags = exports.getMultilingualSchemaFromContext = exports.hasAdapterMultilingualData = exports.getDynamicPatternsSchema = exports.isComponentItemType = exports.getIndexableValue = exports.getUserPatternByItemType = exports.TAG_TYPES = exports.resolveWithPatterns = exports.resolvePayload = exports.resolve = exports.renderToStaticMarkup = exports.ITEM_TYPES = exports.getTags = exports.getPatternBlob = exports.getDefaultItemPayload = exports.getAdapter = exports.filterPageLevelTags = exports.IDENTIFIERS = exports.CONTEXT_PROPS = exports.buildOgImageFullUrl = exports.buildOgImagePreview = void 0;
var resolve_1 = require("./private/renderer/resolve");
Object.defineProperty(exports, "resolve", {
    enumerable: true,
    get: function() {
        return resolve_1.resolve;
    }
});
var resolve_with_patterns_1 = require("./private/renderer/resolve-with-patterns");
Object.defineProperty(exports, "resolveWithPatterns", {
    enumerable: true,
    get: function() {
        return resolve_with_patterns_1.resolveWithPatterns;
    }
});
var resolve_payload_1 = require("./private/renderer/resolve-payload");
Object.defineProperty(exports, "resolvePayload", {
    enumerable: true,
    get: function() {
        return resolve_payload_1.resolvePayload;
    }
});
var TagTypes_1 = require("./private/types/TagTypes");
Object.defineProperty(exports, "TAG_TYPES", {
    enumerable: true,
    get: function() {
        return TagTypes_1.TAG_TYPES;
    }
});
var ContextProps_1 = require("./private/types/ContextProps");
Object.defineProperty(exports, "CONTEXT_PROPS", {
    enumerable: true,
    get: function() {
        return ContextProps_1.CONTEXT_PROPS;
    }
});
var ItemTypes_1 = require("./private/types/ItemTypes");
Object.defineProperty(exports, "ITEM_TYPES", {
    enumerable: true,
    get: function() {
        return ItemTypes_1.ITEM_TYPES;
    }
});
var Identifiers_1 = require("./private/types/Identifiers");
Object.defineProperty(exports, "IDENTIFIERS", {
    enumerable: true,
    get: function() {
        return Identifiers_1.IDENTIFIERS;
    }
});
var filter_page_level_tags_1 = require("./private/tags/filters/filter-page-level-tags");
Object.defineProperty(exports, "filterPageLevelTags", {
    enumerable: true,
    get: function() {
        return filter_page_level_tags_1.filterPageLevelTags;
    }
});
var render_to_static_markup_1 = require("./private/renderer/render-to-static-markup");
Object.defineProperty(exports, "renderToStaticMarkup", {
    enumerable: true,
    get: function() {
        return render_to_static_markup_1.renderToStaticMarkup;
    }
});
var adapters_1 = require("./private/adapters");
Object.defineProperty(exports, "getAdapter", {
    enumerable: true,
    get: function() {
        return adapters_1.getAdapter;
    }
});
var patterns_1 = require("./private/patterns");
Object.defineProperty(exports, "getPatternBlob", {
    enumerable: true,
    get: function() {
        return patterns_1.getPatternBlob;
    }
});
Object.defineProperty(exports, "getUserPatternByItemType", {
    enumerable: true,
    get: function() {
        return patterns_1.getUserPatternByItemType;
    }
});
var build_og_image_1 = require("./private/utils/build-og-image");
Object.defineProperty(exports, "buildOgImageFullUrl", {
    enumerable: true,
    get: function() {
        return build_og_image_1.buildOgImageFullUrl;
    }
});
var build_og_image_preview_1 = require("./private/utils/build-og-image-preview");
Object.defineProperty(exports, "buildOgImagePreview", {
    enumerable: true,
    get: function() {
        return build_og_image_preview_1.buildOgImagePreview;
    }
});
var get_default_item_1 = require("./private/patterns/get-default-item");
Object.defineProperty(exports, "getDefaultItemPayload", {
    enumerable: true,
    get: function() {
        return get_default_item_1.getDefaultItemPayload;
    }
});
var get_tags_async_1 = require("./private/renderer/get-tags-async");
Object.defineProperty(exports, "getTags", {
    enumerable: true,
    get: function() {
        return get_tags_async_1.getTags;
    }
});
var resolve_is_indexable_1 = require("./private/renderer/utils/resolve-is-indexable");
Object.defineProperty(exports, "getIndexableValue", {
    enumerable: true,
    get: function() {
        return resolve_is_indexable_1.getIndexableValue;
    }
});
var is_component_item_type_1 = require("./private/renderer/is-component-item-type");
Object.defineProperty(exports, "isComponentItemType", {
    enumerable: true,
    get: function() {
        return is_component_item_type_1.isComponentItemType;
    }
});
var get_dynamic_patterns_schema_1 = require("./private/renderer/get-dynamic-patterns-schema");
Object.defineProperty(exports, "getDynamicPatternsSchema", {
    enumerable: true,
    get: function() {
        return get_dynamic_patterns_schema_1.getDynamicPatternsSchema;
    }
});
var has_adapter_multilingual_data_1 = require("./private/renderer/has-adapter-multilingual-data");
Object.defineProperty(exports, "hasAdapterMultilingualData", {
    enumerable: true,
    get: function() {
        return has_adapter_multilingual_data_1.hasAdapterMultilingualData;
    }
});
var get_multilingual_tags_1 = require("./private/renderer/get-multilingual-tags");
Object.defineProperty(exports, "getMultilingualSchemaFromContext", {
    enumerable: true,
    get: function() {
        return get_multilingual_tags_1.getMultilingualSchemaFromContext;
    }
});
Object.defineProperty(exports, "getMultilingualTags", {
    enumerable: true,
    get: function() {
        return get_multilingual_tags_1.getMultilingualTags;
    }
});
var setters_1 = require("./private/renderer/setters");
Object.defineProperty(exports, "addSchema", {
    enumerable: true,
    get: function() {
        return setters_1.addSchema;
    }
});
Object.defineProperty(exports, "setCanonical", {
    enumerable: true,
    get: function() {
        return setters_1.setCanonical;
    }
});
Object.defineProperty(exports, "setDescription", {
    enumerable: true,
    get: function() {
        return setters_1.setDescription;
    }
});
Object.defineProperty(exports, "setExternalRouter", {
    enumerable: true,
    get: function() {
        return setters_1.setExternalRouter;
    }
});
Object.defineProperty(exports, "setIndexable", {
    enumerable: true,
    get: function() {
        return setters_1.setIndexable;
    }
});
Object.defineProperty(exports, "setLinks", {
    enumerable: true,
    get: function() {
        return setters_1.setLinks;
    }
});
Object.defineProperty(exports, "setMetaTags", {
    enumerable: true,
    get: function() {
        return setters_1.setMetaTags;
    }
});
Object.defineProperty(exports, "setSchemas", {
    enumerable: true,
    get: function() {
        return setters_1.setSchemas;
    }
});
Object.defineProperty(exports, "setTitle", {
    enumerable: true,
    get: function() {
        return setters_1.setTitle;
    }
});
Object.defineProperty(exports, "removeTagsByType", {
    enumerable: true,
    get: function() {
        return setters_1.removeTagsByType;
    }
});
Object.defineProperty(exports, "updateValueByIdentifier", {
    enumerable: true,
    get: function() {
        return setters_1.updateValueByIdentifier;
    }
});
var getters_1 = require("./private/renderer/getters");
Object.defineProperty(exports, "addMetaTag", {
    enumerable: true,
    get: function() {
        return getters_1.addMetaTag;
    }
});
Object.defineProperty(exports, "getCanonical", {
    enumerable: true,
    get: function() {
        return getters_1.getCanonical;
    }
});
Object.defineProperty(exports, "getDescription", {
    enumerable: true,
    get: function() {
        return getters_1.getDescription;
    }
});
Object.defineProperty(exports, "getLinks", {
    enumerable: true,
    get: function() {
        return getters_1.getLinks;
    }
});
Object.defineProperty(exports, "getMetaTags", {
    enumerable: true,
    get: function() {
        return getters_1.getMetaTags;
    }
});
Object.defineProperty(exports, "getSchemas", {
    enumerable: true,
    get: function() {
        return getters_1.getSchemas;
    }
});
Object.defineProperty(exports, "getStaticMarkup", {
    enumerable: true,
    get: function() {
        return getters_1.getStaticMarkup;
    }
});
Object.defineProperty(exports, "getTitle", {
    enumerable: true,
    get: function() {
        return getters_1.getTitle;
    }
});
Object.defineProperty(exports, "getValueByIdentifier", {
    enumerable: true,
    get: function() {
        return getters_1.getValueByIdentifier;
    }
});
