// @wix/advanced-seo-utils/dist/src/private/item-types/bookings-service/pattern-bookings-service.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.BOOKINGS_SERVICE = void 0;
var builders_1 = require("../../types/builders");
var utils_1 = require("../../adapters/utils");
var adapter_bookings_service_1 = __importDefault(require("./adapter-bookings-service"));
var bookings_service_1 = __importDefault(require("./schema-presets/bookings-service"));
var IDs = adapter_bookings_service_1.default.IDs;
/**
 Here you can find more details about the implementation
 https://docs.google.com/document/d/1fLagCkL9Q_9G1jmVY6umrCqCnIDULCaD0Wt2JPPdtJc/edit#
*/
exports.BOOKINGS_SERVICE = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.NAME + "}} | {{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildStandardMeta('description', "{{" + IDs.DESCRIPTION + "}}"),
        builders_1.buildStandardMeta('robots', 'index')
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:image', "{{" + IDs.IMAGE_URL + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.IMAGE_WIDTH + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.IMAGE_HEIGHT + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'website'),
        builders_1.buildOgMeta('og:url', "{{" + IDs.CALCULATED_SERVICE_URL + "}}"),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.CALCULATED_SERVICE_URL + "}}"
        })
    ], builders_1.buildTwitterMetaRefs(), [
        builders_1.buildStructuredData({
            schema: bookings_service_1.default.schema,
            schemaType: bookings_service_1.default.type,
            disabled: [
                "{{" + IDs.SD_DISABLE_SCHEMA + "}}",
                '{{options.disable-pattern-schema}}',
            ],
        }),
    ]),
};
