// @wix/ambassador-members-v1-member-to-member-block/build/es/http.js
import {
    toURLSearchParams
} from '@wix/metro-runtime';
import {
    serializer
} from '@wix/metro-runtime/ambassador';
import {
    resolveUrl
} from '@wix/metro-runtime';
var _blockMemberRequest = {};
var _blockMemberResponse = {};
var _listCurrentMemberBlockingRequest = {};
var _listCurrentMemberBlockingResponse = {};
var _listMemberBlocksRequest = {};
var _listMemberBlocksResponse = {};
var _unblockMemberRequest = {};
var _unblockMemberResponse = {};

function resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(opts) {
    var domainToMappings = {
        'api._api_base_domain_': [{
            srcPath: '/members/v1/member-to-member-blocks',
            destPath: '',
        }, ],
        'www.wixapis.com': [{
            srcPath: '/members/v1/member-to-member-blocks',
            destPath: '/v1/member-to-member-blocks',
        }, ],
        _: [{
            srcPath: '/_api/members/v1/member-to-member-blocks',
            destPath: '/v1/member-to-member-blocks',
        }, ],
        'members.wixapps.net': [{
            srcPath: '/_api/members/v1/member-to-member-blocks',
            destPath: '/v1/member-to-member-blocks',
        }, ],
    };
    return resolveUrl(Object.assign(opts, {
        domainToMappings: domainToMappings
    }));
}
/** Blocks the specified member from the current member. */
export function blockMember(payload) {
    return function __blockMember(_a) {
        var host = _a.host;
        var toJSON = serializer(_blockMemberRequest, {}).toJSON;
        var fromJSON = serializer(_blockMemberResponse, {}).fromJSON;
        var serializedData = toJSON(payload);
        var metadata = {
            method: 'POST',
            url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl({
                protoPath: '/v1/member-to-member-blocks/members/my',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromJSON,
        };
        return metadata;
    };
}
/** Unblocks the specified member from the current member. */
export function unblockMember(payload) {
    return function __unblockMember(_a) {
        var host = _a.host;
        var toJSON = serializer(_unblockMemberRequest, {}).toJSON;
        var fromJSON = serializer(_unblockMemberResponse, {}).fromJSON;
        var serializedData = toJSON(payload);
        var metadata = {
            method: 'DELETE',
            url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl({
                protoPath: '/v1/member-to-member-blocks/members/my/{memberId}',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromJSON,
        };
        return metadata;
    };
}
/** Returns information about members blocked by the current member. */
export function listCurrentMemberBlocking(payload) {
    return function __listCurrentMemberBlocking(_a) {
        var host = _a.host;
        var toJSON = serializer(_listCurrentMemberBlockingRequest, {}).toJSON;
        var fromJSON = serializer(_listCurrentMemberBlockingResponse, {}).fromJSON;
        var serializedData = toJSON(payload);
        var metadata = {
            method: 'GET',
            url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl({
                protoPath: '/v1/member-to-member-blocks/members/my',
                data: serializedData,
                host: host,
            }),
            params: toURLSearchParams(serializedData),
            transformResponse: fromJSON,
        };
        return metadata;
    };
}
/** Returns IDs of members that are blocking or blocked by the member whose ID was provided. */
export function listMemberBlocks(payload) {
    return function __listMemberBlocks(_a) {
        var host = _a.host;
        var toJSON = serializer(_listMemberBlocksRequest, {}).toJSON;
        var fromJSON = serializer(_listMemberBlocksResponse, {}).fromJSON;
        var serializedData = toJSON(payload);
        var metadata = {
            method: 'GET',
            url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl({
                protoPath: '/v1/member-to-member-blocks/members/{memberId}',
                data: serializedData,
                host: host,
            }),
            params: toURLSearchParams(serializedData),
            transformResponse: fromJSON,
        };
        return metadata;
    };
}
