// @wix/advanced-seo-utils/dist/src/private/item-types/blog-hashtags/pattern-blog-hashtags.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.BLOG_HASHTAGS = void 0;
var utils_1 = require("../../adapters/utils");
var adapter_blog_hashtags_1 = __importDefault(require("./adapter-blog-hashtags"));
var builders_1 = require("../../types/builders");
var utils_2 = require("../utils");
var LinkTagTypes_1 = require("../../types/LinkTagTypes");
var IDs = adapter_blog_hashtags_1.default.IDs;
var SITE_NAME = utils_1.siteIDs.SITE_NAME;
exports.BLOG_HASHTAGS = {
    tags: [
        builders_1.buildTitle("{{" + IDs.TITLE + "}}"),
        builders_1.buildStandardMeta('robots', 'noindex'),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'prev',
            href: "{{" + IDs.PREV_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'next',
            href: "{{" + IDs.NEXT_URL + "}}"
        }),
        builders_1.buildOgMeta('og:title', "{{" + IDs.TITLE + "}}"),
        builders_1.buildOgMeta('og:url', "{{" + IDs.URL + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'website'),
        builders_1.buildLink({
            rel: 'alternate',
            type: LinkTagTypes_1.LINK_TAG_TYPES.RSS,
            href: "{{" + IDs.BLOG_FEED_ROUTE + "}}",
            title: utils_2.getRssTitle("{{" + SITE_NAME + "}}"),
        }),
    ],
};
