// @wix/bookings-analytics-adapter/dist/esm/analyticsAdapter.js
import {
    __assign
} from "tslib";
import {
    BOOKINGS_APP_DEF_ID
} from '@wix/bookings-uou-domain';
import {
    ServiceType,
} from '@wix/bookings-uou-types';
var EVENT_ORIGIN_BOOKINGS = 'Bookings';
export var basePayload = {
    origin: EVENT_ORIGIN_BOOKINGS,
    appDefId: BOOKINGS_APP_DEF_ID,
};
var getVariant = function(serviceType) {
    switch (serviceType) {
        case ServiceType.INDIVIDUAL:
            return 'PRIVATE';
        case ServiceType.GROUP:
            return 'GROUP';
        case ServiceType.COURSE:
        default:
            return 'COURSE';
    }
};
var getServiceTrackingEventFromCatalogServiceDto = function(service, businessName) {
    return ({
        id: service.id,
        name: service.info.name,
        brand: businessName,
        category: '',
        variant: getVariant(service.type),
        price: service.payment.price,
        currency: service.payment.currency,
    });
};
var getServiceTrackingEventFromService = function(service, businessName) {
    return ({
        id: service.id,
        name: service.info.name,
        brand: businessName,
        category: '',
        variant: getVariant(service.info.type),
        price: service.payment.paymentDetails.price,
        currency: service.payment.paymentDetails.currency,
    });
};
export var getTrackingInfoForBookButtonClick = function(_a) {
    var service = _a.service,
        businessName = _a.businessName;
    return {
        eventId: 'InitiateCheckout',
        payload: __assign(__assign({}, basePayload), {
            contents: [
                getServiceTrackingEventFromCatalogServiceDto(service, businessName),
            ]
        }),
    };
};
export var getTrackingInfoForNavigateToServicePageClick = function(_a) {
    var service = _a.service,
        businessName = _a.businessName;
    return {
        eventId: 'ClickProduct',
        payload: __assign(__assign(__assign({}, basePayload), getServiceTrackingEventFromCatalogServiceDto(service, businessName)), {
            list: 'Service Widget',
            position: 0
        }),
    };
};
export var getTrackingInfoForServicePageLoads = function(_a) {
    var service = _a.service,
        businessName = _a.businessName;
    return {
        eventId: 'ViewContent',
        payload: __assign(__assign({}, basePayload), getServiceTrackingEventFromCatalogServiceDto(service, businessName)),
    };
};
export var getTrackingInfoForCalendarNextButtonClick = function(_a) {
    var services = _a.services,
        businessName = _a.businessName;
    return {
        eventId: 'CustomEvent',
        payload: __assign(__assign({}, basePayload), {
            contents: services.map(function(service) {
                return getServiceTrackingEventFromService(service, businessName);
            }),
            eventCategory: 'Enhanced Ecommerce - Bookings',
            eventAction: 'Select Date'
        }),
    };
};
