// @wix/advanced-seo-utils/dist/src/private/item-types/events-page/pattern-events-page.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.EVENTS_PAGE = void 0;
var builders_1 = require("../../types/builders");
var utils_1 = require("../../adapters/utils");
var adapter_events_page_1 = __importDefault(require("./adapter-events-page"));
var events_page_1 = __importDefault(require("./schema-presets/events-page"));
/**
 Here you can find more details about the implementation
 https://docs.google.com/document/d/1n-QE6LXncd_wqz9eQKufxMWnalE-WYxEzdrZ7rixF2A/edit#heading=h.8n2lir49tnp7
*/
var IDs = adapter_events_page_1.default.IDs;
exports.EVENTS_PAGE = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.NAME + "}} | {{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildStandardMeta('description', "{{" + IDs.CALCULATED_SEO_DESCRIPTION + "}}"),
        builders_1.buildStandardMeta('robots', 'index')
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:image', "{{" + IDs.CALCULATED_MAIN_IMAGE_URL + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.IMAGE_WIDTH + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.IMAGE_HEIGHT + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'website'),
        builders_1.buildOgMeta('og:url', "{{" + IDs.CALCULATED_EVENT_URL + "}}"),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.CALCULATED_EVENT_URL + "}}"
        })
    ], builders_1.buildTwitterMetaRefs(), [
        builders_1.buildStructuredData({
            schema: events_page_1.default.schema,
            schemaType: events_page_1.default.type,
            disabled: [
                "{{" + IDs.SD_DISABLE_SCHEMA + "}}",
                '{{options.disable-pattern-schema}}',
            ],
        }),
    ]),
};
