// @wix/advanced-seo-utils/dist/src/private/tags/advanced/get-simplified-tag.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getSimplifiedTag = void 0;
var identify_tag_1 = require("./identify-tag");
var Identifiers_1 = require("../../types/Identifiers");
var render_to_static_markup_1 = require("../../renderer/render-to-static-markup");
exports.getSimplifiedTag = function(seoTag) {
    var type = identify_tag_1.identifyTag(seoTag);
    switch (type) {
        case Identifiers_1.GENERAL_IDENTIFIERS.LINK:
            {
                var _a = seoTag.props,
                    rel = _a.rel,
                    href = _a.href;
                return {
                    label: rel,
                    value: href,
                    isDisabled: !!seoTag.disabled,
                    generalIdentifier: Identifiers_1.GENERAL_IDENTIFIERS.LINK,
                    html: render_to_static_markup_1.renderToStaticMarkup({
                        tags: [seoTag]
                    }),
                    isCustom: !!seoTag.custom,
                };
            }
        case Identifiers_1.GENERAL_IDENTIFIERS.OG_TAG:
            {
                return {
                    label: seoTag.props.property,
                    value: seoTag.props.content,
                    isDisabled: !!seoTag.disabled,
                    generalIdentifier: Identifiers_1.GENERAL_IDENTIFIERS.OG_TAG,
                    html: render_to_static_markup_1.renderToStaticMarkup({
                        tags: [seoTag]
                    }),
                    isCustom: !!seoTag.custom,
                };
            }
        case Identifiers_1.GENERAL_IDENTIFIERS.SATANDARD_META:
            {
                return {
                    label: seoTag.props.name,
                    value: seoTag.props.content,
                    isDisabled: !!seoTag.disabled,
                    generalIdentifier: Identifiers_1.GENERAL_IDENTIFIERS.SATANDARD_META,
                    html: render_to_static_markup_1.renderToStaticMarkup({
                        tags: [seoTag]
                    }),
                    isCustom: !!seoTag.custom,
                };
            }
        default:
            {
                return null;
            }
    }
};
