// @wix/advanced-seo-utils/dist/src/private/item-types/stores-product/schema-presets/stores-product-page.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
var adapter_stores_product_1 = __importDefault(require("../adapter-stores-product"));
var IDs = adapter_stores_product_1.default.IDs,
    SchemaTypes = adapter_stores_product_1.default.SchemaTypes;
var schema = {
    '@context': 'https://schema.org/',
    '@type': 'Product',
    name: "{{" + IDs.NAME + "}}",
    description: "{{" + IDs.DESCRIPTION + "}}",
    sku: "{{" + IDs.SKU + "}}",
    brand: {
        '@type': 'Brand',
        name: "{{" + IDs.BRAND + "}}",
    },
    image: {
        '@type': 'ImageObject',
        height: "{{" + IDs.IMAGE_HEIGHT + "}}",
        width: "{{" + IDs.IMAGE_WIDTH + "}}",
        contentUrl: "{{" + IDs.IMAGE_URL + "}}",
        representativeOfPage: true,
        alt: "{{" + IDs.IMAGE_ALT + "}}",
        thumbnail: {
            '@type': 'ImageObject',
            height: "{{" + IDs.THUMBNAIL_HEIGHT + "}}",
            width: "{{" + IDs.THUMBNAIL_WIDTH + "}}",
            contentUrl: "{{" + IDs.THUMBNAIL_URL + "}}",
            alt: "{{" + IDs.THUMBNAIL_ALT + "}}",
        },
    },
    Offers: {
        '@type': 'Offer',
        url: "{{" + IDs.PRODUCT_PAGE_URL + "}}",
        priceCurrency: "{{" + IDs.CURRENCY + "}}",
        price: "{{" + IDs.PRICE + "}}",
        Availability: "{{" + IDs.AVAILABILITY_SCHEMA + "}}",
        seller: {
            '@type': 'Organization',
            name: "{{" + IDs.BUSINESS_NAME + "}}",
            sameAs: "{{" + IDs.SITE_URL + "}}",
        },
    },
};
exports.default = {
    type: SchemaTypes.STORES_PRODUCT_PAGE,
    schema: schema,
};
