// @wix/advanced-seo-utils/dist/src/private/tags/values/get-by-identifier.js
"use strict";
var __assign = (this && this.__assign) || function() {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s)
                if (Object.prototype.hasOwnProperty.call(s, p))
                    t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getTagsByIdentifier = exports.getSchemasArray = exports.getMetaByIdentifier = exports.getValueByIdentifierFromContext = exports.getDisabledByIdentifier = exports.getValueByIdentifier = void 0;
var is_empty_1 = require("../../utils/is-empty");
var validate_tag_structure_1 = require("../utils/validate-tag-structure");
var TagSchemas_1 = require("../../types/TagSchemas");
var resolve_payload_1 = require("../../renderer/resolve-payload");
var fill_in_pattern_blob_1 = require("../../patterns/fill-in-pattern-blob");
var Identifiers_1 = require("../../types/Identifiers");
var consts_1 = require("../../consts");
var resolve_structured_data_1 = require("../structured-data/resolve-structured-data");
var resolve_sd_schema_1 = require("../../patterns/structured-data/resolve-sd-schema");

function getValueByIdentifier(tags, identifier) {
    var result = findTag(tags, identifier);
    return result ? getValueFromTag(result, identifier) : result;
}
exports.getValueByIdentifier = getValueByIdentifier;

function getDisabledByIdentifier(_a, identifier) {
    var tags = (_a === void 0 ? {} : _a).tags;
    var result = findTag(tags, identifier);
    return result ? getDisabledTag(result) : false;
}
exports.getDisabledByIdentifier = getDisabledByIdentifier;

function getDisabledTag(tag) {
    if (is_empty_1.isEmpty(tag)) {
        return false;
    }
    return tag.disabled;
}

function getValueByIdentifierFromContext(payload, identifier, context, referencePath) {
    var tags = resolve_payload_1.resolvePayload(payload).tags;
    var result = findTag(tags, identifier);
    return result ?
        getValueFromTagUsingContext(result, identifier, context, tags, referencePath) :
        result;
}
exports.getValueByIdentifierFromContext = getValueByIdentifierFromContext;

function getMetaByIdentifier(tags, identifier) {
    var result = findTag(tags, identifier);
    return (result ? result.meta : null) || {};
}
exports.getMetaByIdentifier = getMetaByIdentifier;

function getSchemasArray(tags, _a) {
    var _b = _a === void 0 ? {} : _a,
        _c = _b.withMeta,
        withMeta = _c === void 0 ? false : _c,
        defaultSchema = _b.defaultSchema,
        _d = _b.logError,
        logError = _d === void 0 ? function() {} : _d;
    if (is_empty_1.isEmpty(tags)) {
        return [];
    }
    var tagSchema = TagSchemas_1.TAG_SCHEMAS[Identifiers_1.IDENTIFIERS.STRUCTURED_DATA];
    return resolve_structured_data_1.resolveStructuredData(tags)
        .filter(function(tag) {
            return validate_tag_structure_1.validateTagStructure(tag, tagSchema);
        })
        .map(function(tag) {
            var _a = tag.meta || {},
                schemaType = _a.schemaType,
                variants = _a.variants,
                selectedVariant = _a.selectedVariant,
                displayName = _a.displayName;
            var schema = '';
            try {
                schema = JSON.parse(resolve_sd_schema_1.resolveSdSchema(tagSchema.getValue(tag)));
                if (!Object.keys(schema).length) {
                    throw new Error('Error parsing structured data schema');
                }
            } catch (error) {
                logError({
                    error: error,
                    data: {
                        value: tag
                    }
                });
                schema = getDefaultSchema(defaultSchema === null || defaultSchema === void 0 ? void 0 : defaultSchema.tags, selectedVariant) || '{}';
            }
            return withMeta ?
                __assign(__assign(__assign(__assign({
                    schema: schema,
                    schemaType: schemaType
                }, (variants ? {
                    variants: variants
                } : {})), (selectedVariant ? {
                    selectedVariant: selectedVariant
                } : {})), (displayName ? {
                    displayName: displayName
                } : {})), {
                    disabled: isDisabled(tag.disabled)
                }) : schema;
        });
}
exports.getSchemasArray = getSchemasArray;

function getDefaultSchema(defaultTags, selectedVariant) {
    var _a, _b, _c;
    if (!defaultTags) {
        return '';
    }
    if (!selectedVariant) {
        return getValueByIdentifier(defaultTags, Identifiers_1.IDENTIFIERS.STRUCTURED_DATA);
    }
    var defaultTag = findTag(defaultTags, Identifiers_1.IDENTIFIERS.STRUCTURED_DATA);
    return (((_c = (_b = (_a = defaultTag === null || defaultTag === void 0 ? void 0 : defaultTag.meta) === null || _a === void 0 ? void 0 : _a.variants) === null || _b === void 0 ? void 0 : _b.find(function(variant) {
        return variant.schemaType === selectedVariant;
    })) === null || _c === void 0 ? void 0 : _c.schema) || '');
}

function findTag(tags, identifier) {
    if (!tags || !identifier) {
        return null;
    }
    var tag = getTagByIdentifier(tags, identifier);
    if (is_empty_1.isEmpty(tag)) {
        return undefined;
    }
    return tag;
}

function getTagsByIdentifier(tags, identifier) {
    var tagSchema = TagSchemas_1.TAG_SCHEMAS[identifier];
    if (is_empty_1.isEmpty(tagSchema) || is_empty_1.isEmpty(tags) || !Array.isArray(tags)) {
        return [];
    }
    return tags.filter(function(tag) {
        return validate_tag_structure_1.validateTagStructure(tag, tagSchema);
    });
}
exports.getTagsByIdentifier = getTagsByIdentifier;

function getTagByIdentifier(tags, identifier) {
    var tagSchema = TagSchemas_1.TAG_SCHEMAS[identifier];
    if (is_empty_1.isEmpty(tagSchema) || is_empty_1.isEmpty(tags) || !Array.isArray(tags)) {
        return {};
    }
    return tags
        .slice()
        .reverse()
        .find(function(tag) {
            return validate_tag_structure_1.validateTagStructure(tag, tagSchema);
        });
}

function getValueFromTagUsingContext(tag, identifier, context, tags, referencePath) {
    var tagSchema = identifier && TagSchemas_1.TAG_SCHEMAS[identifier];
    if (is_empty_1.isEmpty(tag) || is_empty_1.isEmpty(tagSchema)) {
        return '';
    }
    var value = tagSchema.getValue(tag);
    if (value.search(/{{.+?}}/g) !== -1) {
        return fill_in_pattern_blob_1.fillPatternValue(value, context, tags, referencePath);
    } else {
        return value === 'undefined' ? undefined : value;
    }
}

function getValueFromTag(tag, identifier) {
    var tagSchema = identifier && TagSchemas_1.TAG_SCHEMAS[identifier];
    if (is_empty_1.isEmpty(tag) || is_empty_1.isEmpty(tagSchema)) {
        return '';
    }
    return tagSchema.getValue(tag);
}

function isDisabled(value) {
    if (!value || value === 'false' || Array.isArray(value)) {
        return false;
    }
    if (typeof value === 'string') {
        var isVariable = [consts_1.PATTERN_PREFIX, consts_1.PATTERN_SUFFIX].every(function(affix) {
            return value === null || value === void 0 ? void 0 : value.includes(affix);
        });
        return !isVariable;
    }
    return Boolean(value);
}
