// @wix/bookings-uou-mappers/dist/esm/service/service-pricing-plans.mapper.js
import {
    PricingPlanStatus,
    ScheduleStatus,
} from '@wix/ambassador-services-catalog-server/types';
export function mapServicePricingPlansDto(serviceResponse, options) {
    var _a;
    if (options === void 0) {
        options = {
            onlyActive: false,
        };
    }
    var schedule = serviceResponse.schedules.find(function(serviceSchedule) {
        return serviceSchedule.status === ScheduleStatus.CREATED;
    });
    var pricingPlans = serviceResponse.pricingPlans || [];
    if (options.onlyActive) {
        pricingPlans = pricingPlans.filter(function(pricingPlan) {
            return pricingPlan.status === PricingPlanStatus.ACTIVE;
        });
    }
    var pricingPlansDto = pricingPlans.map(function(pricingPlan) {
        return ({
            id: pricingPlan.id,
            name: pricingPlan.name,
            visible: pricingPlan.visible,
        });
    });
    return {
        displayText: (_a = schedule.rate) === null || _a === void 0 ? void 0 : _a.priceText,
        pricingPlans: pricingPlansDto,
    };
}
