// @wix/advanced-seo-utils/dist/src/private/item-types/search-page/adapter-search-page.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.IDs = void 0;
var utils_1 = require("../../adapters/utils");
var get_1 = require("../../utils/get");
exports.IDs = {
    SEARCH_TERM: 'search-page.search-term',
    ALL_RESULTS_URL: 'search-page.all-results-url',
    PAGE_URL: 'search-page.page-url',
    RESULTS_TOTAL: 'search-page.results-total',
};
var SchemaTypes = {};
var userVisible = [exports.IDs.SEARCH_TERM, exports.IDs.RESULTS_TOTAL];
exports.default = {
    IDs: exports.IDs,
    SchemaTypes: SchemaTypes,
    getData: function(item) {
        var _a;
        return _a = {},
            _a[exports.IDs.SEARCH_TERM] = get_1.get(item, 'searchTerm'),
            _a[exports.IDs.ALL_RESULTS_URL] = get_1.get(item, 'allResultsUrl'),
            _a[exports.IDs.PAGE_URL] = get_1.get(item, 'pageUrl'),
            _a[exports.IDs.RESULTS_TOTAL] = get_1.get(item, 'resultsTotal'),
            _a;
    },
    getSdStatus: function() {},
    getLegacySeoBlob: function() {},
    getSlug: function() {
        return '';
    },
    updateItemDataWithSlug: function(item) {
        return item;
    },
    getBiData: function() {},
    getKeys: function() {
        return utils_1.getKeys(exports.IDs, userVisible);
    },
    getSdKeys: function() {
        return [];
    },
};
