// @wix/wixstores-client-storefront-sdk/dist/es/src/viewer-script/wixcode/product/openQuickView.js
import {
    __awaiter,
    __generator
} from "tslib";
import {
    getCartIconControllerConfig
} from '../getCartIconControllerConfig';
import {
    QuickViewActions
} from '../../../actions/QuickViewActions/QuickViewActions';
import {
    ProductsApi
} from '../../../apis/ProductsApi/ProductsApi';
export var openQuickView = function(_a) {
    var context = _a.context,
        origin = _a.origin;
    return function(productId, options) {
        return __awaiter(void 0, void 0, void 0, function() {
            var getProductsApi, cartIconControllerConfig, compId, config, product;
            return __generator(this, function(_a) {
                switch (_a.label) {
                    case 0:
                        getProductsApi = function(siteStore) {
                            return new ProductsApi({
                                siteStore: siteStore,
                                origin: origin
                            });
                        };
                        cartIconControllerConfig = getCartIconControllerConfig(context);
                        if (context.siteStore.isMobile()) {
                            throw Error('Cannot open quick-view in mobile');
                        }
                        if (!cartIconControllerConfig) {
                            throw Error('Cannot use openQuickView() without cart-icon on site');
                        }
                        compId = cartIconControllerConfig.compId, config = cartIconControllerConfig.config;
                        return [4 /*yield*/ , getProductsApi(context.siteStore).fetchProductSlug(productId)];
                    case 1:
                        product = _a.sent();
                        if (!product) {
                            throw Error('Product with provided productId does not exist');
                        }
                        return [2 /*return*/ , new QuickViewActions(context.siteStore).quickViewProduct({
                            origin: origin,
                            urlPart: product.urlPart,
                            compId: compId,
                            externalId: config.externalId,
                            selectionIds: [],
                            quantity: options === null || options === void 0 ? void 0 : options.quantity,
                        })];
                }
            });
        });
    };
};
