// @wix/ambassador-calendar-server/types.js
/* eslint-disable */
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Transparency = exports.StructNullValue = exports.Status = exports.SortOrder = exports.SessionType = exports.LocationsLocationType = exports.LocationStatus = exports.LocationLocationType = exports.DayOfWeek = exports.ConferenceType = exports.ApprovalStatus = void 0;
var ApprovalStatus;
(function(ApprovalStatus) {
    ApprovalStatus["UNDEFINED"] = "UNDEFINED";
    ApprovalStatus["PENDING"] = "PENDING";
    ApprovalStatus["APPROVED"] = "APPROVED";
    ApprovalStatus["DECLINED"] = "DECLINED";
})(ApprovalStatus = exports.ApprovalStatus || (exports.ApprovalStatus = {}));
var ConferenceType;
(function(ConferenceType) {
    ConferenceType["UNDEFINED"] = "UNDEFINED";
    ConferenceType["ONLINE_MEETING_PROVIDER"] = "ONLINE_MEETING_PROVIDER";
    ConferenceType["CUSTOM"] = "CUSTOM";
})(ConferenceType = exports.ConferenceType || (exports.ConferenceType = {}));
var DayOfWeek;
(function(DayOfWeek) {
    DayOfWeek["MONDAY"] = "MONDAY";
    DayOfWeek["TUESDAY"] = "TUESDAY";
    DayOfWeek["WEDNESDAY"] = "WEDNESDAY";
    DayOfWeek["THURSDAY"] = "THURSDAY";
    DayOfWeek["FRIDAY"] = "FRIDAY";
    DayOfWeek["SATURDAY"] = "SATURDAY";
    DayOfWeek["SUNDAY"] = "SUNDAY";
})(DayOfWeek = exports.DayOfWeek || (exports.DayOfWeek = {}));
var LocationLocationType;
(function(LocationLocationType) {
    LocationLocationType["UNDEFINED"] = "UNDEFINED";
    LocationLocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationLocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationLocationType["CUSTOM"] = "CUSTOM";
})(LocationLocationType = exports.LocationLocationType || (exports.LocationLocationType = {}));
var LocationStatus;
(function(LocationStatus) {
    LocationStatus["ACTIVE"] = "ACTIVE";
    LocationStatus["IN_ACTIVE"] = "IN_ACTIVE";
})(LocationStatus = exports.LocationStatus || (exports.LocationStatus = {}));
var LocationsLocationType;
(function(LocationsLocationType) {
    LocationsLocationType["UNKNOWN"] = "UNKNOWN";
    LocationsLocationType["BRANCH"] = "BRANCH";
    LocationsLocationType["OFFICES"] = "OFFICES";
    LocationsLocationType["RECEPTION"] = "RECEPTION";
    LocationsLocationType["HEADQUARTERS"] = "HEADQUARTERS";
})(LocationsLocationType = exports.LocationsLocationType || (exports.LocationsLocationType = {}));
var SessionType;
(function(SessionType) {
    SessionType["UNDEFINED"] = "UNDEFINED";
    SessionType["EVENT"] = "EVENT";
    SessionType["WORKING_HOURS"] = "WORKING_HOURS";
    SessionType["TIME_AVAILABILITY"] = "TIME_AVAILABILITY";
    SessionType["AVAILABILITY"] = "AVAILABILITY";
})(SessionType = exports.SessionType || (exports.SessionType = {}));
var SortOrder;
(function(SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder = exports.SortOrder || (exports.SortOrder = {}));
var Status;
(function(Status) {
    Status["UNDEFINED"] = "UNDEFINED";
    Status["CONFIRMED"] = "CONFIRMED";
    Status["CANCELLED"] = "CANCELLED";
})(Status = exports.Status || (exports.Status = {}));
var StructNullValue;
(function(StructNullValue) {
    StructNullValue["NULL_VALUE"] = "NULL_VALUE";
})(StructNullValue = exports.StructNullValue || (exports.StructNullValue = {}));
var Transparency;
(function(Transparency) {
    Transparency["UNDEFINED"] = "UNDEFINED";
    Transparency["FREE"] = "FREE";
    Transparency["BUSY"] = "BUSY";
})(Transparency = exports.Transparency || (exports.Transparency = {}));
