// @wix/advanced-seo-utils/dist/src/private/adapters/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
var _a;
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.componentAdapterHandlers = exports.optionalHandlers = exports.adaptersHandlers = exports.getAdapter = exports.adapters = void 0;
var ItemTypes_1 = require("../types/ItemTypes");
var utils_1 = require("./utils");
var adapter_default_1 = __importDefault(require("../item-types/default/adapter-default"));
var adapter_stores_product_1 = __importDefault(require("../item-types/stores-product/adapter-stores-product"));
var adapter_static_page_1 = __importDefault(require("../item-types/static-page/adapter-static-page"));
var adapter_static_page_v2_1 = __importDefault(require("../item-types/static-page-v2/adapter-static-page-v2"));
var adapter_forum_category_1 = __importDefault(require("../item-types/forum-category/adapter-forum-category"));
var adapter_forum_post_1 = __importDefault(require("../item-types/forum-post/adapter-forum-post"));
var adapter_pro_gallery_item_1 = __importDefault(require("../item-types/pro-gallery-item/adapter-pro-gallery-item"));
var adapter_blog_category_1 = __importDefault(require("../item-types/blog-category/adapter-blog-category"));
var adapter_blog_post_1 = __importDefault(require("../item-types/blog-post/adapter-blog-post"));
var adapter_blog_tags_1 = __importDefault(require("../item-types/blog-tags/adapter-blog-tags"));
var adapter_blog_archive_1 = __importDefault(require("../item-types/blog-archive/adapter-blog-archive"));
var adapter_groups_post_1 = __importDefault(require("../item-types/groups-post/adapter-groups-post"));
var adapter_groups_page_1 = __importDefault(require("../item-types/groups-page/adapter-groups-page"));
var adapter_events_page_1 = __importDefault(require("../item-types/events-page/adapter-events-page"));
var adapter_challenges_page_1 = __importDefault(require("../item-types/challenges-page/adapter-challenges-page"));
var adapter_bookings_service_1 = __importDefault(require("../item-types/bookings-service/adapter-bookings-service"));
var adapter_breadcrumbs_component_1 = __importDefault(require("../item-types/breadcrumbs-component/adapter-breadcrumbs-component"));
var adapter_search_page_1 = __importDefault(require("../item-types/search-page/adapter-search-page"));
var adapter_blog_hashtags_1 = __importDefault(require("../item-types/blog-hashtags/adapter-blog-hashtags"));
var adapter_bookings_calendar_1 = __importDefault(require("../item-types/bookings-calendar/adapter-bookings-calendar"));
var adapter_restaurants_order_page_1 = __importDefault(require("../item-types/restaurants-order-page/adapter-restaurants-order-page"));
var adapter_bookings_form_1 = __importDefault(require("../item-types/bookings-form/adapter-bookings-form"));
exports.adapters = (_a = {},
    _a[ItemTypes_1.ITEM_TYPES.STORES_PRODUCT] = adapter_stores_product_1.default,
    _a[ItemTypes_1.ITEM_TYPES.STATIC_PAGE] = adapter_static_page_1.default,
    _a[ItemTypes_1.ITEM_TYPES.STATIC_PAGE_V2] = adapter_static_page_v2_1.default,
    _a[ItemTypes_1.ITEM_TYPES.FORUM_CATEGORY] = adapter_forum_category_1.default,
    _a[ItemTypes_1.ITEM_TYPES.FORUM_POST] = adapter_forum_post_1.default,
    _a[ItemTypes_1.ITEM_TYPES.PRO_GALLERY_ITEM] = adapter_pro_gallery_item_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BLOG_CATEGORY] = adapter_blog_category_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BLOG_POST] = adapter_blog_post_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BLOG_TAGS] = adapter_blog_tags_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BLOG_ARCHIVE] = adapter_blog_archive_1.default,
    _a[ItemTypes_1.ITEM_TYPES.GROUPS_POST] = adapter_groups_post_1.default,
    _a[ItemTypes_1.ITEM_TYPES.GROUPS_PAGE] = adapter_groups_page_1.default,
    _a[ItemTypes_1.ITEM_TYPES.EVENTS_PAGE] = adapter_events_page_1.default,
    _a[ItemTypes_1.ITEM_TYPES.CHALLENGES_PAGE] = adapter_challenges_page_1.default,
    _a[ItemTypes_1.ITEM_TYPES.SEARCH_PAGE] = adapter_search_page_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BOOKINGS_SERVICE] = adapter_bookings_service_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BREADCRUMBS_COMPONENT] = adapter_breadcrumbs_component_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BLOG_HASHTAGS] = adapter_blog_hashtags_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BOOKINGS_CALENDAR] = adapter_bookings_calendar_1.default,
    _a[ItemTypes_1.ITEM_TYPES.BOOKINGS_FORM] = adapter_bookings_form_1.default,
    _a[ItemTypes_1.ITEM_TYPES.RESTAURANTS_ORDER_PAGE] = adapter_restaurants_order_page_1.default,
    _a);
/**
 *
 * @param {keyof typeof ITEM_TYPES} itemType
 * @returns {import('../../../index').Adapter}
 */
function getAdapter(itemType) {
    var adapter = exports.adapters[itemType];
    return adapter ? utils_1.enrichWithSiteLevelData(adapter) : adapter_default_1.default;
}
exports.getAdapter = getAdapter;
exports.adaptersHandlers = {
    getData: 'function',
    getSdStatus: 'function',
    getLegacySeoBlob: 'function',
    getBiData: 'function',
    getKeys: 'function',
    getSlug: 'function',
    updateItemDataWithSlug: 'function',
    getMissingSchemaVariables: 'function',
    getDynamicPatternsSchema: 'function',
    hasMultilingual: 'function',
};
exports.optionalHandlers = [
    'getDynamicPatternsSchema',
    'hasMultilingual',
    'getSdKeys',
];
exports.componentAdapterHandlers = {
    getData: 'function',
    getLegacySeoBlob: 'function',
    getBiData: 'function',
    getKeys: 'function',
    getMissingSchemaVariables: 'function',
};
