// @wix/advanced-seo-utils/dist/src/private/item-types/forum-category/adapter-forum-category.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
var get_1 = require("../../utils/get");
var ContextProps_1 = require("../../types/ContextProps");
var utils_1 = require("../../adapters/utils");
var consts_1 = require("../../consts");
var IDs = {
    SITE_NAME: 'site.name',
    INDEX_SITE: 'site.index',
    INDEX_PAGE: 'page.index',
    NAME: 'category.name',
    DESCRIPTION: 'category.description',
    IMAGE: 'category.image',
    IMAGE_WIDTH: 'category.image.width',
    IMAGE_HEIGHT: 'category.image.height',
    CATEGORY_PAGE_URL: 'category.url',
    POST_AMOUNT: 'category.posts.amount',
    VIEW_AMOUNT: 'category.views.amount',
};
var userVisible = [
    IDs.NAME,
    IDs.POST_AMOUNT,
    IDs.VIEW_AMOUNT,
    IDs.DESCRIPTION,
    IDs.CATEGORY_PAGE_URL,
    IDs.IMAGE,
    IDs.SITE_NAME,
];
exports.default = {
    IDs: IDs,
    getData: function(item) {
        var _a;
        return _a = {},
            _a[IDs.SITE_NAME] = get_1.get(item, "context." + ContextProps_1.CONTEXT_PROPS.SITE_NAME),
            _a[IDs.INDEX_SITE] = get_1.getRawValue(item, "context." + ContextProps_1.CONTEXT_PROPS.INDEX_SITE),
            _a[IDs.INDEX_PAGE] = get_1.getRawValue(item, "context." + ContextProps_1.CONTEXT_PROPS.INDEX_PAGE),
            _a[IDs.NAME] = get_1.get(item, 'category.title'),
            _a[IDs.DESCRIPTION] = get_1.get(item, 'category.description'),
            _a[IDs.IMAGE] = get_1.get(item, 'category.images.0.src'),
            _a[IDs.IMAGE_WIDTH] = get_1.get(item, 'category.images.0.width'),
            _a[IDs.IMAGE_HEIGHT] = get_1.get(item, 'category.images.0.height'),
            _a[IDs.CATEGORY_PAGE_URL] = get_1.get(item, 'category.refersTo'),
            _a[IDs.POST_AMOUNT] = get_1.get(item, 'category.totalPosts'),
            _a[IDs.VIEW_AMOUNT] = get_1.get(item, 'category.totalViews'),
            _a;
    },
    getSdStatus: function() {
        return consts_1.SD_STATUS.DOES_NOT_EXIST;
    },
    getLegacySeoBlob: function() {},
    getSlug: function() {
        return '';
    },
    updateItemDataWithSlug: function(item) {
        return item;
    },
    getBiData: function() {},
    getKeys: function() {
        return utils_1.getKeys(IDs, userVisible);
    },
    getSdKeys: function() {
        return [];
    },
};
