// message-channel/dist/src/connect/send-factory.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = sendFactory;

var _utils = require('../utils');

var _v = require('uuid/v4');

var _v2 = _interopRequireDefault(_v);

function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}

function sendFactory(port, options) {
    var messageMaxTimeout = options.messageMaxTimeout;

    var messages = {};

    port.onmessage = function(e) {
        var _parseChannelMessage = (0, _utils.parseChannelMessage)(e.data),
            id = _parseChannelMessage.id,
            payload = _parseChannelMessage.payload;

        if (messages[id]) {
            var modifiedEvent = {
                data: payload,
                origin: e.origin,
                lastEventId: e.lastEventId,
                source: e.source,
                ports: e.ports
            };

            messages[id](modifiedEvent);
        }
    };

    return function(message, transferList) {
        return new Promise(function(resolve, reject) {
            setTimeout(function() {
                reject(new Error('max timeout of ' + messageMaxTimeout + 'ms exceeded'));
            }, messageMaxTimeout);

            var messageId = (0, _v2.default)();
            var packaged = (0, _utils.constructChannelMessage)(message, messageId);
            messages[messageId] = resolve;
            return port.postMessage(packaged, transferList);
        });
    };
}
