// @wix/chat-sdk/lib/src/constants/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.STAR_LABEL_ID = exports.DEFAULT_THROTTLE_THRESHOLD = exports.FetchRoomMessagesPageSize = exports.FetchChatroomsPageSizeDefaultValueInt = exports.FetchChatroomsPageSizeDefaultValue = exports.IndicationEvent = exports.HistorySinceRealtimeConnectionWasEstablished = exports.MessagePositionInGroup = exports.MediaProviders = exports.experiments = exports.MessageOrientation = exports.MessageSendStatus = exports.MessageDirections = exports.MessageTypes = exports.MimeTypes = void 0;
var MimeTypes = {
    Image: 'uri/image',
    PlainText: 'text/plain',
    File: 'uri/file',
};
exports.MimeTypes = MimeTypes;
var MessageTypes = {
    Message: 'message',
    Templates: {
        Standard: 'template/standard-event',
        Form: 'template/form',
        Minimal: 'template/minimal',
        Custom: 'template/custom',
        System: 'template/system',
    },
};
exports.MessageTypes = MessageTypes;
var MessageDirections = {
    Outgoing: 'right',
    Incoming: 'left',
    System: 'center',
};
exports.MessageDirections = MessageDirections;
var MessageOrientation = {
    Portrait: 'portrait',
    Landscape: 'landscape',
};
exports.MessageOrientation = MessageOrientation;
var MessageSendStatus;
(function(MessageSendStatus) {
    MessageSendStatus["SENT"] = "SENT";
    MessageSendStatus["SENDING"] = "SENDING";
    MessageSendStatus["FAILED"] = "FAILED";
})(MessageSendStatus || (MessageSendStatus = {}));
exports.MessageSendStatus = MessageSendStatus;
var experiments = {
    CrmmChatRenderOfflineActivity: 'specs.woa.CrmmChatRenderOfflineActivity',
    CrmmChatShowCartIndicator: 'specs.woa.CrmmChatShowCartIndicator',
    UnseenMessagesButton: 'specs.chat.UnseenMessagesButton',
    AvailabilityHours: 'specs.chat.AvailabilityHours',
    LazyLoadEmojiPanel: 'specs.chat.LazyLoadEmojiPanel',
    UxEnhacements: 'specs.chat.UxEnhacements',
    ShowCTAButtonForOldDataFormat: 'specs.com.ShowCTAButtonForOldDataFormat',
    ChatDebugMeasureTimeToPush: 'specs.woa.ChatDebugMeasureTimeToPush',
    EnrichedChatroom: 'specs.chat.ChatroomParticipants',
    DrillInMessageIndications: 'specs.chat.DrillInMessageIndications',
    NoHistoryOnRealtimeConnect: 'specs.chat.NoHistoryOnRealtimeConnect',
    InboxErrorIndications: 'specs.chat.InboxErrorIndications',
    FetchChatroomsPageSize: 'specs.chat.FetchChatroomsPageSize',
    FirebaseAtomicServerTime: 'specs.chat.FirebaseAtomicServerTime',
    DragNDropUpload: 'specs.chat.DragNDropUpload',
    DragNDropUploadInbox: 'specs.chat.DragNDropUploadInbox',
    FilterByUnread: 'specs.chat.SdkFilterByUnread',
    InboxAllowStarConversations: 'specs.chat.InboxAllowStarConversations',
};
exports.experiments = experiments;
var MediaProviders = {
    WIX_MEDIA: 'WixMedia',
    WIX_MEDIA_PUBLIC: 'WixMediaPublic',
};
exports.MediaProviders = MediaProviders;
var MessagePositionInGroup = {
    START: 'start',
    MIDDLE: 'middle',
    END: 'end',
    SINGLE: 'single',
};
exports.MessagePositionInGroup = MessagePositionInGroup;
var HistorySinceRealtimeConnectionWasEstablished = 30000;
exports.HistorySinceRealtimeConnectionWasEstablished = HistorySinceRealtimeConnectionWasEstablished;
var FetchChatroomsPageSizeDefaultValue = '30';
exports.FetchChatroomsPageSizeDefaultValue = FetchChatroomsPageSizeDefaultValue;
var FetchChatroomsPageSizeDefaultValueInt = parseInt(FetchChatroomsPageSizeDefaultValue);
exports.FetchChatroomsPageSizeDefaultValueInt = FetchChatroomsPageSizeDefaultValueInt;
var FetchRoomMessagesPageSize = 50;
exports.FetchRoomMessagesPageSize = FetchRoomMessagesPageSize;
var IndicationEvent;
(function(IndicationEvent) {
    IndicationEvent["SENT"] = "Sent";
    IndicationEvent["SEEN"] = "Seen";
    IndicationEvent["FAILED"] = "Failed";
})(IndicationEvent || (IndicationEvent = {}));
exports.IndicationEvent = IndicationEvent;
var DEFAULT_THROTTLE_THRESHOLD = 5 * 1000;
exports.DEFAULT_THROTTLE_THRESHOLD = DEFAULT_THROTTLE_THRESHOLD;
var STAR_LABEL_ID = '7553d8c6-0ba2-424e-8e81-d1f7ea14bfe4';
exports.STAR_LABEL_ID = STAR_LABEL_ID;
