// @wix/wixstores-client-storefront-sdk/dist/es/src/viewer-script/wixcode/product/wixcode.product.js
import {
    getDefaultProduct
} from './getDefaultProduct';
import {
    openQuickView
} from './openQuickView';
import {
    getExtendedProductBySlug
} from './getExtendedProductBySlug';
import {
    getAbsoluteProductUrl
} from './getAbsoluteProductUrl';
import {
    getRelativeProductUrl
} from './getRelativeProductUrl';
import {
    getContextualProducts
} from './getContextualProducts';
export function createProductExports(_a) {
    var context = _a.context,
        origin = _a.origin;
    var getAbsoluteProductUrlFn = getAbsoluteProductUrl({
        context: context,
        origin: origin
    });
    var getRelativeProductUrlFn = getRelativeProductUrl({
        context: context,
        origin: origin
    });
    return {
        openQuickView: openQuickView({
            context: context,
            origin: origin
        }),
        getDefaultProduct: getDefaultProduct({
            context: context,
            origin: origin
        }),
        getExtendedProductBySlug: getExtendedProductBySlug({
            context: context,
            origin: origin
        }),
        getAbsoluteProductUrl: getAbsoluteProductUrlFn,
        getRelativeProductUrl: getRelativeProductUrlFn,
        getContextualProducts: getContextualProducts({
            context: context,
            getAbsoluteProductUrl: getAbsoluteProductUrlFn,
            getRelativeProductUrl: getRelativeProductUrlFn,
        }),
    };
}
