// @wix/wixstores-client-storefront-sdk/dist/es/src/viewer-script/wixcode/createWixcodeExports.js
import {
    createCartExports
} from './cart/wixcode.cart';
import {
    createCheckoutExports
} from './checkout/wixcode.checkout';
import {
    createNavigateExports
} from './navigate/wixcode.navigate';
import {
    createProductExports
} from './product/wixcode.product';
import {
    createBackInStockExports
} from './backInStock/wixcode.backInStock';
export var createWixcodeExports = function(_a) {
    var context = _a.context,
        origin = _a.origin;
    return ({
        cart: createCartExports({
            context: context,
            origin: origin
        }),
        navigate: createNavigateExports({
            context: context,
            origin: origin
        }),
        product: createProductExports({
            context: context,
            origin: origin
        }),
        checkout: createCheckoutExports({
            context: context,
            origin: origin
        }),
        backInStock: createBackInStockExports({
            context: context,
            origin: origin
        }),
    });
};
