// @wix/chat-web/dist/es/src/services/media-utils.js
import {
    attachments
} from '@wix/inbox-common';
var ATTACHMENT_FILE_LIMIT = 1024 * 1024 * 15;
export function isFileTooLarge(file) {
    return file.size > ATTACHMENT_FILE_LIMIT;
}
export function areFilesTooLarge(files) {
    return (files.reduce(function(acc, file) {
        acc += file.size;
        return acc;
    }, 0) > ATTACHMENT_FILE_LIMIT);
}
export function getFileSecureUrl(url) {
    return url.replace('http://', 'https://');
}
export var formattedFileSize = function(fileSize) {
    var kb = 1024;
    var mb = kb * 1024;
    if (fileSize >= mb) {
        return (fileSize / mb).toFixed(2) + " MB";
    }
    return (fileSize / kb).toFixed(2) + " KB";
};
export var allAttachmentTypes = [
    attachments.attachmentTypes.IMAGE,
    attachments.attachmentTypes.DOC,
    attachments.attachmentTypes.PPT,
    attachments.attachmentTypes.XLS,
    attachments.attachmentTypes.PDF,
];
export function toDialogFileType(attachmentType) {
    switch (attachmentType) {
        case attachments.attachmentTypes.IMAGE:
            return 'image/*';
        case attachments.attachmentTypes.XLS:
            return '.xlsx,.xls';
        case attachments.attachmentTypes.DOC:
            return '.doc,.docx';
        case attachments.attachmentTypes.PPT:
            return '.ppt,.pptx';
        case attachments.attachmentTypes.PDF:
            return '.pdf';
        default:
            return '';
    }
}
export function toDialogFileTypes(attachmentTypes) {
    return attachmentTypes.map(function(type) {
        return toDialogFileType(type);
    }).join(',');
}
export function isFileTypeSupported(file, attachmentTypes) {
    return attachmentTypes.indexOf(attachments.getAttachmentType(file.type)) > -1;
}
