// @wix/thunderbolt-commons/dist/hoverBoxUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.compIdToCssSelector = exports.isDefaultCompId = exports.getDefaultCompId = exports.getHoverCompId = exports.HOVER_DATA_MODE = exports.DEFAULT_DATA_MODE = exports.HOVER_SUFFIX = exports.MODES_TYPE = void 0;
exports.MODES_TYPE = {
    hover: 'HOVER',
    default: 'DEFAULT',
};
exports.HOVER_SUFFIX = ':hover';
exports.DEFAULT_DATA_MODE = '[data-mode=default]';
exports.HOVER_DATA_MODE = '[data-mode=hover]';
var getHoverCompId = function(compId) {
    return "" + compId + exports.HOVER_SUFFIX;
};
exports.getHoverCompId = getHoverCompId;
var getDefaultCompId = function(compId) {
    return (compId ? compId.replace(exports.HOVER_SUFFIX, '') : compId);
};
exports.getDefaultCompId = getDefaultCompId;
var isDefaultCompId = function(compId) {
    return !compId.endsWith(exports.HOVER_SUFFIX);
};
exports.isDefaultCompId = isDefaultCompId;
var COMP_SELECTOR_PLACEHOLDER = '#<%= compId %>';
var compIdToCssSelector = function(compId, shouldReturnPlaceholderSelector) {
    var defaultCompId = (0, exports.getDefaultCompId)(compId);
    var fallbackSelector = shouldReturnPlaceholderSelector ? "" + COMP_SELECTOR_PLACEHOLDER : "#" + compId;
    return compId.endsWith(exports.HOVER_SUFFIX) ? exports.HOVER_DATA_MODE + " #" + defaultCompId : fallbackSelector;
};
exports.compIdToCssSelector = compIdToCssSelector;
