// @wix/advanced-seo-utils/dist/src/private/item-types/blog-tags/pattern-blog-tags.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.BLOG_TAGS = void 0;
var builders_1 = require("../../types/builders");
var adapter_blog_tags_1 = __importDefault(require("./adapter-blog-tags"));
var utils_1 = require("../../adapters/utils");
var utils_2 = require("../utils");
var LinkTagTypes_1 = require("../../types/LinkTagTypes");
var IDs = adapter_blog_tags_1.default.IDs;
var SITE_NAME = utils_1.siteIDs.SITE_NAME;
var NAME = IDs.NAME,
    PAGE_URL = IDs.PAGE_URL,
    PREV_LINK = IDs.PREV_LINK,
    NEXT_LINK = IDs.NEXT_LINK,
    BLOG_FEED_ROUTE = IDs.BLOG_FEED_ROUTE;
exports.BLOG_TAGS = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + NAME + "}} | {{" + SITE_NAME + "}}"),
        builders_1.buildStandardMeta('robots', 'noindex')
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:site_name', "{{" + SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'website'),
        builders_1.buildOgMeta('og:url', "{{" + PAGE_URL + "}}"),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'prev',
            href: "{{" + PREV_LINK + "}}"
        }),
        builders_1.buildLink({
            rel: 'next',
            href: "{{" + NEXT_LINK + "}}"
        }),
        builders_1.buildLink({
            rel: 'alternate',
            type: LinkTagTypes_1.LINK_TAG_TYPES.RSS,
            href: "{{" + BLOG_FEED_ROUTE + "}}",
            title: utils_2.getRssTitle("{{" + SITE_NAME + "}}"),
        })
    ], builders_1.buildTwitterMetaRefs()),
};
