// @wix/chat-web/dist/es/src/components/attachments/AttachmentArea/AttachmentArea.js
import React from 'react';
import {
    Box
} from 'wix-style-react';
import {
    MimeTypes
} from '@wix/chat-sdk/lib/src/constants/constants';
import ImageAttachment from '../ImageAttachment/ImageAttachment';
import FileAttachment from '../FileAttachment/FileAttachment';
export var AttachmentArea = function(_a) {
    var attachments = _a.attachments,
        onRemoveAttachment = _a.onRemoveAttachment,
        instanceId = _a.instanceId;
    return (attachments === null || attachments === void 0 ? void 0 : attachments.length) ? (React.createElement(Box, {
            display: "block",
            width: "100%",
            dataHook: "attachment-area",
            overflow: "auto"
        },
        React.createElement(Box, {
            width: "100%",
            flexWrap: "wrap"
        }, attachments.map(function(attachment) {
            var _a;
            return (React.createElement(Box, {
                key: "attachment-" + attachment.id,
                position: "relative",
                verticalAlign: "bottom",
                marginRight: "6px"
            }, ((_a = attachment === null || attachment === void 0 ? void 0 : attachment.data) === null || _a === void 0 ? void 0 : _a[0].mimeType) === MimeTypes.Image ? (React.createElement(ImageAttachment, {
                messagePart: attachment.messageParts[0],
                transit: attachment.transit,
                instanceId: instanceId || 'no-instance',
                onRemove: function() {
                    return onRemoveAttachment(attachment);
                }
            })) : (React.createElement(FileAttachment, {
                messagePart: attachment.messageParts[0],
                transit: attachment.transit,
                instanceId: instanceId || 'no-instance',
                onRemove: function() {
                    return onRemoveAttachment(attachment);
                }
            }))));
        })))) : null;
};
