// @wix/advanced-seo-utils/dist/src/private/item-types/events-page/structured-data/get-json-offers.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getTicketPrice = exports.getTicketFinalPrice = exports.getJsonOffers = void 0;
var get_1 = require("../../../utils/get");
var get_event_url_1 = require("../calculated/get-event-url");
var get_stock_status_1 = require("../calculated/get-stock-status");
exports.getJsonOffers = function(item, ticketsLength) {
    if (item === void 0) {
        item = {};
    }
    if (ticketsLength === void 0) {
        ticketsLength = 0;
    }
    var tickets = item.tickets;
    if (!ticketsLength) {
        return '';
    }
    var offers = tickets.map(function(ticket) {
        var url = get_event_url_1.getEventUrl(item);
        var offer = {
            '@type': 'offer',
            name: get_1.get(ticket, 'name'),
            price: "" + exports.getTicketFinalPrice(ticket),
            priceCurrency: get_1.get(ticket, 'price.currency'),
            availability: "https://schema.org/" + get_stock_status_1.getStockStatus(ticket),
            validFrom: get_1.get(item, 'event.created'),
        };
        if (url) {
            offer.url = url;
        }
        return offer;
    });
    var json = {
        '@type': 'AggregateOffer',
        highPrice: get_1.get(item, 'event.registration.ticketing.highestTicketPrice.amount'),
        lowPrice: get_1.get(item, 'event.registration.ticketing.lowestTicketPrice.amount'),
        offerCount: "" + ticketsLength,
        priceCurrency: get_1.get(item, 'event.registration.ticketing.currency'),
        availability: tickets.some(function(ticket) {
                return get_stock_status_1.getStockStatus(ticket) === get_stock_status_1.STOCK_STATUS.IN_STOCK;
            }) ?
            "https://schema.org/" + get_stock_status_1.STOCK_STATUS.IN_STOCK :
            "https://schema.org/" + get_stock_status_1.STOCK_STATUS.SOLD_OUT,
        validFrom: get_1.get(item, 'event.created'),
        url: get_1.get(item, 'event.eventPageUrl.base') +
            get_1.get(item, 'event.eventPageUrl.path'),
    };
    if (offers.length) {
        json.offers = offers;
    }
    return json;
};
exports.getTicketFinalPrice = function(ticket) {
    return parseFloat(get_1.getRawValue(ticket, 'finalPrice.amount') || 0);
};
exports.getTicketPrice = function(ticket) {
    return parseFloat(get_1.getRawValue(ticket, 'price.amount') || 0);
};
