// wix-ui-tpa/dist/src/components/Tooltip/Tooltip.js
"use strict";
var __extends = (this && this.__extends) || (function() {
    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({
                    __proto__: []
                }
                instanceof Array && function(d, b) {
                    d.__proto__ = b;
                }) ||
            function(d, b) {
                for (var p in b)
                    if (b.hasOwnProperty(p)) d[p] = b[p];
            };
        return extendStatics(d, b);
    };
    return function(d, b) {
        extendStatics(d, b);

        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function() {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s)
                if (Object.prototype.hasOwnProperty.call(s, p))
                    t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function(s, e) {
    var t = {};
    for (var p in s)
        if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Tooltip = void 0;
var React = require("react");
var tooltip_1 = require("wix-ui-core/dist/src/components/tooltip");
var Arrows_1 = require("./Arrows");
var Tooltip_st_css_1 = require("./Tooltip.st.css");
var types_1 = require("./types");

function customArrow(placement, arrowProps) {
    var ArrowSvg = Arrows_1.getArrowByPlacement(placement);
    return (React.createElement("div", __assign({
            className: Tooltip_st_css_1.classes.tpaArrow
        }, arrowProps),
        React.createElement(ArrowSvg, null)));
}
var Tooltip = /** @class */ (function(_super) {
    __extends(Tooltip, _super);

    function Tooltip() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Tooltip.prototype.render = function() {
        var _a = this.props,
            className = _a.className,
            rest = __rest(_a, ["className"]);
        return (React.createElement(tooltip_1.Tooltip, __assign({
            className: Tooltip_st_css_1.st(Tooltip_st_css_1.classes.root, {
                skin: this.props.skin
            }, className),
            timeout: {
                enter: 120,
                exit: 80
            },
            customArrow: customArrow
        }, rest)));
    };
    Tooltip.displayName = 'Tooltip';
    Tooltip.defaultProps = {
        skin: types_1.TooltipSkin.Standard,
        maxWidth: 240,
    };
    return Tooltip;
}(React.Component));
exports.Tooltip = Tooltip;
