// @wix/advanced-seo-utils/dist/src/private/item-types/blog-post/pattern-blog-post.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.BLOG_POST = void 0;
var builders_1 = require("../../types/builders");
var adapter_blog_post_1 = __importDefault(require("./adapter-blog-post"));
var utils_1 = require("../../adapters/utils");
var blog_post_blog_posting_1 = __importDefault(require("./schema-presets/blog-post-blog-posting"));
var blog_post_news_article_1 = __importDefault(require("./schema-presets/blog-post-news-article"));
var blog_post_article_1 = __importDefault(require("./schema-presets/blog-post-article"));
var LinkTagTypes_1 = require("../../types/LinkTagTypes");
var utils_2 = require("../utils");
var IDs = adapter_blog_post_1.default.IDs;
exports.BLOG_POST = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.TITLE + "}}"),
        builders_1.buildStandardMeta('description', "{{" + IDs.EXCERPT + "}}")
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:url', "{{" + IDs.POST_PAGE_URL + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'article'),
        builders_1.buildOgMeta('og:image', "{{" + IDs.COVER_IMAGE + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.COVER_IMAGE_HEIGHT + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.COVER_IMAGE_WIDTH + "}}"),
        builders_1.buildOgMeta('article:author', "{{" + IDs.AUTHOR + "}}"),
        builders_1.buildOgMeta('article:published_time', "{{" + IDs.PUBLISH_DATE + "}}"),
        builders_1.buildOgMeta('article:modified_time', "{{" + IDs.LAST_MODIFIED_DATE + "}}"),
        builders_1.buildLink({
            rel: 'amphtml',
            href: "{{" + IDs.AMP_URL + "}}",
            disabled: "{{" + IDs.DISABLE_AMP + "}}",
        }),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.POST_PAGE_URL + "}}"
        }),
        builders_1.buildLink({
            rel: 'alternate',
            type: LinkTagTypes_1.LINK_TAG_TYPES.RSS,
            href: "{{" + IDs.BLOG_FEED_ROUTE + "}}",
            title: utils_2.getRssTitle("{{" + utils_1.siteIDs.SITE_NAME + "}}"),
        }),
        builders_1.buildStandardMeta('robots', "{{" + IDs.INDEX_POST + "}}")
    ], builders_1.buildTwitterMetaRefs(), [
        builders_1.buildStructuredData({
            schema: blog_post_blog_posting_1.default.schema,
            schemaType: blog_post_blog_posting_1.default.type,
            variants: [{
                    schema: blog_post_news_article_1.default.schema,
                    schemaType: blog_post_news_article_1.default.type,
                    disabled: [
                        "{{" + IDs.SD_DISABLE_SCHEMA + "}}",
                        '{{options.disable-pattern-schema}}',
                    ],
                },
                {
                    schema: blog_post_article_1.default.schema,
                    schemaType: blog_post_article_1.default.type,
                    disabled: [
                        "{{" + IDs.SD_DISABLE_SCHEMA + "}}",
                        '{{options.disable-pattern-schema}}',
                    ],
                },
            ],
            disabled: [
                "{{" + IDs.SD_DISABLE_SCHEMA + "}}",
                '{{options.disable-pattern-schema}}',
            ],
        }),
    ]),
};
