// @wix/advanced-seo-utils/dist/src/private/tags/defaults/get-by-identifier.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getDefaultValueByIdentifier = void 0;
var Identifiers_1 = require("../../types/Identifiers");
var ContextProps_1 = require("../../types/ContextProps");
var get_by_identifier_1 = require("../values/get-by-identifier");
var sanitize_url_1 = require("../../utils/sanitize-url");
var HANDLERS = (_a = {},
    _a[Identifiers_1.IDENTIFIERS.TITLE] = getDefaultTitle,
    _a[Identifiers_1.IDENTIFIERS.DESCRIPTION] = getDefaultDescription,
    _a[Identifiers_1.IDENTIFIERS.OG_TITLE] = getDefaultOgTitle,
    _a[Identifiers_1.IDENTIFIERS.OG_DESCRIPTION] = getDefaultOgDescription,
    _a[Identifiers_1.IDENTIFIERS.OG_IMAGE] = getDefaultOgImage,
    _a[Identifiers_1.IDENTIFIERS.FB_ADMINS] = getDefaultFbAdmins,
    _a[Identifiers_1.IDENTIFIERS.CANONICAL] = getDefaultCanonicalUrl,
    _a);
exports.getDefaultValueByIdentifier = function(tags, identifier, context) {
    return (HANDLERS[identifier] || getDefault)(context, tags);
};

function getDefault() {
    return undefined;
}

function getDefaultTitle(context) {
    if (context[ContextProps_1.CONTEXT_PROPS.TITLE]) {
        return context[ContextProps_1.CONTEXT_PROPS.TITLE];
    }
    if (context[ContextProps_1.CONTEXT_PROPS.IS_HOME_PAGE]) {
        return context[ContextProps_1.CONTEXT_PROPS.SITE_NAME] || '';
    }
    return [
            context[ContextProps_1.CONTEXT_PROPS.PAGE_NAME],
            context[ContextProps_1.CONTEXT_PROPS.HOME_PAGE_TITLE] || context[ContextProps_1.CONTEXT_PROPS.SITE_NAME],
        ]
        .filter(function(_) {
            return !!_;
        })
        .join(' | ');
}

function getDefaultDescription(context) {
    return context[ContextProps_1.CONTEXT_PROPS.DESCRIPTION];
}

function getDefaultOgTitle(context, tags) {
    return (get_by_identifier_1.getValueByIdentifier(tags, Identifiers_1.IDENTIFIERS.TITLE) ||
        getDefaultTitle(context, tags));
}

function getDefaultOgDescription(context, tags) {
    return (get_by_identifier_1.getValueByIdentifier(tags, Identifiers_1.IDENTIFIERS.DESCRIPTION) ||
        getDefaultDescription(context, tags));
}

function getDefaultOgImage(context) {
    return (context[ContextProps_1.CONTEXT_PROPS.OG_IMAGE] || context[ContextProps_1.CONTEXT_PROPS.SITE_OG_IMAGE]);
}

function getDefaultCanonicalUrl(context) {
    return sanitize_url_1.sanitizeUrl(context[ContextProps_1.CONTEXT_PROPS.DEFAULT_URL]);
}

function getDefaultFbAdmins(context) {
    return context[ContextProps_1.CONTEXT_PROPS.FB_ADMINS];
}
