// @wix/advanced-seo-utils/dist/src/private/item-types/stores-product/pattern-stores-product.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.STORES_PRODUCT = void 0;
var builders_1 = require("../../types/builders");
var adapter_stores_product_1 = __importDefault(require("./adapter-stores-product"));
var stores_product_page_1 = __importDefault(require("./schema-presets/stores-product-page"));
var IDs = adapter_stores_product_1.default.IDs;
exports.STORES_PRODUCT = {
    tags: __spreadArrays([
        builders_1.buildTitle("{{" + IDs.NAME + "}} | {{" + IDs.SITE_NAME + "}}"),
        builders_1.buildStandardMeta('description', "{{" + IDs.DESCRIPTION + "}}")
    ], builders_1.buildOgMetaRefs(), [
        builders_1.buildOgMeta('og:url', "{{" + IDs.PRODUCT_PAGE_URL + "}}"),
        builders_1.buildOgMeta('og:site_name', "{{" + IDs.SITE_NAME + "}}"),
        builders_1.buildOgMeta('og:type', 'product'),
        builders_1.buildStandardMeta('robots', 'index'),
        builders_1.buildLink({
            rel: 'canonical',
            href: "{{" + IDs.PRODUCT_PAGE_URL + "}}"
        }),
        builders_1.buildOgMeta('product:price:amount', "{{" + IDs.PRICE + "}}"),
        builders_1.buildOgMeta('product:price:currency', "{{" + IDs.CURRENCY + "}}"),
        builders_1.buildOgMeta('og:availability', "{{" + IDs.AVAILABILITY_META_TAG + "}}"),
        builders_1.buildOgMeta('og:image', "{{" + IDs.IMAGE_URL + "}}"),
        builders_1.buildOgMeta('og:image:width', "{{" + IDs.IMAGE_WIDTH + "}}"),
        builders_1.buildOgMeta('og:image:height', "{{" + IDs.IMAGE_HEIGHT + "}}")
    ], builders_1.buildTwitterMetaRefs(), [
        builders_1.buildStructuredData({
            schema: stores_product_page_1.default.schema,
            schemaType: stores_product_page_1.default.type,
            disabled: [
                "{{" + IDs.DISABLE_SCHEMA + "}}",
                '{{options.disable-pattern-schema}}',
            ],
        }),
    ]),
};
