// @wix/ambassador-gateway/types.js
/* eslint-disable */
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ValueType = exports.SelectedPaymentOption = exports.Platform = exports.PaymentStatus = exports.LocationType = exports.ConferenceType = exports.BookingStatus = exports.Actor = void 0;
var Actor;
(function(Actor) {
    Actor["UNDEFINED_ACTOR"] = "UNDEFINED_ACTOR";
    Actor["BUSINESS"] = "BUSINESS";
    Actor["CUSTOMER"] = "CUSTOMER";
})(Actor = exports.Actor || (exports.Actor = {}));
var BookingStatus;
(function(BookingStatus) {
    BookingStatus["CREATED"] = "CREATED";
    BookingStatus["PENDING_ECOM_CHECKOUT"] = "PENDING_ECOM_CHECKOUT";
    BookingStatus["CONFIRMED"] = "CONFIRMED";
    BookingStatus["CANCELED"] = "CANCELED";
    BookingStatus["PENDING"] = "PENDING";
    BookingStatus["DECLINED"] = "DECLINED";
    BookingStatus["WAITING_LIST"] = "WAITING_LIST";
})(BookingStatus = exports.BookingStatus || (exports.BookingStatus = {}));
var ConferenceType;
(function(ConferenceType) {
    ConferenceType["UNDEFINED"] = "UNDEFINED";
    ConferenceType["ONLINE_MEETING_PROVIDER"] = "ONLINE_MEETING_PROVIDER";
    ConferenceType["CUSTOM"] = "CUSTOM";
})(ConferenceType = exports.ConferenceType || (exports.ConferenceType = {}));
var LocationType;
(function(LocationType) {
    LocationType["UNDEFINED"] = "UNDEFINED";
    LocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationType["CUSTOM"] = "CUSTOM";
})(LocationType = exports.LocationType || (exports.LocationType = {}));
var PaymentStatus;
(function(PaymentStatus) {
    PaymentStatus["NOT_PAID"] = "NOT_PAID";
    PaymentStatus["PAID"] = "PAID";
    PaymentStatus["PARTIALLY_PAID"] = "PARTIALLY_PAID";
})(PaymentStatus = exports.PaymentStatus || (exports.PaymentStatus = {}));
var Platform;
(function(Platform) {
    Platform["UNDEFINED_PLATFORM"] = "UNDEFINED_PLATFORM";
    Platform["WEB"] = "WEB";
    Platform["MOBILE_APP"] = "MOBILE_APP";
})(Platform = exports.Platform || (exports.Platform = {}));
var SelectedPaymentOption;
(function(SelectedPaymentOption) {
    SelectedPaymentOption["UNDEFINED"] = "UNDEFINED";
    SelectedPaymentOption["OFFLINE"] = "OFFLINE";
    SelectedPaymentOption["ONLINE"] = "ONLINE";
    SelectedPaymentOption["CUSTOM"] = "CUSTOM";
    SelectedPaymentOption["WIX_PAID_PLAN"] = "WIX_PAID_PLAN";
})(SelectedPaymentOption = exports.SelectedPaymentOption || (exports.SelectedPaymentOption = {}));
var ValueType;
(function(ValueType) {
    ValueType["SHORT_TEXT"] = "SHORT_TEXT";
    ValueType["LONG_TEXT"] = "LONG_TEXT";
    ValueType["CHECK_BOX"] = "CHECK_BOX";
})(ValueType = exports.ValueType || (exports.ValueType = {}));
