// @wix/tpa-settings/dist/src/contexts/StylesContext/styles/createStyleParamGetter.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.createStyleParamGetter = exports.createStyleParamDefaultValueGetter = void 0;
var tslib_1 = require("tslib");
var types_1 = require("../../../types");
var generateKey_1 = require("../../../generateKey");
var paramShouldInheritDesktop_1 = require("../../../utils/paramShouldInheritDesktop");

function createStyleParamDefaultValueGetter(_a) {
    var styles = _a.styles,
        textPresets = _a.textPresets,
        colors = _a.colors,
        isMobile = _a.isMobile,
        isRTL = _a.isRTL;
    return function(param) {
        return param.getDefaultValue({
            colors: colors,
            textPresets: textPresets,
            isMobile: isMobile,
            isRTL: isRTL,
            getStyleParamValue: function(depParam, options) {
                if (options === void 0) {
                    options = {};
                }
                var getParam = createStyleParamGetter(tslib_1.__assign({
                    storage: styles,
                    colors: colors,
                    textPresets: textPresets,
                    isMobile: isMobile,
                    isRTL: isRTL,
                    useDefaultValue: false
                }, options));
                return getParam(depParam);
            },
        });
    };
}
exports.createStyleParamDefaultValueGetter = createStyleParamDefaultValueGetter;

function createStyleParamGetter(_a) {
    var styles = _a.storage,
        textPresets = _a.textPresets,
        colors = _a.colors,
        isMobile = _a.isMobile,
        isRTL = _a.isRTL,
        _b = _a.useDefaultValue,
        useDefaultValue = _b === void 0 ? true : _b;
    var getNumberParam = function(key) {
        return (styles.numbers || {})[key];
    };
    var getBooleanParam = function(key) {
        return (styles.booleans || {})[key];
    };
    var getFontParam = function(key) {
        return (styles.fonts || {})[key];
    };
    var getColorParam = function(key) {
        return (styles.colors || {})[key];
    };
    return function(param) {
        var _a;
        var getter = (_a = {},
            _a[types_1.StyleParamType.Number] = getNumberParam,
            _a[types_1.StyleParamType.Boolean] = getBooleanParam,
            _a[types_1.StyleParamType.Font] = getFontParam,
            _a[types_1.StyleParamType.Color] = getColorParam,
            _a)[param.type];
        var key = param.key || param.name;
        var generatedKey = generateKey_1.generateKey(tslib_1.__assign(tslib_1.__assign({}, param), {
            key: key
        }), {
            isMobile: isMobile
        });
        var value = getter(generatedKey);
        var shouldInheritDesktop = paramShouldInheritDesktop_1.paramShouldInheritDesktop(param);
        if (generatedKey !== key &&
            shouldInheritDesktop &&
            typeof value === 'undefined') {
            value = getter(key);
        }
        if (useDefaultValue &&
            typeof value === 'undefined' &&
            param.getDefaultValue) {
            var getDefaultValue = createStyleParamDefaultValueGetter({
                styles: styles,
                textPresets: textPresets,
                colors: colors,
                isMobile: isMobile,
                isRTL: isRTL,
            });
            return getDefaultValue(param);
        }
        return value;
    };
}
exports.createStyleParamGetter = createStyleParamGetter;
