// @wix/advanced-seo-utils/dist/src/private/item-types/restaurants-order-page/adapter-restaurants-order-page.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.IDs = void 0;
var utils_1 = require("../../adapters/utils");
var ContextProps_1 = require("../../types/ContextProps");
var get_1 = require("../../utils/get");
exports.IDs = {
    PAGE_NAME: 'page.name',
    ORDER_PAGE_URL: 'restaurants-order-page.page-url',
};
var SchemaTypes = {};
var userVisible = [];
exports.default = {
    IDs: exports.IDs,
    SchemaTypes: SchemaTypes,
    getData: function(item) {
        var _a;
        if (!get_1.get(item, 'menu')) {
            return {};
        }
        return _a = {},
            _a[exports.IDs.ORDER_PAGE_URL] = get_1.get(item, 'menu'),
            _a[exports.IDs.PAGE_NAME] = get_1.get(item, "context." + ContextProps_1.CONTEXT_PROPS.PAGE_NAME),
            _a;
    },
    getSdStatus: function() {},
    getLegacySeoBlob: function() {},
    getSlug: function() {
        return '';
    },
    updateItemDataWithSlug: function(item) {
        return item;
    },
    getBiData: function() {},
    getKeys: function() {
        return utils_1.getKeys(exports.IDs, userVisible);
    },
    getSdKeys: function() {
        return [];
    },
};
