// @wix/tpa-settings/dist/src/generateKey.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.generateKeyVariations = exports.isParamTranslatable = exports.generateKey = void 0;
var settings_key_format_1 = require("@wix/settings-key-format");
var types_1 = require("./types");
var DEFAULT_LANGUAGE = 'en';
exports.generateKey = function(param, _a) {
    var isMobile = _a.isMobile,
        _b = _a.language,
        language = _b === void 0 ? DEFAULT_LANGUAGE : _b;
    if (typeof param.dangerousKeyTransformationOverride === 'function') {
        // Please, use `dangerousKeyTransformationOverride` only for migration purposes!
        return param.dangerousKeyTransformationOverride(param.key, {
            isMobile: isMobile,
            language: language,
        });
    }
    return settings_key_format_1.generateKey(param.key, {
        m: isMobile,
        l: language
    }, {
        m: false,
        l: DEFAULT_LANGUAGE
    });
};
exports.isParamTranslatable = function(param) {
    return param.type === types_1.SettingsParamType.Text;
};
exports.generateKeyVariations = function(param, options) {
    var languageAndMobile = exports.generateKey(param, {
        isMobile: options.isMobile,
        language: options.language,
    });
    var language = exports.generateKey(param, {
        isMobile: false,
        language: options.language,
    });
    var mobile = exports.generateKey(param, {
        isMobile: options.isMobile,
    });
    return {
        languageAndMobile: languageAndMobile,
        language: language,
        mobile: mobile,
        original: param.key,
    };
};
