// @wix/wixstores-graphql-schema-node/dist/src/graphql-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ProductFieldName = exports.ProductBrandFieldName = exports.PlanDiscountMode = exports.PickupMethod = exports.PaymentStatusEnumPaymentStatus = exports.PaymentStatus = exports.PaymentOptionType = exports.OrderStatus = exports.OrderSortField = exports.OrderInHistoryStatus = exports.NotRefundableReason = exports.MissingTaxRatesReason = exports.MissingShippingRatesReason = exports.MediaItemType = exports.MediaItemEnum = exports.MeasurementUnitType = exports.ManualCalculationReason = exports.LockedOperations = exports.LockableOperation = exports.LineItemEnum = exports.LengthUnit = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.InventoryUpdatingMode = exports.InventoryStatus = exports.GiftCardErrorType = exports.GiftCardError = exports.FulfillmentStatus = exports.FilterOperators = exports.FileType = exports.FallbackReason = exports.ErrorTypes = exports.EmailStatusType = exports.EmailNotificationsConfiguration = exports.DiscountType = exports.DiscountReason = exports.DiscountMode = exports.DescriptionLineType = exports.CreateOrderFulfillmentStatus = exports.CouponErrorType = exports.ContentTypes = exports.ChargeType = exports.ChannelType = exports.CategoryFieldName = exports.BackInStockRequestStatus = exports.AttributionSource = exports.AppliedDiscountDiscountType = exports.ActivityType = exports.ActiveStatus = exports.AbandonedCartsSortField = void 0;
exports.WeightUnit = exports.VatTypeInput = exports.VatType = exports.V1DiscountType = exports.SubscriptionPlanFrequency = exports.SubscriptionFrequency = exports.StringOperator = exports.StoreVariantSortField = exports.Source = exports.SortDirection = exports.ShippingStatus = exports.ShippingLabelUnavailabilityReason = exports.SalesChannels = exports.RuleType = exports.RestrictableOperation = exports.RestockType = exports.RefundAbilityStatus = exports.RecoveryStatusType = exports.ReadStatus = exports.RateType = exports.ProductType = exports.ProductSortField = exports.ProductOptionType = exports.ProductGroupRateCalculationSettingType = void 0;
var AbandonedCartsSortField;
(function(AbandonedCartsSortField) {
    AbandonedCartsSortField["AbandonedDate"] = "AbandonedDate";
    AbandonedCartsSortField["AbandonTime"] = "AbandonTime";
})(AbandonedCartsSortField = exports.AbandonedCartsSortField || (exports.AbandonedCartsSortField = {}));
var ActiveStatus;
(function(ActiveStatus) {
    ActiveStatus["Active"] = "Active";
    ActiveStatus["Archived"] = "Archived";
})(ActiveStatus = exports.ActiveStatus || (exports.ActiveStatus = {}));
var ActivityType;
(function(ActivityType) {
    ActivityType["ORDER_REFUNDED"] = "ORDER_REFUNDED";
    ActivityType["ORDER_PLACED"] = "ORDER_PLACED";
    ActivityType["ORDER_PAID"] = "ORDER_PAID";
    ActivityType["ORDER_FULFILLED"] = "ORDER_FULFILLED";
    ActivityType["ORDER_NOT_FULFILLED"] = "ORDER_NOT_FULFILLED";
    ActivityType["ORDER_CANCELED"] = "ORDER_CANCELED";
    ActivityType["DOWNLOAD_LINK_SENT"] = "DOWNLOAD_LINK_SENT";
    ActivityType["TRACKING_NUMBER_ADDED"] = "TRACKING_NUMBER_ADDED";
    ActivityType["TRACKING_NUMBER_EDITED"] = "TRACKING_NUMBER_EDITED";
    ActivityType["TRACKING_LINK_ADDED"] = "TRACKING_LINK_ADDED";
    ActivityType["SHIPPING_CONFIRMATION_EMAIL_SENT"] = "SHIPPING_CONFIRMATION_EMAIL_SENT";
    ActivityType["INVOICE_ADDED"] = "INVOICE_ADDED";
    ActivityType["INVOICE_REMOVED"] = "INVOICE_REMOVED";
    ActivityType["INVOICE_SENT"] = "INVOICE_SENT";
    ActivityType["FULFILLER_EMAIL_SENT"] = "FULFILLER_EMAIL_SENT";
    ActivityType["SHIPPING_ADDRESS_EDITED"] = "SHIPPING_ADDRESS_EDITED";
    ActivityType["EMAIL_EDITED"] = "EMAIL_EDITED";
    ActivityType["PICKUP_READY_EMAIL_SENT"] = "PICKUP_READY_EMAIL_SENT";
    ActivityType["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
    ActivityType["MERCHANT_COMMENT"] = "MERCHANT_COMMENT";
    ActivityType["ORDER_CREATED_FROM_EXCHANGE"] = "ORDER_CREATED_FROM_EXCHANGE";
    ActivityType["NEW_EXCHANGE_ORDER_CREATED"] = "NEW_EXCHANGE_ORDER_CREATED";
})(ActivityType = exports.ActivityType || (exports.ActivityType = {}));
var AppliedDiscountDiscountType;
(function(AppliedDiscountDiscountType) {
    AppliedDiscountDiscountType["GLOBAL"] = "GLOBAL";
    AppliedDiscountDiscountType["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
    AppliedDiscountDiscountType["SHIPPING"] = "SHIPPING";
})(AppliedDiscountDiscountType = exports.AppliedDiscountDiscountType || (exports.AppliedDiscountDiscountType = {}));
var AttributionSource;
(function(AttributionSource) {
    AttributionSource["UNSPECIFIED"] = "UNSPECIFIED";
    AttributionSource["FACEBOOK_ADS"] = "FACEBOOK_ADS";
})(AttributionSource = exports.AttributionSource || (exports.AttributionSource = {}));
var BackInStockRequestStatus;
(function(BackInStockRequestStatus) {
    BackInStockRequestStatus["UNSPECIFIED"] = "UNSPECIFIED";
    BackInStockRequestStatus["RECEIVED"] = "RECEIVED";
    BackInStockRequestStatus["PROCESSING"] = "PROCESSING";
    BackInStockRequestStatus["NOTIFICATION_SENT"] = "NOTIFICATION_SENT";
    BackInStockRequestStatus["FAILED"] = "FAILED";
})(BackInStockRequestStatus = exports.BackInStockRequestStatus || (exports.BackInStockRequestStatus = {}));
var CategoryFieldName;
(function(CategoryFieldName) {
    CategoryFieldName["id"] = "id";
    CategoryFieldName["name"] = "name";
    CategoryFieldName["numOfProducts"] = "numOfProducts";
})(CategoryFieldName = exports.CategoryFieldName || (exports.CategoryFieldName = {}));
var ChannelType;
(function(ChannelType) {
    ChannelType["UNSPECIFIED"] = "UNSPECIFIED";
    ChannelType["WEB"] = "WEB";
    ChannelType["POS"] = "POS";
    ChannelType["EBAY"] = "EBAY";
    ChannelType["AMAZON"] = "AMAZON";
    ChannelType["WISH"] = "WISH";
    ChannelType["OTHER_PLATFORM"] = "OTHER_PLATFORM";
    ChannelType["WIX_APP_STORE"] = "WIX_APP_STORE";
    ChannelType["WIX_INVOICES"] = "WIX_INVOICES";
    ChannelType["BACKOFFICE_MERCHANT"] = "BACKOFFICE_MERCHANT";
})(ChannelType = exports.ChannelType || (exports.ChannelType = {}));
var ChargeType;
(function(ChargeType) {
    ChargeType["HANDLING_FEE"] = "HANDLING_FEE";
    ChargeType["INSURANCE"] = "INSURANCE";
})(ChargeType = exports.ChargeType || (exports.ChargeType = {}));
var ContentTypes;
(function(ContentTypes) {
    ContentTypes["gift"] = "gift";
    ContentTypes["merchandise"] = "merchandise";
    ContentTypes["other"] = "other";
})(ContentTypes = exports.ContentTypes || (exports.ContentTypes = {}));
var CouponErrorType;
(function(CouponErrorType) {
    CouponErrorType["EXPIRED"] = "EXPIRED";
})(CouponErrorType = exports.CouponErrorType || (exports.CouponErrorType = {}));
var CreateOrderFulfillmentStatus;
(function(CreateOrderFulfillmentStatus) {
    CreateOrderFulfillmentStatus["NOT_FULFILLED"] = "NOT_FULFILLED";
    CreateOrderFulfillmentStatus["FULFILLED"] = "FULFILLED";
    CreateOrderFulfillmentStatus["PARTIALLY_FULFILLED"] = "PARTIALLY_FULFILLED";
})(CreateOrderFulfillmentStatus = exports.CreateOrderFulfillmentStatus || (exports.CreateOrderFulfillmentStatus = {}));
var DescriptionLineType;
(function(DescriptionLineType) {
    DescriptionLineType["UNRECOGNISED"] = "UNRECOGNISED";
    DescriptionLineType["PLAIN_TEXT"] = "PLAIN_TEXT";
    DescriptionLineType["COLOR"] = "COLOR";
})(DescriptionLineType = exports.DescriptionLineType || (exports.DescriptionLineType = {}));
var DiscountMode;
(function(DiscountMode) {
    DiscountMode["Amount"] = "Amount";
    DiscountMode["Percent"] = "Percent";
})(DiscountMode = exports.DiscountMode || (exports.DiscountMode = {}));
var DiscountReason;
(function(DiscountReason) {
    DiscountReason["UNSPECIFIED"] = "UNSPECIFIED";
    DiscountReason["EXCHANGED_ITEMS"] = "EXCHANGED_ITEMS";
})(DiscountReason = exports.DiscountReason || (exports.DiscountReason = {}));
var DiscountType;
(function(DiscountType) {
    DiscountType["GLOBAL"] = "GLOBAL";
    DiscountType["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
    DiscountType["SHIPPING"] = "SHIPPING";
})(DiscountType = exports.DiscountType || (exports.DiscountType = {}));
var EmailNotificationsConfiguration;
(function(EmailNotificationsConfiguration) {
    EmailNotificationsConfiguration["ON_ORDER_CREATED"] = "ON_ORDER_CREATED";
    EmailNotificationsConfiguration["ON_ORDER_PAID"] = "ON_ORDER_PAID";
    EmailNotificationsConfiguration["OFF"] = "OFF";
})(EmailNotificationsConfiguration = exports.EmailNotificationsConfiguration || (exports.EmailNotificationsConfiguration = {}));
var EmailStatusType;
(function(EmailStatusType) {
    EmailStatusType["SENT"] = "SENT";
    EmailStatusType["NOT_SENT"] = "NOT_SENT";
    EmailStatusType["SCHEDULED"] = "SCHEDULED";
})(EmailStatusType = exports.EmailStatusType || (exports.EmailStatusType = {}));
var ErrorTypes;
(function(ErrorTypes) {
    ErrorTypes["MISSING_PARAMETER"] = "MISSING_PARAMETER";
    ErrorTypes["OUT_OF_RANGE"] = "OUT_OF_RANGE";
    ErrorTypes["SERVICE_UNAVAILABLE"] = "SERVICE_UNAVAILABLE";
})(ErrorTypes = exports.ErrorTypes || (exports.ErrorTypes = {}));
var FallbackReason;
(function(FallbackReason) {
    FallbackReason["AUTO_TAX_FAILED"] = "AUTO_TAX_FAILED";
    FallbackReason["AUTO_TAX_DEACTIVATED"] = "AUTO_TAX_DEACTIVATED";
})(FallbackReason = exports.FallbackReason || (exports.FallbackReason = {}));
var FileType;
(function(FileType) {
    FileType["UNSPECIFIED"] = "UNSPECIFIED";
    FileType["SECURE_PICTURE"] = "SECURE_PICTURE";
    FileType["SECURE_VIDEO"] = "SECURE_VIDEO";
    FileType["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
    FileType["SECURE_MUSIC"] = "SECURE_MUSIC";
    FileType["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
})(FileType = exports.FileType || (exports.FileType = {}));
var FilterOperators;
(function(FilterOperators) {
    FilterOperators["EQUALS"] = "EQUALS";
    FilterOperators["NEQUALS"] = "NEQUALS";
    FilterOperators["GT"] = "GT";
    FilterOperators["GTE"] = "GTE";
    FilterOperators["LT"] = "LT";
    FilterOperators["LTE"] = "LTE";
    FilterOperators["BETWEEN"] = "BETWEEN";
    FilterOperators["IN"] = "IN";
    FilterOperators["IN_ALL"] = "IN_ALL";
    FilterOperators["CONTAINS"] = "CONTAINS";
    FilterOperators["START_WITH"] = "START_WITH";
    FilterOperators["MATCH"] = "MATCH";
    FilterOperators["CONTAINS_ALL"] = "CONTAINS_ALL";
    FilterOperators["CONTAINS_ANY"] = "CONTAINS_ANY";
    FilterOperators["EXISTS"] = "EXISTS";
})(FilterOperators = exports.FilterOperators || (exports.FilterOperators = {}));
var FulfillmentStatus;
(function(FulfillmentStatus) {
    FulfillmentStatus["Fulfilled"] = "Fulfilled";
    FulfillmentStatus["NotFulfilled"] = "NotFulfilled";
    FulfillmentStatus["PartiallyFulfilled"] = "PartiallyFulfilled";
    FulfillmentStatus["Canceled"] = "Canceled";
})(FulfillmentStatus = exports.FulfillmentStatus || (exports.FulfillmentStatus = {}));
var GiftCardError;
(function(GiftCardError) {
    GiftCardError["GIFT_CARD_GENERAL_ERROR"] = "GIFT_CARD_GENERAL_ERROR";
    GiftCardError["GIFT_CARD_INSUFFICIENT_BALANCE"] = "GIFT_CARD_INSUFFICIENT_BALANCE";
    GiftCardError["GIFT_CARD_CODE_INVALID"] = "GIFT_CARD_CODE_INVALID";
    GiftCardError["GIFT_CARD_CURRENCY_INVALID"] = "GIFT_CARD_CURRENCY_INVALID";
    GiftCardError["GIFT_CARD_EXPIRED"] = "GIFT_CARD_EXPIRED";
    GiftCardError["GIFT_CARD_DISABLED"] = "GIFT_CARD_DISABLED";
    GiftCardError["GIFT_CARD_IS_EMPTY"] = "GIFT_CARD_IS_EMPTY";
    GiftCardError["GIFT_CARD_SERVICE_UNAVAILABLE"] = "GIFT_CARD_SERVICE_UNAVAILABLE";
})(GiftCardError = exports.GiftCardError || (exports.GiftCardError = {}));
var GiftCardErrorType;
(function(GiftCardErrorType) {
    GiftCardErrorType["EXPIRED"] = "EXPIRED";
})(GiftCardErrorType = exports.GiftCardErrorType || (exports.GiftCardErrorType = {}));
var InventoryStatus;
(function(InventoryStatus) {
    InventoryStatus["IN_STOCK_STATUS"] = "IN_STOCK_STATUS";
    InventoryStatus["OUT_OF_STOCK_STATUS"] = "OUT_OF_STOCK_STATUS";
    InventoryStatus["NOT_OUT_OF_STOCK_STATUS"] = "NOT_OUT_OF_STOCK_STATUS";
    InventoryStatus["PARTIALLY_OUT_OF_STOCK_STATUS"] = "PARTIALLY_OUT_OF_STOCK_STATUS";
})(InventoryStatus = exports.InventoryStatus || (exports.InventoryStatus = {}));
var InventoryUpdatingMode;
(function(InventoryUpdatingMode) {
    InventoryUpdatingMode["OnOrderPaid"] = "OnOrderPaid";
    InventoryUpdatingMode["OnOrderCreated"] = "OnOrderCreated";
    InventoryUpdatingMode["Unknown"] = "Unknown";
})(InventoryUpdatingMode = exports.InventoryUpdatingMode || (exports.InventoryUpdatingMode = {}));
var ItemAvailabilityStatus;
(function(ItemAvailabilityStatus) {
    ItemAvailabilityStatus["AVAILABLE"] = "AVAILABLE";
    ItemAvailabilityStatus["NOT_FOUND"] = "NOT_FOUND";
    ItemAvailabilityStatus["NOT_AVAILABLE"] = "NOT_AVAILABLE";
    ItemAvailabilityStatus["PARTIALLY_AVAILABLE"] = "PARTIALLY_AVAILABLE";
})(ItemAvailabilityStatus = exports.ItemAvailabilityStatus || (exports.ItemAvailabilityStatus = {}));
var ItemTypeItemType;
(function(ItemTypeItemType) {
    ItemTypeItemType["UNRECOGNISED"] = "UNRECOGNISED";
    ItemTypeItemType["PHYSICAL"] = "PHYSICAL";
    ItemTypeItemType["DIGITAL"] = "DIGITAL";
    ItemTypeItemType["GIFT_CARD"] = "GIFT_CARD";
})(ItemTypeItemType = exports.ItemTypeItemType || (exports.ItemTypeItemType = {}));
var LengthUnit;
(function(LengthUnit) {
    LengthUnit["UNSPECIFIED_LENGTH_UNIT"] = "UNSPECIFIED_LENGTH_UNIT";
    LengthUnit["KM"] = "KM";
    LengthUnit["MI"] = "MI";
})(LengthUnit = exports.LengthUnit || (exports.LengthUnit = {}));
var LineItemEnum;
(function(LineItemEnum) {
    LineItemEnum["UNSPECIFIED_LINE_ITEM_TYPE"] = "UNSPECIFIED_LINE_ITEM_TYPE";
    LineItemEnum["PHYSICAL"] = "PHYSICAL";
    LineItemEnum["DIGITAL"] = "DIGITAL";
    LineItemEnum["CUSTOM_AMOUNT_ITEM"] = "CUSTOM_AMOUNT_ITEM";
})(LineItemEnum = exports.LineItemEnum || (exports.LineItemEnum = {}));
var LockableOperation;
(function(LockableOperation) {
    LockableOperation["LOCKED"] = "LOCKED";
    LockableOperation["UNLOCKED"] = "UNLOCKED";
})(LockableOperation = exports.LockableOperation || (exports.LockableOperation = {}));
var LockedOperations;
(function(LockedOperations) {
    LockedOperations["ManageProductPricing"] = "ManageProductPricing";
    LockedOperations["ManageProductVariants"] = "ManageProductVariants";
    LockedOperations["ManageProductInventory"] = "ManageProductInventory";
    LockedOperations["DuplicateProduct"] = "DuplicateProduct";
})(LockedOperations = exports.LockedOperations || (exports.LockedOperations = {}));
var ManualCalculationReason;
(function(ManualCalculationReason) {
    ManualCalculationReason["GLOBAL_SETTING_TO_MANUAL"] = "GLOBAL_SETTING_TO_MANUAL";
    ManualCalculationReason["REGION_SETTING_TO_MANUAL"] = "REGION_SETTING_TO_MANUAL";
})(ManualCalculationReason = exports.ManualCalculationReason || (exports.ManualCalculationReason = {}));
var MeasurementUnitType;
(function(MeasurementUnitType) {
    MeasurementUnitType["ml"] = "ml";
    MeasurementUnitType["cl"] = "cl";
    MeasurementUnitType["l"] = "l";
    MeasurementUnitType["cbm"] = "cbm";
    MeasurementUnitType["mg"] = "mg";
    MeasurementUnitType["g"] = "g";
    MeasurementUnitType["kg"] = "kg";
    MeasurementUnitType["mm"] = "mm";
    MeasurementUnitType["cm"] = "cm";
    MeasurementUnitType["m"] = "m";
    MeasurementUnitType["sqm"] = "sqm";
    MeasurementUnitType["oz"] = "oz";
    MeasurementUnitType["lb"] = "lb";
    MeasurementUnitType["floz"] = "floz";
    MeasurementUnitType["pt"] = "pt";
    MeasurementUnitType["qt"] = "qt";
    MeasurementUnitType["gal"] = "gal";
    MeasurementUnitType["in"] = "in";
    MeasurementUnitType["ft"] = "ft";
    MeasurementUnitType["yd"] = "yd";
    MeasurementUnitType["sqft"] = "sqft";
})(MeasurementUnitType = exports.MeasurementUnitType || (exports.MeasurementUnitType = {}));
var MediaItemEnum;
(function(MediaItemEnum) {
    MediaItemEnum["PHOTO"] = "PHOTO";
    MediaItemEnum["UNSUPPORTED"] = "UNSUPPORTED";
})(MediaItemEnum = exports.MediaItemEnum || (exports.MediaItemEnum = {}));
var MediaItemType;
(function(MediaItemType) {
    MediaItemType["unspecified_media_item_type"] = "unspecified_media_item_type";
    MediaItemType["image"] = "image";
    MediaItemType["video"] = "video";
    MediaItemType["audio"] = "audio";
    MediaItemType["document"] = "document";
    MediaItemType["zip"] = "zip";
})(MediaItemType = exports.MediaItemType || (exports.MediaItemType = {}));
var MissingShippingRatesReason;
(function(MissingShippingRatesReason) {
    MissingShippingRatesReason["UNRECOGNISED"] = "UNRECOGNISED";
    MissingShippingRatesReason["MISSING_ADDRESS"] = "MISSING_ADDRESS";
})(MissingShippingRatesReason = exports.MissingShippingRatesReason || (exports.MissingShippingRatesReason = {}));
var MissingTaxRatesReason;
(function(MissingTaxRatesReason) {
    MissingTaxRatesReason["UNRECOGNISED"] = "UNRECOGNISED";
    MissingTaxRatesReason["MISSING_ADDRESS"] = "MISSING_ADDRESS";
})(MissingTaxRatesReason = exports.MissingTaxRatesReason || (exports.MissingTaxRatesReason = {}));
var NotRefundableReason;
(function(NotRefundableReason) {
    NotRefundableReason["ALREADY_REFUNDED"] = "ALREADY_REFUNDED";
    NotRefundableReason["PROVIDER_IS_DOWN"] = "PROVIDER_IS_DOWN";
    NotRefundableReason["NOT_SUPPORTED"] = "NOT_SUPPORTED";
    NotRefundableReason["EXPIRED"] = "EXPIRED";
    NotRefundableReason["NOT_FOUND"] = "NOT_FOUND";
    NotRefundableReason["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    NotRefundableReason["NOT_PAID"] = "NOT_PAID";
    NotRefundableReason["ACCESS_DENIED"] = "ACCESS_DENIED";
    NotRefundableReason["ZERO_PRICE"] = "ZERO_PRICE";
    NotRefundableReason["OFFLINE"] = "OFFLINE";
    NotRefundableReason["DISABLE_BY_VENDOR"] = "DISABLE_BY_VENDOR";
})(NotRefundableReason = exports.NotRefundableReason || (exports.NotRefundableReason = {}));
var OrderInHistoryStatus;
(function(OrderInHistoryStatus) {
    OrderInHistoryStatus["Confirmed"] = "Confirmed";
    OrderInHistoryStatus["Shipped"] = "Shipped";
    OrderInHistoryStatus["Fulfilled"] = "Fulfilled";
    OrderInHistoryStatus["Tracking"] = "Tracking";
    OrderInHistoryStatus["Canceled"] = "Canceled";
})(OrderInHistoryStatus = exports.OrderInHistoryStatus || (exports.OrderInHistoryStatus = {}));
var OrderSortField;
(function(OrderSortField) {
    OrderSortField["IncrementId"] = "IncrementId";
    OrderSortField["Quantity"] = "Quantity";
    OrderSortField["CreationDate"] = "CreationDate";
    OrderSortField["CustomerName"] = "CustomerName";
    OrderSortField["PaymentStatus"] = "PaymentStatus";
    OrderSortField["ShippingStatus"] = "ShippingStatus";
    OrderSortField["Total"] = "Total";
    OrderSortField["FulfillmentStatus"] = "FulfillmentStatus";
    OrderSortField["LastUpdated"] = "LastUpdated";
})(OrderSortField = exports.OrderSortField || (exports.OrderSortField = {}));
var OrderStatus;
(function(OrderStatus) {
    OrderStatus["INITIALIZED"] = "INITIALIZED";
    OrderStatus["APPROVED"] = "APPROVED";
    OrderStatus["CANCELED"] = "CANCELED";
})(OrderStatus = exports.OrderStatus || (exports.OrderStatus = {}));
var PaymentOptionType;
(function(PaymentOptionType) {
    PaymentOptionType["FULL_PAYMENT_ONLINE"] = "FULL_PAYMENT_ONLINE";
    PaymentOptionType["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
    PaymentOptionType["DEPOSIT_ONLINE"] = "DEPOSIT_ONLINE";
})(PaymentOptionType = exports.PaymentOptionType || (exports.PaymentOptionType = {}));
var PaymentStatus;
(function(PaymentStatus) {
    PaymentStatus["Paid"] = "Paid";
    PaymentStatus["NotPaid"] = "NotPaid";
    PaymentStatus["Refunded"] = "Refunded";
})(PaymentStatus = exports.PaymentStatus || (exports.PaymentStatus = {}));
var PaymentStatusEnumPaymentStatus;
(function(PaymentStatusEnumPaymentStatus) {
    PaymentStatusEnumPaymentStatus["UNSPECIFIED"] = "UNSPECIFIED";
    PaymentStatusEnumPaymentStatus["NOT_PAID"] = "NOT_PAID";
    PaymentStatusEnumPaymentStatus["PAID"] = "PAID";
    PaymentStatusEnumPaymentStatus["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
    PaymentStatusEnumPaymentStatus["FULLY_REFUNDED"] = "FULLY_REFUNDED";
    PaymentStatusEnumPaymentStatus["PENDING"] = "PENDING";
})(PaymentStatusEnumPaymentStatus = exports.PaymentStatusEnumPaymentStatus || (exports.PaymentStatusEnumPaymentStatus = {}));
var PickupMethod;
(function(PickupMethod) {
    PickupMethod["STORE_PICKUP"] = "STORE_PICKUP";
    PickupMethod["PICKUP_POINT"] = "PICKUP_POINT";
    PickupMethod["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
})(PickupMethod = exports.PickupMethod || (exports.PickupMethod = {}));
var PlanDiscountMode;
(function(PlanDiscountMode) {
    PlanDiscountMode["AMOUNT"] = "AMOUNT";
    PlanDiscountMode["PERCENT"] = "PERCENT";
})(PlanDiscountMode = exports.PlanDiscountMode || (exports.PlanDiscountMode = {}));
var ProductBrandFieldName;
(function(ProductBrandFieldName) {
    ProductBrandFieldName["brand"] = "brand";
})(ProductBrandFieldName = exports.ProductBrandFieldName || (exports.ProductBrandFieldName = {}));
var ProductFieldName;
(function(ProductFieldName) {
    ProductFieldName["name"] = "name";
    ProductFieldName["creationDate"] = "creationDate";
    ProductFieldName["createVersion"] = "createVersion";
    ProductFieldName["price"] = "price";
    ProductFieldName["comparePrice"] = "comparePrice";
    ProductFieldName["discountedPrice"] = "discountedPrice";
    ProductFieldName["description"] = "description";
    ProductFieldName["ribbon"] = "ribbon";
    ProductFieldName["sku"] = "sku";
    ProductFieldName["visible"] = "visible";
    ProductFieldName["deleted"] = "deleted";
    ProductFieldName["created"] = "created";
    ProductFieldName["categories"] = "categories";
    ProductFieldName["id"] = "id";
    ProductFieldName["productType"] = "productType";
    ProductFieldName["inventoryStatus"] = "inventoryStatus";
    ProductFieldName["optionChoices"] = "optionChoices";
    ProductFieldName["slug"] = "slug";
    ProductFieldName["taxGroupId"] = "taxGroupId";
    ProductFieldName["handleId"] = "handleId";
    ProductFieldName["lastUpdated"] = "lastUpdated";
    ProductFieldName["shippingProductGroupId"] = "shippingProductGroupId";
    ProductFieldName["planIds"] = "planIds";
    ProductFieldName["allowOneTimePurchase"] = "allowOneTimePurchase";
    ProductFieldName["fulfillerId"] = "fulfillerId";
    ProductFieldName["readyToSellExternalAppId"] = "readyToSellExternalAppId";
    ProductFieldName["hideFromPos"] = "hideFromPos";
    ProductFieldName["brand"] = "brand";
})(ProductFieldName = exports.ProductFieldName || (exports.ProductFieldName = {}));
var ProductGroupRateCalculationSettingType;
(function(ProductGroupRateCalculationSettingType) {
    ProductGroupRateCalculationSettingType["APPLY_RATE_TO_EACH_PRODUCT"] = "APPLY_RATE_TO_EACH_PRODUCT";
    ProductGroupRateCalculationSettingType["APPLY_RATE_ONCE_PER_GROUP"] = "APPLY_RATE_ONCE_PER_GROUP";
    ProductGroupRateCalculationSettingType["APPLY_HIGHEST_RATE_ONCE"] = "APPLY_HIGHEST_RATE_ONCE";
})(ProductGroupRateCalculationSettingType = exports.ProductGroupRateCalculationSettingType || (exports.ProductGroupRateCalculationSettingType = {}));
var ProductOptionType;
(function(ProductOptionType) {
    ProductOptionType["DROP_DOWN"] = "DROP_DOWN";
    ProductOptionType["COLOR"] = "COLOR";
})(ProductOptionType = exports.ProductOptionType || (exports.ProductOptionType = {}));
var ProductSortField;
(function(ProductSortField) {
    ProductSortField["Name"] = "Name";
    ProductSortField["CreationDate"] = "CreationDate";
    ProductSortField["CreateVersion"] = "CreateVersion";
    ProductSortField["Price"] = "Price";
    ProductSortField["ComparePrice"] = "ComparePrice";
    ProductSortField["DiscountedPrice"] = "DiscountedPrice";
    ProductSortField["Id"] = "Id";
    ProductSortField["LastUpdated"] = "LastUpdated";
    ProductSortField["name"] = "name";
    ProductSortField["creationDate"] = "creationDate";
    ProductSortField["createVersion"] = "createVersion";
    ProductSortField["price"] = "price";
    ProductSortField["comparePrice"] = "comparePrice";
    ProductSortField["discountedPrice"] = "discountedPrice";
    ProductSortField["id"] = "id";
    ProductSortField["lastUpdated"] = "lastUpdated";
})(ProductSortField = exports.ProductSortField || (exports.ProductSortField = {}));
var ProductType;
(function(ProductType) {
    ProductType["Physical"] = "Physical";
    ProductType["Digital"] = "Digital";
})(ProductType = exports.ProductType || (exports.ProductType = {}));
var RateType;
(function(RateType) {
    RateType["NO_TAX_COLLECTED"] = "NO_TAX_COLLECTED";
    RateType["MANUAL_RATE"] = "MANUAL_RATE";
    RateType["AUTO_RATE"] = "AUTO_RATE";
    RateType["FALLBACK_RATE"] = "FALLBACK_RATE";
})(RateType = exports.RateType || (exports.RateType = {}));
var ReadStatus;
(function(ReadStatus) {
    ReadStatus["Read"] = "Read";
    ReadStatus["NotRead"] = "NotRead";
})(ReadStatus = exports.ReadStatus || (exports.ReadStatus = {}));
var RecoveryStatusType;
(function(RecoveryStatusType) {
    RecoveryStatusType["RECOVERED"] = "RECOVERED";
    RecoveryStatusType["NOT_RECOVERED"] = "NOT_RECOVERED";
})(RecoveryStatusType = exports.RecoveryStatusType || (exports.RecoveryStatusType = {}));
var RefundAbilityStatus;
(function(RefundAbilityStatus) {
    RefundAbilityStatus["REFUNDABLE"] = "REFUNDABLE";
    RefundAbilityStatus["MANUAL"] = "MANUAL";
    RefundAbilityStatus["NOT_REFUNDABLE"] = "NOT_REFUNDABLE";
})(RefundAbilityStatus = exports.RefundAbilityStatus || (exports.RefundAbilityStatus = {}));
var RestockType;
(function(RestockType) {
    RestockType["NO_ITEMS"] = "NO_ITEMS";
    RestockType["ALL_ITEMS"] = "ALL_ITEMS";
    RestockType["SOME_ITEMS"] = "SOME_ITEMS";
})(RestockType = exports.RestockType || (exports.RestockType = {}));
var RestrictableOperation;
(function(RestrictableOperation) {
    RestrictableOperation["ALLOW"] = "ALLOW";
    RestrictableOperation["WARN"] = "WARN";
})(RestrictableOperation = exports.RestrictableOperation || (exports.RestrictableOperation = {}));
var RuleType;
(function(RuleType) {
    RuleType["VALIDATION"] = "VALIDATION";
    RuleType["OTHER"] = "OTHER";
    RuleType["MAX"] = "MAX";
    RuleType["MIN"] = "MIN";
    RuleType["MAX_LENGTH"] = "MAX_LENGTH";
    RuleType["MIN_LENGTH"] = "MIN_LENGTH";
    RuleType["MAX_SIZE"] = "MAX_SIZE";
    RuleType["MIN_SIZE"] = "MIN_SIZE";
    RuleType["FORMAT"] = "FORMAT";
    RuleType["DECIMAL_LTE"] = "DECIMAL_LTE";
    RuleType["DECIMAL_GTE"] = "DECIMAL_GTE";
    RuleType["DECIMAL_LT"] = "DECIMAL_LT";
    RuleType["DECIMAL_GT"] = "DECIMAL_GT";
    RuleType["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
    RuleType["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
    RuleType["REQUIRED_FIELD"] = "REQUIRED_FIELD";
})(RuleType = exports.RuleType || (exports.RuleType = {}));
var SalesChannels;
(function(SalesChannels) {
    SalesChannels["Web"] = "Web";
    SalesChannels["Pos"] = "Pos";
    SalesChannels["Ebay"] = "Ebay";
    SalesChannels["Amazon"] = "Amazon";
    SalesChannels["Wish"] = "Wish";
    SalesChannels["OtherPlatform"] = "OtherPlatform";
    SalesChannels["WixAppStore"] = "WixAppStore";
})(SalesChannels = exports.SalesChannels || (exports.SalesChannels = {}));
var ShippingLabelUnavailabilityReason;
(function(ShippingLabelUnavailabilityReason) {
    ShippingLabelUnavailabilityReason["FEATURE_KILLED"] = "FEATURE_KILLED";
    ShippingLabelUnavailabilityReason["INELIGIBLE_STORE"] = "INELIGIBLE_STORE";
})(ShippingLabelUnavailabilityReason = exports.ShippingLabelUnavailabilityReason || (exports.ShippingLabelUnavailabilityReason = {}));
var ShippingStatus;
(function(ShippingStatus) {
    ShippingStatus["Fulfilled"] = "Fulfilled";
    ShippingStatus["NotFulfilled"] = "NotFulfilled";
    ShippingStatus["Canceled"] = "Canceled";
})(ShippingStatus = exports.ShippingStatus || (exports.ShippingStatus = {}));
var SortDirection;
(function(SortDirection) {
    SortDirection["Ascending"] = "Ascending";
    SortDirection["Descending"] = "Descending";
})(SortDirection = exports.SortDirection || (exports.SortDirection = {}));
var Source;
(function(Source) {
    Source["UNSUPPORTED_SOURCE"] = "UNSUPPORTED_SOURCE";
    Source["WIX_FACEBOOK_ADS"] = "WIX_FACEBOOK_ADS";
})(Source = exports.Source || (exports.Source = {}));
var StoreVariantSortField;
(function(StoreVariantSortField) {
    StoreVariantSortField["ProductNumericId"] = "ProductNumericId";
    StoreVariantSortField["VariantId"] = "VariantId";
    StoreVariantSortField["productNumericId"] = "productNumericId";
    StoreVariantSortField["variantId"] = "variantId";
})(StoreVariantSortField = exports.StoreVariantSortField || (exports.StoreVariantSortField = {}));
var StringOperator;
(function(StringOperator) {
    StringOperator["Equals"] = "Equals";
    StringOperator["Contains"] = "Contains";
})(StringOperator = exports.StringOperator || (exports.StringOperator = {}));
var SubscriptionFrequency;
(function(SubscriptionFrequency) {
    SubscriptionFrequency["UNDEFINED"] = "UNDEFINED";
    SubscriptionFrequency["DAY"] = "DAY";
    SubscriptionFrequency["WEEK"] = "WEEK";
    SubscriptionFrequency["MONTH"] = "MONTH";
    SubscriptionFrequency["YEAR"] = "YEAR";
})(SubscriptionFrequency = exports.SubscriptionFrequency || (exports.SubscriptionFrequency = {}));
var SubscriptionPlanFrequency;
(function(SubscriptionPlanFrequency) {
    SubscriptionPlanFrequency["MONTH"] = "MONTH";
    SubscriptionPlanFrequency["DAY"] = "DAY";
    SubscriptionPlanFrequency["WEEK"] = "WEEK";
    SubscriptionPlanFrequency["YEAR"] = "YEAR";
    SubscriptionPlanFrequency["UNDEFINED"] = "UNDEFINED";
})(SubscriptionPlanFrequency = exports.SubscriptionPlanFrequency || (exports.SubscriptionPlanFrequency = {}));
var V1DiscountType;
(function(V1DiscountType) {
    V1DiscountType["UNDEFINED"] = "UNDEFINED";
    V1DiscountType["GLOBAL"] = "GLOBAL";
    V1DiscountType["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
    V1DiscountType["SHIPPING"] = "SHIPPING";
})(V1DiscountType = exports.V1DiscountType || (exports.V1DiscountType = {}));
var VatType;
(function(VatType) {
    VatType["UNSPECIFIED_TAX_TYPE"] = "UNSPECIFIED_TAX_TYPE";
    VatType["CPF"] = "CPF";
    VatType["CNPJ"] = "CNPJ";
})(VatType = exports.VatType || (exports.VatType = {}));
var VatTypeInput;
(function(VatTypeInput) {
    VatTypeInput["UNSPECIFIED"] = "UNSPECIFIED";
    VatTypeInput["CPF"] = "CPF";
    VatTypeInput["CNPJ"] = "CNPJ";
})(VatTypeInput = exports.VatTypeInput || (exports.VatTypeInput = {}));
var WeightUnit;
(function(WeightUnit) {
    WeightUnit["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
    WeightUnit["KG"] = "KG";
    WeightUnit["LB"] = "LB";
})(WeightUnit = exports.WeightUnit || (exports.WeightUnit = {}));
