// @wix/ambassador-checkout/http.js
/* eslint-disable */
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, {
        enumerable: true,
        get: function() {
            return m[k];
        }
    });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m)
        if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __awaiter = (this && this.__awaiter) || function(thisArg, _arguments, P, generator) {
    function adopt(value) {
        return value instanceof P ? value : new P(function(resolve) {
            resolve(value);
        });
    }
    return new(P || (P = Promise))(function(resolve, reject) {
        function fulfilled(value) {
            try {
                step(generator.next(value));
            } catch (e) {
                reject(e);
            }
        }

        function rejected(value) {
            try {
                step(generator["throw"](value));
            } catch (e) {
                reject(e);
            }
        }

        function step(result) {
            result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected);
        }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function(thisArg, body) {
    var _ = {
            label: 0,
            sent: function() {
                if (t[0] & 1) throw t[1];
                return t[1];
            },
            trys: [],
            ops: []
        },
        f, y, t, g;
    return g = {
        next: verb(0),
        "throw": verb(1),
        "return": verb(2)
    }, typeof Symbol === "function" && (g[Symbol.iterator] = function() {
        return this;
    }), g;

    function verb(n) {
        return function(v) {
            return step([n, v]);
        };
    }

    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0:
                case 1:
                    t = op;
                    break;
                case 4:
                    _.label++;
                    return {
                        value: op[1],
                        done: false
                    };
                case 5:
                    _.label++;
                    y = op[1];
                    op = [0];
                    continue;
                case 7:
                    op = _.ops.pop();
                    _.trys.pop();
                    continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                        _ = 0;
                        continue;
                    }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                        _.label = op[1];
                        break;
                    }
                    if (op[0] === 6 && _.label < t[1]) {
                        _.label = t[1];
                        t = op;
                        break;
                    }
                    if (t && _.label < t[2]) {
                        _.label = t[2];
                        _.ops.push(op);
                        break;
                    }
                    if (t[2]) _.ops.pop();
                    _.trys.pop();
                    continue;
            }
            op = body.call(thisArg, _);
        } catch (e) {
            op = [6, e];
            y = 0;
        } finally {
            f = t = 0;
        }
        if (op[0] & 5) throw op[1];
        return {
            value: op[0] ? op[1] : void 0,
            done: true
        };
    }
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Checkout = void 0;
var json_proto_serializer_1 = require("@wix/json-proto-serializer");
var http_1 = require("@wix/ambassador/runtime/http");
var Checkout = function(baseUrl, httpClientConfig) {
    var httpClient = new http_1.AmbassadorHTTPClient(baseUrl, httpClientConfig);
    return {
        CheckoutService: function() {
            return function(headers) {
                var service = {
                    createCheckout: function(createCheckoutRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var requestTransformations, serializedPayload, response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        requestTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkoutInfo', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkoutInfo', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkoutInfo', 'updatedDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['customLineItems', 'media', 'urlExpirationDate']
                                            },
                                        ];
                                        serializedPayload = json_proto_serializer_1.batchToJson(createCheckoutRequest, requestTransformations);
                                        return [4 /*yield*/ , httpClient.invoke('POST', '/v1/checkouts', serializedPayload, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                    createOrder: function(createOrderRequest) {
                        return httpClient.invoke('POST', '/v1/checkouts/{id}/create-order', createOrderRequest, headers);
                    },
                    createOrderAndCharge: function(createOrderAndChargeRequest) {
                        return httpClient.invoke('POST', '/v1/checkouts/{id}/create-order-and-charge', createOrderAndChargeRequest, headers);
                    },
                    getCheckout: function(getCheckoutRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        return [4 /*yield*/ , httpClient.invoke('GET', '/v1/checkouts/{id}', getCheckoutRequest, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                    getWixCheckoutURL: function(getWixCheckoutURLRequest) {
                        return httpClient.invoke('GET', '/v1/checkouts/{id}/wix-checkout-url', getWixCheckoutURLRequest, headers);
                    },
                    markCheckoutAsCompleted: function(markCheckoutAsCompletedRequest) {
                        return httpClient.invoke('POST', '/v1/checkouts/{id}/mark-as-completed', markCheckoutAsCompletedRequest, headers);
                    },
                    removeCoupon: function(removeCouponRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        return [4 /*yield*/ , httpClient.invoke('POST', '/v1/checkouts/{id}/remove-coupon', removeCouponRequest, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                    removeGiftCard: function(removeGiftCardRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        return [4 /*yield*/ , httpClient.invoke('POST', '/v1/checkouts/{id}/remove-gift-card', removeGiftCardRequest, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                    removeLineItems: function(removeLineItemsRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        return [4 /*yield*/ , httpClient.invoke('POST', '/v1/checkouts/{id}/remove-line-items', removeLineItemsRequest, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                    updateCheckout: function(updateCheckoutRequest) {
                        return __awaiter(void 0, void 0, void 0, function() {
                            var requestTransformations, serializedPayload, response, responseTransformations, deSerializedResponse;
                            return __generator(this, function(_a) {
                                switch (_a.label) {
                                    case 0:
                                        requestTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.FieldMaskArray',
                                                pathInPayload: ['fieldMask']
                                            },
                                        ];
                                        serializedPayload = json_proto_serializer_1.batchToJson(updateCheckoutRequest, requestTransformations);
                                        return [4 /*yield*/ , httpClient.invoke('PATCH', '/v1/checkouts/{checkout.id}', serializedPayload, headers)];
                                    case 1:
                                        response = _a.sent();
                                        responseTransformations = [{
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'lineItems', 'media', 'urlExpirationDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'createdDate']
                                            },
                                            {
                                                typeName: 'google.protobuf.Timestamp',
                                                pathInPayload: ['checkout', 'updatedDate']
                                            },
                                        ];
                                        deSerializedResponse = json_proto_serializer_1.batchFromJson(response, responseTransformations);
                                        return [2 /*return*/ , deSerializedResponse];
                                }
                            });
                        });
                    },
                };
                service.createCheckout.__http_method = 'POST';
                service.createCheckout.__http_url = '/v1/checkouts';
                service.createOrder.__http_method = 'POST';
                service.createOrder.__http_url = '/v1/checkouts/{id}/create-order';
                service.createOrderAndCharge.__http_method = 'POST';
                service.createOrderAndCharge.__http_url = '/v1/checkouts/{id}/create-order-and-charge';
                service.getCheckout.__http_method = 'GET';
                service.getCheckout.__http_url = '/v1/checkouts/{id}';
                service.getWixCheckoutURL.__http_method = 'GET';
                service.getWixCheckoutURL.__http_url = '/v1/checkouts/{id}/wix-checkout-url';
                service.markCheckoutAsCompleted.__http_method = 'POST';
                service.markCheckoutAsCompleted.__http_url = '/v1/checkouts/{id}/mark-as-completed';
                service.removeCoupon.__http_method = 'POST';
                service.removeCoupon.__http_url = '/v1/checkouts/{id}/remove-coupon';
                service.removeGiftCard.__http_method = 'POST';
                service.removeGiftCard.__http_url = '/v1/checkouts/{id}/remove-gift-card';
                service.removeLineItems.__http_method = 'POST';
                service.removeLineItems.__http_url = '/v1/checkouts/{id}/remove-line-items';
                service.updateCheckout.__http_method = 'PATCH';
                service.updateCheckout.__http_url = '/v1/checkouts/{checkout.id}';
                return service;
            };
        },
    };
};
exports.Checkout = Checkout;
exports.Checkout.__transport = 'http';
__exportStar(require("./types"), exports);
