// @wix/ambassador-checkout/types.js
/* eslint-disable */
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.WeightUnit = exports.V2IdentityType = exports.V1IdentityType = exports.SubscriptionFrequency = exports.RuleType = exports.RateType = exports.PickupMethod = exports.PaymentStatus = exports.PaymentOptionType = exports.MediaItemType = exports.ManualCalculationReason = exports.LineItemType = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.FulfillmentStatus = exports.FileType = exports.FallbackReason = exports.DiscountDiscountType = exports.DescriptionLineType = exports.CommonVatType = exports.ChargeType = exports.ChannelInfoChannelType = exports.AppliedDiscountDiscountType = exports.ApiVatType = exports.ApiChannelType = void 0;
var ApiChannelType;
(function(ApiChannelType) {
    ApiChannelType["UNSPECIFIED"] = "UNSPECIFIED";
    ApiChannelType["WEB"] = "WEB";
    ApiChannelType["POS"] = "POS";
    ApiChannelType["EBAY"] = "EBAY";
    ApiChannelType["AMAZON"] = "AMAZON";
    ApiChannelType["OTHER_PLATFORM"] = "OTHER_PLATFORM";
    ApiChannelType["WIX_APP_STORE"] = "WIX_APP_STORE";
    ApiChannelType["WIX_INVOICES"] = "WIX_INVOICES";
    ApiChannelType["BACKOFFICE_MERCHANT"] = "BACKOFFICE_MERCHANT";
    ApiChannelType["WISH"] = "WISH";
})(ApiChannelType = exports.ApiChannelType || (exports.ApiChannelType = {}));
var ApiVatType;
(function(ApiVatType) {
    ApiVatType["UNSPECIFIED_TAX_TYPE"] = "UNSPECIFIED_TAX_TYPE";
    ApiVatType["CPF"] = "CPF";
    ApiVatType["CNPJ"] = "CNPJ";
})(ApiVatType = exports.ApiVatType || (exports.ApiVatType = {}));
var AppliedDiscountDiscountType;
(function(AppliedDiscountDiscountType) {
    AppliedDiscountDiscountType["GLOBAL"] = "GLOBAL";
    AppliedDiscountDiscountType["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
    AppliedDiscountDiscountType["SHIPPING"] = "SHIPPING";
})(AppliedDiscountDiscountType = exports.AppliedDiscountDiscountType || (exports.AppliedDiscountDiscountType = {}));
var ChannelInfoChannelType;
(function(ChannelInfoChannelType) {
    ChannelInfoChannelType["UNSPECIFIED"] = "UNSPECIFIED";
    ChannelInfoChannelType["WEB"] = "WEB";
    ChannelInfoChannelType["OTHER_PLATFORM"] = "OTHER_PLATFORM";
    ChannelInfoChannelType["WIX_APP_STORE"] = "WIX_APP_STORE";
})(ChannelInfoChannelType = exports.ChannelInfoChannelType || (exports.ChannelInfoChannelType = {}));
var ChargeType;
(function(ChargeType) {
    ChargeType["HANDLING_FEE"] = "HANDLING_FEE";
    ChargeType["INSURANCE"] = "INSURANCE";
})(ChargeType = exports.ChargeType || (exports.ChargeType = {}));
var CommonVatType;
(function(CommonVatType) {
    CommonVatType["UNSPECIFIED"] = "UNSPECIFIED";
    CommonVatType["CPF"] = "CPF";
    CommonVatType["CNPJ"] = "CNPJ";
})(CommonVatType = exports.CommonVatType || (exports.CommonVatType = {}));
var DescriptionLineType;
(function(DescriptionLineType) {
    DescriptionLineType["UNRECOGNISED"] = "UNRECOGNISED";
    DescriptionLineType["PLAIN_TEXT"] = "PLAIN_TEXT";
    DescriptionLineType["COLOR"] = "COLOR";
})(DescriptionLineType = exports.DescriptionLineType || (exports.DescriptionLineType = {}));
var DiscountDiscountType;
(function(DiscountDiscountType) {
    DiscountDiscountType["UNDEFINED"] = "UNDEFINED";
    DiscountDiscountType["AMOUNT"] = "AMOUNT";
    DiscountDiscountType["PERCENT"] = "PERCENT";
})(DiscountDiscountType = exports.DiscountDiscountType || (exports.DiscountDiscountType = {}));
var FallbackReason;
(function(FallbackReason) {
    FallbackReason["AUTO_TAX_FAILED"] = "AUTO_TAX_FAILED";
    FallbackReason["AUTO_TAX_DEACTIVATED"] = "AUTO_TAX_DEACTIVATED";
})(FallbackReason = exports.FallbackReason || (exports.FallbackReason = {}));
var FileType;
(function(FileType) {
    FileType["UNSPECIFIED"] = "UNSPECIFIED";
    FileType["SECURE_PICTURE"] = "SECURE_PICTURE";
    FileType["SECURE_VIDEO"] = "SECURE_VIDEO";
    FileType["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
    FileType["SECURE_MUSIC"] = "SECURE_MUSIC";
    FileType["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
})(FileType = exports.FileType || (exports.FileType = {}));
var FulfillmentStatus;
(function(FulfillmentStatus) {
    FulfillmentStatus["NOT_FULFILLED"] = "NOT_FULFILLED";
    FulfillmentStatus["FULFILLED"] = "FULFILLED";
    FulfillmentStatus["CANCELED"] = "CANCELED";
    FulfillmentStatus["PARTIALLY_FULFILLED"] = "PARTIALLY_FULFILLED";
})(FulfillmentStatus = exports.FulfillmentStatus || (exports.FulfillmentStatus = {}));
var ItemAvailabilityStatus;
(function(ItemAvailabilityStatus) {
    ItemAvailabilityStatus["AVAILABLE"] = "AVAILABLE";
    ItemAvailabilityStatus["NOT_FOUND"] = "NOT_FOUND";
    ItemAvailabilityStatus["NOT_AVAILABLE"] = "NOT_AVAILABLE";
    ItemAvailabilityStatus["PARTIALLY_AVAILABLE"] = "PARTIALLY_AVAILABLE";
})(ItemAvailabilityStatus = exports.ItemAvailabilityStatus || (exports.ItemAvailabilityStatus = {}));
var ItemTypeItemType;
(function(ItemTypeItemType) {
    ItemTypeItemType["UNRECOGNISED"] = "UNRECOGNISED";
    ItemTypeItemType["PHYSICAL"] = "PHYSICAL";
    ItemTypeItemType["DIGITAL"] = "DIGITAL";
    ItemTypeItemType["GIFT_CARD"] = "GIFT_CARD";
})(ItemTypeItemType = exports.ItemTypeItemType || (exports.ItemTypeItemType = {}));
var LineItemType;
(function(LineItemType) {
    LineItemType["UNSPECIFIED_LINE_ITEM_TYPE"] = "UNSPECIFIED_LINE_ITEM_TYPE";
    LineItemType["PHYSICAL"] = "PHYSICAL";
    LineItemType["DIGITAL"] = "DIGITAL";
    LineItemType["CUSTOM_AMOUNT_ITEM"] = "CUSTOM_AMOUNT_ITEM";
})(LineItemType = exports.LineItemType || (exports.LineItemType = {}));
var ManualCalculationReason;
(function(ManualCalculationReason) {
    ManualCalculationReason["GLOBAL_SETTING_TO_MANUAL"] = "GLOBAL_SETTING_TO_MANUAL";
    ManualCalculationReason["REGION_SETTING_TO_MANUAL"] = "REGION_SETTING_TO_MANUAL";
})(ManualCalculationReason = exports.ManualCalculationReason || (exports.ManualCalculationReason = {}));
var MediaItemType;
(function(MediaItemType) {
    MediaItemType["UNSPECIFIED_MEDIA_TYPE_ITEM"] = "UNSPECIFIED_MEDIA_TYPE_ITEM";
    MediaItemType["IMAGE"] = "IMAGE";
})(MediaItemType = exports.MediaItemType || (exports.MediaItemType = {}));
var PaymentOptionType;
(function(PaymentOptionType) {
    PaymentOptionType["FULL_PAYMENT_ONLINE"] = "FULL_PAYMENT_ONLINE";
    PaymentOptionType["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
})(PaymentOptionType = exports.PaymentOptionType || (exports.PaymentOptionType = {}));
var PaymentStatus;
(function(PaymentStatus) {
    PaymentStatus["UNSPECIFIED_PAYMENT_STATUS"] = "UNSPECIFIED_PAYMENT_STATUS";
    PaymentStatus["PENDING"] = "PENDING";
    PaymentStatus["NOT_PAID"] = "NOT_PAID";
    PaymentStatus["PAID"] = "PAID";
    PaymentStatus["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
    PaymentStatus["FULLY_REFUNDED"] = "FULLY_REFUNDED";
})(PaymentStatus = exports.PaymentStatus || (exports.PaymentStatus = {}));
var PickupMethod;
(function(PickupMethod) {
    PickupMethod["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
    PickupMethod["STORE_PICKUP"] = "STORE_PICKUP";
    PickupMethod["PICKUP_POINT"] = "PICKUP_POINT";
})(PickupMethod = exports.PickupMethod || (exports.PickupMethod = {}));
var RateType;
(function(RateType) {
    RateType["NO_TAX_COLLECTED"] = "NO_TAX_COLLECTED";
    RateType["MANUAL_RATE"] = "MANUAL_RATE";
    RateType["AUTO_RATE"] = "AUTO_RATE";
    RateType["FALLBACK_RATE"] = "FALLBACK_RATE";
})(RateType = exports.RateType || (exports.RateType = {}));
var RuleType;
(function(RuleType) {
    RuleType["VALIDATION"] = "VALIDATION";
    RuleType["OTHER"] = "OTHER";
    RuleType["MAX"] = "MAX";
    RuleType["MIN"] = "MIN";
    RuleType["MAX_LENGTH"] = "MAX_LENGTH";
    RuleType["MIN_LENGTH"] = "MIN_LENGTH";
    RuleType["MAX_SIZE"] = "MAX_SIZE";
    RuleType["MIN_SIZE"] = "MIN_SIZE";
    RuleType["FORMAT"] = "FORMAT";
    RuleType["DECIMAL_LTE"] = "DECIMAL_LTE";
    RuleType["DECIMAL_GTE"] = "DECIMAL_GTE";
    RuleType["DECIMAL_LT"] = "DECIMAL_LT";
    RuleType["DECIMAL_GT"] = "DECIMAL_GT";
    RuleType["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
    RuleType["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
    RuleType["REQUIRED_FIELD"] = "REQUIRED_FIELD";
})(RuleType = exports.RuleType || (exports.RuleType = {}));
var SubscriptionFrequency;
(function(SubscriptionFrequency) {
    SubscriptionFrequency["UNDEFINED"] = "UNDEFINED";
    SubscriptionFrequency["DAY"] = "DAY";
    SubscriptionFrequency["WEEK"] = "WEEK";
    SubscriptionFrequency["MONTH"] = "MONTH";
    SubscriptionFrequency["YEAR"] = "YEAR";
})(SubscriptionFrequency = exports.SubscriptionFrequency || (exports.SubscriptionFrequency = {}));
var V1IdentityType;
(function(V1IdentityType) {
    V1IdentityType["UNSPECIFIED_IDENTITY_TYPE"] = "UNSPECIFIED_IDENTITY_TYPE";
    V1IdentityType["MEMBER"] = "MEMBER";
    V1IdentityType["CONTACT"] = "CONTACT";
})(V1IdentityType = exports.V1IdentityType || (exports.V1IdentityType = {}));
var V2IdentityType;
(function(V2IdentityType) {
    V2IdentityType["UNSPECIFIED_IDENTITY_TYPE"] = "UNSPECIFIED_IDENTITY_TYPE";
    V2IdentityType["MEMBER"] = "MEMBER";
    V2IdentityType["CONTACT"] = "CONTACT";
})(V2IdentityType = exports.V2IdentityType || (exports.V2IdentityType = {}));
var WeightUnit;
(function(WeightUnit) {
    WeightUnit["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
    WeightUnit["KG"] = "KG";
    WeightUnit["LB"] = "LB";
})(WeightUnit = exports.WeightUnit || (exports.WeightUnit = {}));
