// @wix/advanced-seo-utils/dist/src/private/types/TagSchemas.js
"use strict";
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
var _a, _b;
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.TAG_SCHEMAS = void 0;
var Props_1 = __importDefault(require("./Props"));
var TagTypes_1 = require("./TagTypes");
var Identifiers_1 = require("./Identifiers");
var Validators_1 = require("./Validators");
var setters_1 = require("./setters");
exports.TAG_SCHEMAS = (_a = {},
    _a[Identifiers_1.IDENTIFIERS.TITLE] = {
        type: TagTypes_1.TAG_TYPES.TITLE,
        children: true,
        getValue: function(tag) {
            return tag && tag.children;
        },
        setValue: function(tag, value) {
            if (tag && typeof tag === 'object') {
                tag.children = value;
                return {
                    isValid: true
                };
            }
            return {
                isValid: false
            };
        },
    },
    _a[Identifiers_1.IDENTIFIERS.DESCRIPTION] = buildStandardMeta('description'),
    _a[Identifiers_1.IDENTIFIERS.OG_TITLE] = buildOgMeta('og:title'),
    _a[Identifiers_1.IDENTIFIERS.OG_DESCRIPTION] = buildOgMeta('og:description'),
    _a[Identifiers_1.IDENTIFIERS.OG_IMAGE] = buildOgMeta('og:image'),
    _a[Identifiers_1.IDENTIFIERS.OG_IMAGE_WIDTH] = buildOgMeta('og:image:width'),
    _a[Identifiers_1.IDENTIFIERS.OG_IMAGE_HEIGHT] = buildOgMeta('og:image:height'),
    _a[Identifiers_1.IDENTIFIERS.TWITTER_CARD] = buildTwitterMeta('twitter:card'),
    _a[Identifiers_1.IDENTIFIERS.TWITTER_TITLE] = buildTwitterMeta('twitter:title'),
    _a[Identifiers_1.IDENTIFIERS.TWITTER_DESCRIPTION] = buildTwitterMeta('twitter:description'),
    _a[Identifiers_1.IDENTIFIERS.TWITTER_IMAGE] = buildTwitterMeta('twitter:image'),
    _a[Identifiers_1.IDENTIFIERS.FB_ADMINS] = buildOgMeta('fb:admins'),
    _a[Identifiers_1.IDENTIFIERS.ROBOTS] = buildStandardMeta('robots'),
    _a[Identifiers_1.IDENTIFIERS.STRUCTURED_DATA] = {
        type: TagTypes_1.TAG_TYPES.SCRIPT,
        props: (_b = {}, _b[Props_1.default.TYPE] = 'application/ld+json', _b),
        children: true,
        getValue: function(tag) {
            return tag.children;
        },
        setValue: function(tag, value, enableValidation) {
            var validator = {
                isValid: true
            };
            if (enableValidation) {
                validator = Validators_1.validateJson(value);
            }
            if (validator.isValid) {
                tag.children = value;
            }
            return validator;
        },
    },
    _a[Identifiers_1.IDENTIFIERS.CANONICAL] = buildLink('canonical'),
    _a[Identifiers_1.GENERAL_IDENTIFIERS.LINK] = function(rel) {
        return buildLink(rel);
    },
    _a[Identifiers_1.GENERAL_IDENTIFIERS.SATANDARD_META] = function(name) {
        return buildStandardMeta(name);
    },
    _a[Identifiers_1.GENERAL_IDENTIFIERS.OG_TAG] = function(name) {
        return buildOgMeta(name);
    },
    _a);

function buildOgMeta(name) {
    return buildStandardMeta(name, Props_1.default.PROPERTY);
}

function buildStandardMeta(name, prop) {
    var _a;
    if (prop === void 0) {
        prop = Props_1.default.NAME;
    }
    return {
        type: TagTypes_1.TAG_TYPES.META,
        props: (_a = {}, _a[prop] = name, _a[Props_1.default.CONTENT] = '', _a),
        getValue: function(tag) {
            return tag.props[Props_1.default.CONTENT];
        },
        setValue: function(tag, value) {
            return setters_1.setters[Identifiers_1.GENERAL_IDENTIFIERS.STANDARD_META](tag, value);
        },
    };
}

function buildTwitterMeta(name, prop) {
    return buildStandardMeta(name, prop);
}

function buildLink(rel) {
    var _a;
    return {
        type: TagTypes_1.TAG_TYPES.LINK,
        props: (_a = {}, _a[Props_1.default.REL] = rel, _a[Props_1.default.HREF] = '', _a),
        getValue: function(tag) {
            return tag.props[Props_1.default.HREF];
        },
        setValue: function(tag, value, enableValidation) {
            return setters_1.setters[Identifiers_1.GENERAL_IDENTIFIERS.LINK](tag, value, enableValidation);
        },
    };
}
