// @wix/advanced-seo-utils/dist/src/private/renderer/setters.js
"use strict";
var __assign = (this && this.__assign) || function() {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s)
                if (Object.prototype.hasOwnProperty.call(s, p))
                    t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function(mod) {
    return (mod && mod.__esModule) ? mod : {
        "default": mod
    };
};
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.updateValueByIdentifier = exports.removeTagsByType = exports.setTitle = exports.setSchemas = exports.setMetaTags = exports.setLinks = exports.setIndexable = exports.setExternalRouter = exports.setDescription = exports.setCanonical = exports.addSchema = void 0;
var tags_1 = require("../tags");
var Identifiers_1 = require("../types/Identifiers");
var TagTypes_1 = require("../types/TagTypes");
var Props_1 = __importDefault(require("../types/Props"));
var clone_1 = require("./utils/clone");
var is_multilingual_tag_1 = require("../tags/advanced/is-multilingual-tag");
var robots_tag_1 = require("../tags/robots-tag/robots-tag");
var consts_1 = require("../consts/consts");

function updateValueByIdentifier(data, identifier, value, meta, enableValidation, config) {
    if (data === void 0) {
        data = {};
    }
    var tags = (data && data.tags) || [];
    var updatedTags = tags_1.updateValueByIdentifier(tags, identifier, value, meta, enableValidation, config);
    return __assign(__assign({}, data), {
        tags: updatedTags
    });
}
exports.updateValueByIdentifier = updateValueByIdentifier;

function setTitle(data, title) {
    return updateValueByIdentifier(data, Identifiers_1.IDENTIFIERS.TITLE, title);
}
exports.setTitle = setTitle;

function setMetaTags(data, metaTags) {
    if (!metaTags || !metaTags.length) {
        return data;
    }
    var tags = (data.tags || []).filter(function(tag) {
        return tag.type !== TagTypes_1.TAG_TYPES.META;
    });
    metaTags.forEach(function(metaTag) {
        return tags.push({
            type: TagTypes_1.TAG_TYPES.META,
            props: __assign({}, metaTag),
        });
    });
    return __assign(__assign({}, data), {
        tags: tags
    });
}
exports.setMetaTags = setMetaTags;

function addSchema(data, json) {
    var _a;
    if (!json || typeof json !== 'object' || Array.isArray(json)) {
        return data;
    }
    var result = clone_1.clone(data);
    result.tags.push({
        type: TagTypes_1.TAG_TYPES.SCRIPT,
        props: (_a = {}, _a[Props_1.default.TYPE] = 'application/ld+json', _a),
        children: JSON.stringify(json),
        allowMultiple: true,
    });
    return result;
}
exports.addSchema = addSchema;

function setSchemas(data, schemas) {
    if (!schemas || !schemas.length) {
        return data;
    }
    var result = __assign(__assign({}, data), {
        tags: (data.tags || []).filter(function(tag) {
            return tag.type !== TagTypes_1.TAG_TYPES.SCRIPT;
        })
    });
    schemas.forEach(function(schema) {
        return (result = addSchema(result, schema));
    });
    return result;
}
exports.setSchemas = setSchemas;

function getUniqueLinks(links) {
    var seenLinks = new Set();
    return links
        .reverse()
        .filter(function(link) {
            if (!link.rel || !link.href) {
                return false;
            }
            var duplicate = seenLinks.has(link.rel);
            seenLinks.add(link.rel);
            return !duplicate;
        })
        .reverse();
}

function setLinks(data, links) {
    if (links === void 0) {
        links = [];
    }
    if (!links || !links.length) {
        return data;
    }
    var tags = (data.tags || []).filter(function(tag) {
        return tag.type !== TagTypes_1.TAG_TYPES.LINK;
    });
    getUniqueLinks(links).forEach(function(link) {
        return tags.push({
            type: TagTypes_1.TAG_TYPES.LINK,
            props: __assign({}, link),
        });
    });
    return __assign(__assign({}, data), {
        tags: tags
    });
}
exports.setLinks = setLinks;

function setExternalRouter(data, payload) {
    var tags = (data === null || data === void 0 ? void 0 : data.tags) || [];
    var title = payload.title,
        metaTags = payload.metaTags,
        links = payload.links,
        structuredData = payload.structuredData;
    if ((title === null || title === void 0 ? void 0 : title.constructor.name) === String.name) {
        tags = setTitle({
            tags: tags
        }, title).tags;
    }
    if ((metaTags === null || metaTags === void 0 ? void 0 : metaTags.constructor.name) === Array.name) {
        tags = setMetaTags({
            tags: tags
        }, metaTags).tags;
    }
    if ((links === null || links === void 0 ? void 0 : links.constructor.name) === Array.name) {
        tags = setLinks({
            tags: tags
        }, links).tags;
    }
    if ((structuredData === null || structuredData === void 0 ? void 0 : structuredData.constructor.name) === Array.name) {
        tags = setSchemas({
            tags: tags
        }, structuredData).tags;
    }
    return __assign(__assign({}, data), {
        tags: tags
    });
}
exports.setExternalRouter = setExternalRouter;

function setDescription(data, description) {
    return updateValueByIdentifier(data, Identifiers_1.IDENTIFIERS.DESCRIPTION, description);
}
exports.setDescription = setDescription;

function setCanonical(data, canonical) {
    return updateValueByIdentifier(data, Identifiers_1.IDENTIFIERS.CANONICAL, canonical);
}
exports.setCanonical = setCanonical;

function setIndexable(data, indexable) {
    var robotsTagValue = tags_1.getValueByIdentifier(data, Identifiers_1.IDENTIFIERS.ROBOTS);
    var newRobotsTagValue = indexable ?
        robots_tag_1.getRobotsTagValueWithDirective(robotsTagValue, consts_1.INDEX) :
        robots_tag_1.getRobotsTagValueWithDirective(robotsTagValue, robots_tag_1.ROBOTS_DIRECTIVES.NOINDEX);
    return updateValueByIdentifier(data, Identifiers_1.IDENTIFIERS.ROBOTS, newRobotsTagValue);
}
exports.setIndexable = setIndexable;

function removeTagsByType(data, type) {
    if (!data) {
        return data;
    }
    if (type === TagTypes_1.TAG_TYPES.LINK) {
        return removeLinkTags(data);
    }
    return __assign(__assign({}, data), {
        tags: (data.tags || []).filter(function(tag) {
            return tag.type !== type;
        })
    });
}
exports.removeTagsByType = removeTagsByType;

function removeLinkTags(data) {
    return __assign(__assign({}, data), {
        tags: (data.tags || []).filter(function(tag) {
            return tag.type !== TagTypes_1.TAG_TYPES.LINK || is_multilingual_tag_1.isMultilingualTag(tag);
        })
    });
}
