// @wix/wixstores-platform-common/dist/src/productVariants/productVariants.mapper.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.productVariantsUpdatedMapper = exports.productVariantsParamMapper = exports.productVariantsMapper = void 0;
exports.productVariantsMapper = function(item) {
    return item.variants.map(function(variantData) {
        return ({
            _id: variantData.id,
            choices: variantData.choices,
            variant: variantData.variant ?
                {
                    currency: variantData.variant.priceData.currency,
                    price: variantData.variant.priceData.price,
                    discountedPrice: variantData.variant.priceData.discountedPrice,
                    pricePerUnit: variantData.variant.priceData.pricePerUnit,
                    formattedPrice: variantData.variant.priceData.formatted.price,
                    formattedDiscountedPrice: variantData.variant.priceData.formatted.discountedPrice,
                    formattedPricePerUnit: variantData.variant.priceData.formatted.pricePerUnit,
                    weight: variantData.variant.weight,
                    sku: variantData.variant.sku,
                    visible: variantData.variant.visible,
                } :
                undefined,
        });
    });
};
exports.productVariantsParamMapper = function(item) {
    return ({
        choices: item.choices,
        variantIds: item.variantIds,
        paging: item.paging ?
            {
                limit: item.paging.limit,
                offset: item.paging.skip,
            } :
            undefined,
    });
};
exports.productVariantsUpdatedMapper = function(item) {
    return {
        productId: item.productId,
        variants: item.variants.map(function(variant) {
            return ({
                variantId: variant.variantId,
                choices: variant.choices,
                updatedFields: variant.changedFields,
            });
        }),
    };
};
