// @wix/wixstores-platform-common/dist/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
var collection_mapper_1 = require("./collections/collection.mapper");
Object.defineProperty(exports, "collectionMapper", {
    enumerable: true,
    get: function() {
        return collection_mapper_1.collectionMapper;
    }
});
var collectionEvent_mapper_1 = require("./collections/collectionEvent.mapper");
Object.defineProperty(exports, "collectionEventMapper", {
    enumerable: true,
    get: function() {
        return collectionEvent_mapper_1.collectionEventMapper;
    }
});
Object.defineProperty(exports, "collectionUpdatedEventMapper", {
    enumerable: true,
    get: function() {
        return collectionEvent_mapper_1.collectionUpdatedEventMapper;
    }
});
var productToUpdate_mapper_1 = require("./products/productToUpdate.mapper");
Object.defineProperty(exports, "productToUpdateMapper", {
    enumerable: true,
    get: function() {
        return productToUpdate_mapper_1.productToUpdateMapper;
    }
});
var product_mapper_1 = require("./products/product.mapper");
Object.defineProperty(exports, "productMapper", {
    enumerable: true,
    get: function() {
        return product_mapper_1.productMapper;
    }
});
Object.defineProperty(exports, "productWithVariantsMapper", {
    enumerable: true,
    get: function() {
        return product_mapper_1.productWithVariantsMapper;
    }
});
Object.defineProperty(exports, "productDbDriverMapper", {
    enumerable: true,
    get: function() {
        return product_mapper_1.productDbDriverMapper;
    }
});
var productEvent_mapper_1 = require("./productEvent/productEvent.mapper");
Object.defineProperty(exports, "productEventMapper", {
    enumerable: true,
    get: function() {
        return productEvent_mapper_1.productEventMapper;
    }
});
Object.defineProperty(exports, "productUpdatedEventMapper", {
    enumerable: true,
    get: function() {
        return productEvent_mapper_1.productUpdatedEventMapper;
    }
});
var order_mapper_1 = require("./orders/order.mapper");
Object.defineProperty(exports, "orderMapperDbDriver", {
    enumerable: true,
    get: function() {
        return order_mapper_1.orderMapperDbDriver;
    }
});
Object.defineProperty(exports, "orderMapper", {
    enumerable: true,
    get: function() {
        return order_mapper_1.orderMapper;
    }
});
Object.defineProperty(exports, "createOrderParamsMapper", {
    enumerable: true,
    get: function() {
        return order_mapper_1.createOrderParamsMapper;
    }
});
var coupons_mapper_1 = require("./coupons/coupons.mapper");
Object.defineProperty(exports, "couponMapper", {
    enumerable: true,
    get: function() {
        return coupons_mapper_1.couponMapper;
    }
});
var productOptionsAvailability_mapper_1 = require("./productOptionsAvailability/productOptionsAvailability.mapper");
Object.defineProperty(exports, "productOptionsAvailabilityMapper", {
    enumerable: true,
    get: function() {
        return productOptionsAvailability_mapper_1.productOptionsAvailabilityMapper;
    }
});
var cart_mapper_1 = require("./carts/cart.mapper");
Object.defineProperty(exports, "cartMapper", {
    enumerable: true,
    get: function() {
        return cart_mapper_1.cartMapper;
    }
});
var cart_mapper_client_1 = require("./carts/cart.mapper.client");
Object.defineProperty(exports, "cartMapperClient", {
    enumerable: true,
    get: function() {
        return cart_mapper_client_1.cartMapperClient;
    }
});
var abandonedCart_mapper_1 = require("./abandonedCarts/abandonedCart.mapper");
Object.defineProperty(exports, "abandonedCartMapper", {
    enumerable: true,
    get: function() {
        return abandonedCart_mapper_1.abandonedCartMapper;
    }
});
var productVariants_mapper_1 = require("./productVariants/productVariants.mapper");
Object.defineProperty(exports, "productVariantsMapper", {
    enumerable: true,
    get: function() {
        return productVariants_mapper_1.productVariantsMapper;
    }
});
Object.defineProperty(exports, "productVariantsUpdatedMapper", {
    enumerable: true,
    get: function() {
        return productVariants_mapper_1.productVariantsUpdatedMapper;
    }
});
var inventory_mapper_1 = require("./inventory/inventory.mapper");
Object.defineProperty(exports, "inventoryMapper", {
    enumerable: true,
    get: function() {
        return inventory_mapper_1.inventoryMapper;
    }
});
Object.defineProperty(exports, "inventoryDbDriverMapper", {
    enumerable: true,
    get: function() {
        return inventory_mapper_1.inventoryDbDriverMapper;
    }
});
var media_mapper_1 = require("./Media/media.mapper");
Object.defineProperty(exports, "mediaIdsMapper", {
    enumerable: true,
    get: function() {
        return media_mapper_1.mediaIdsMapper;
    }
});
Object.defineProperty(exports, "updateMediaItemMapper", {
    enumerable: true,
    get: function() {
        return media_mapper_1.updateMediaItemMapper;
    }
});
Object.defineProperty(exports, "mediaChoicesMapper", {
    enumerable: true,
    get: function() {
        return media_mapper_1.mediaChoicesMapper;
    }
});
var variants_mapper_1 = require("./variants/variants.mapper");
Object.defineProperty(exports, "variantMapper", {
    enumerable: true,
    get: function() {
        return variants_mapper_1.variantMapper;
    }
});
