// @wix/bookings-uou-types/dist/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.TimezoneType = exports.BusinessLocationType = exports.SessionOccupancyStatus = exports.ConferencePlatform = exports.WeekDay = exports.PaymentType = exports.ReservedLocationIds = exports.ServiceLocationType = exports.OfferedAsType = exports.ServiceType = void 0;
var catalog_service_dto_1 = require("./catalog-service.dto");
Object.defineProperty(exports, "ServiceType", {
    enumerable: true,
    get: function() {
        return catalog_service_dto_1.ServiceType;
    }
});
Object.defineProperty(exports, "OfferedAsType", {
    enumerable: true,
    get: function() {
        return catalog_service_dto_1.OfferedAsType;
    }
});
var service_location_dto_1 = require("./service-location.dto");
Object.defineProperty(exports, "ServiceLocationType", {
    enumerable: true,
    get: function() {
        return service_location_dto_1.ServiceLocationType;
    }
});
Object.defineProperty(exports, "ReservedLocationIds", {
    enumerable: true,
    get: function() {
        return service_location_dto_1.ReservedLocationIds;
    }
});
var service_payment_dto_1 = require("./service-payment.dto");
Object.defineProperty(exports, "PaymentType", {
    enumerable: true,
    get: function() {
        return service_payment_dto_1.PaymentType;
    }
});
var service_schedule_header_dto_1 = require("./service-schedule-header.dto");
Object.defineProperty(exports, "WeekDay", {
    enumerable: true,
    get: function() {
        return service_schedule_header_dto_1.WeekDay;
    }
});
var service_schedule_policy_dto_1 = require("./service-schedule-policy.dto");
Object.defineProperty(exports, "ConferencePlatform", {
    enumerable: true,
    get: function() {
        return service_schedule_policy_dto_1.ConferencePlatform;
    }
});
var session_dto_1 = require("./session.dto");
Object.defineProperty(exports, "SessionOccupancyStatus", {
    enumerable: true,
    get: function() {
        return session_dto_1.SessionOccupancyStatus;
    }
});
var business_dto_1 = require("./business.dto");
Object.defineProperty(exports, "BusinessLocationType", {
    enumerable: true,
    get: function() {
        return business_dto_1.BusinessLocationType;
    }
});
Object.defineProperty(exports, "TimezoneType", {
    enumerable: true,
    get: function() {
        return business_dto_1.TimezoneType;
    }
});
